<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2020-05-14" />

<title>Merge samples</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">humanCardiacFibroblasts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Merge samples</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">14 May 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>humanCardiacFibroblasts/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210903code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210903)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210903code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210903)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststreed9e1a985b145083b335fcb581ada30ab27f1bc59targetblankd9e1a98a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/d9e1a985b145083b335fcb581ada30ab27f1bc59" target="_blank">d9e1a98</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststreed9e1a985b145083b335fcb581ada30ab27f1bc59targetblankd9e1a98a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/d9e1a985b145083b335fcb581ada30ab27f1bc59" target="_blank">d9e1a98</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/humanFibroblast/

Untracked files:
    Untracked:  analysis/assignLabelshumanHeartsPlusGrazInt.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(SingleCellExperiment)
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(ggsci)
  library(ggpubr)
  library(pheatmap)
  library(viridis)
  library(sctransform)
})</code></pre>
</div>
<div id="load-data" class="section level2">
<h2>load data</h2>
<pre class="r"><code>basedir &lt;- here()
seurat &lt;- readRDS(file = paste0(basedir, 
                              &quot;/data/humanHeartsPlusGraz_intPatients_merged&quot;, 
                              &quot;_seurat.rds&quot;))
Idents(seurat) &lt;- seurat$seurat_clusters</code></pre>
</div>
<div id="assign-labels" class="section level2">
<h2>assign labels</h2>
<pre class="r"><code>seurat$label &lt;- &quot;other&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;2&quot;,&quot;7&quot;,&quot;9&quot;,&quot;10&quot;))] &lt;- &quot;Endothelial&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;8&quot;))] &lt;- &quot;Tcell&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;3&quot;))] &lt;- &quot;Cardiomyocyte&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;0&quot;))] &lt;- &quot;Fibroblast&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;1&quot;,&quot;6&quot;))] &lt;- &quot;Perivascular&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;4&quot;))] &lt;- &quot;MonocyteMacrophage&quot;</code></pre>
</div>
<div id="color-vectors" class="section level2">
<h2>color vectors</h2>
<pre class="r"><code>colPal &lt;- pal_igv()(length(levels(seurat)))
colTec &lt;- pal_jama()(length(unique(seurat$technique)))
colSmp &lt;- c(pal_uchicago()(8), pal_npg()(8), pal_aaas()(10))[1:length(unique(seurat$dataset))]
colCond &lt;- pal_npg()(length(unique(seurat$cond)))
colID &lt;- c(pal_jco()(10), pal_npg()(10), pal_jama()(10))[1:length(unique(seurat$ID))]
colOrig &lt;- pal_aaas()(length(unique(seurat$origin)))
colIso &lt;- pal_nejm()(length(unique(seurat$isolation)))
colProc &lt;- pal_aaas()(length(unique(seurat$processing)))
colLab &lt;- pal_futurama()(length(unique(seurat$label)))

names(colPal) &lt;- levels(seurat)
names(colTec) &lt;- unique(seurat$technique)
names(colSmp) &lt;- unique(seurat$dataset)
names(colCond) &lt;- unique(seurat$cond)
names(colID) &lt;- unique(seurat$ID)
names(colOrig) &lt;- unique(seurat$origin)
names(colIso) &lt;- unique(seurat$isolation)
names(colProc) &lt;- unique(seurat$processing)
names(colLab) &lt;- unique(seurat$label)</code></pre>
</div>
<div id="vis-data" class="section level2 tabset">
<h2 class="tabset">vis data</h2>
<div id="clusters" class="section level3">
<h3>clusters</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="label" class="section level3">
<h3>label</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;,  group.by = &quot;label&quot;, cols=colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20label-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="technique" class="section level3">
<h3>technique</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;technique&quot;, cols=colTec)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20tecnique-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sample" class="section level3">
<h3>Sample</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;dataset&quot;, cols=colSmp)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20sample-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="id" class="section level3">
<h3>ID</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;ID&quot;, cols=colID)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20ID-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="origin" class="section level3">
<h3>Origin</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;origin&quot;, cols=colOrig)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20origin-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="isolation" class="section level3">
<h3>isolation</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;isolation&quot;, cols=colIso)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20isolation-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cond" class="section level3">
<h3>cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;cond&quot;, cols=colCond)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20cond-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="processing" class="section level3">
<h3>processing</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;processing&quot;, cols=colProc)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20proc-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cnt-tables" class="section level2 tabset">
<h2 class="tabset">cnt tables</h2>
<div id="per-patient" class="section level3">
<h3>per patient</h3>
<pre class="r"><code>## total cells per patient
knitr::kable(table(seurat$ID))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GZ1</td>
<td align="right">2740</td>
</tr>
<tr class="even">
<td align="left">GZ10</td>
<td align="right">3731</td>
</tr>
<tr class="odd">
<td align="left">GZ11</td>
<td align="right">3991</td>
</tr>
<tr class="even">
<td align="left">GZ12</td>
<td align="right">3818</td>
</tr>
<tr class="odd">
<td align="left">GZ2</td>
<td align="right">1684</td>
</tr>
<tr class="even">
<td align="left">GZ3</td>
<td align="right">2396</td>
</tr>
<tr class="odd">
<td align="left">GZ4</td>
<td align="right">545</td>
</tr>
<tr class="even">
<td align="left">GZ5</td>
<td align="right">781</td>
</tr>
<tr class="odd">
<td align="left">GZ6</td>
<td align="right">491</td>
</tr>
<tr class="even">
<td align="left">GZ7</td>
<td align="right">653</td>
</tr>
<tr class="odd">
<td align="left">GZ8</td>
<td align="right">3921</td>
</tr>
<tr class="even">
<td align="left">SG16</td>
<td align="right">1321</td>
</tr>
<tr class="odd">
<td align="left">SG20</td>
<td align="right">1536</td>
</tr>
<tr class="even">
<td align="left">SG21</td>
<td align="right">826</td>
</tr>
<tr class="odd">
<td align="left">SG2224</td>
<td align="right">2381</td>
</tr>
<tr class="even">
<td align="left">SG2325</td>
<td align="right">3131</td>
</tr>
<tr class="odd">
<td align="left">SG26</td>
<td align="right">238</td>
</tr>
<tr class="even">
<td align="left">SG27</td>
<td align="right">1545</td>
</tr>
<tr class="odd">
<td align="left">SG28</td>
<td align="right">1254</td>
</tr>
<tr class="even">
<td align="left">SG29</td>
<td align="right">1242</td>
</tr>
<tr class="odd">
<td align="left">SG30</td>
<td align="right">236</td>
</tr>
<tr class="even">
<td align="left">SG31</td>
<td align="right">1192</td>
</tr>
<tr class="odd">
<td align="left">SG32</td>
<td align="right">1428</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype per patient counts
knitr::kable(table(seurat$label, seurat$ID))</code></pre>
<table style="width:100%;">
<colgroup>
<col width="14%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="5%" />
<col width="5%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">GZ1</th>
<th align="right">GZ10</th>
<th align="right">GZ11</th>
<th align="right">GZ12</th>
<th align="right">GZ2</th>
<th align="right">GZ3</th>
<th align="right">GZ4</th>
<th align="right">GZ5</th>
<th align="right">GZ6</th>
<th align="right">GZ7</th>
<th align="right">GZ8</th>
<th align="right">SG16</th>
<th align="right">SG20</th>
<th align="right">SG21</th>
<th align="right">SG2224</th>
<th align="right">SG2325</th>
<th align="right">SG26</th>
<th align="right">SG27</th>
<th align="right">SG28</th>
<th align="right">SG29</th>
<th align="right">SG30</th>
<th align="right">SG31</th>
<th align="right">SG32</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="right">207</td>
<td align="right">1094</td>
<td align="right">298</td>
<td align="right">114</td>
<td align="right">106</td>
<td align="right">233</td>
<td align="right">44</td>
<td align="right">38</td>
<td align="right">12</td>
<td align="right">88</td>
<td align="right">710</td>
<td align="right">21</td>
<td align="right">96</td>
<td align="right">41</td>
<td align="right">295</td>
<td align="right">396</td>
<td align="right">16</td>
<td align="right">70</td>
<td align="right">78</td>
<td align="right">93</td>
<td align="right">46</td>
<td align="right">178</td>
<td align="right">40</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="right">446</td>
<td align="right">398</td>
<td align="right">805</td>
<td align="right">1140</td>
<td align="right">422</td>
<td align="right">568</td>
<td align="right">183</td>
<td align="right">239</td>
<td align="right">153</td>
<td align="right">121</td>
<td align="right">773</td>
<td align="right">103</td>
<td align="right">223</td>
<td align="right">263</td>
<td align="right">455</td>
<td align="right">637</td>
<td align="right">68</td>
<td align="right">552</td>
<td align="right">424</td>
<td align="right">163</td>
<td align="right">73</td>
<td align="right">253</td>
<td align="right">584</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="right">760</td>
<td align="right">982</td>
<td align="right">809</td>
<td align="right">1106</td>
<td align="right">637</td>
<td align="right">819</td>
<td align="right">140</td>
<td align="right">260</td>
<td align="right">150</td>
<td align="right">231</td>
<td align="right">1136</td>
<td align="right">142</td>
<td align="right">606</td>
<td align="right">234</td>
<td align="right">829</td>
<td align="right">774</td>
<td align="right">79</td>
<td align="right">304</td>
<td align="right">256</td>
<td align="right">145</td>
<td align="right">39</td>
<td align="right">347</td>
<td align="right">313</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="right">291</td>
<td align="right">463</td>
<td align="right">588</td>
<td align="right">219</td>
<td align="right">79</td>
<td align="right">115</td>
<td align="right">56</td>
<td align="right">31</td>
<td align="right">61</td>
<td align="right">16</td>
<td align="right">342</td>
<td align="right">13</td>
<td align="right">153</td>
<td align="right">40</td>
<td align="right">162</td>
<td align="right">230</td>
<td align="right">7</td>
<td align="right">13</td>
<td align="right">38</td>
<td align="right">259</td>
<td align="right">14</td>
<td align="right">126</td>
<td align="right">65</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="right">131</td>
<td align="right">183</td>
<td align="right">161</td>
<td align="right">151</td>
<td align="right">91</td>
<td align="right">123</td>
<td align="right">11</td>
<td align="right">42</td>
<td align="right">13</td>
<td align="right">71</td>
<td align="right">153</td>
<td align="right">976</td>
<td align="right">22</td>
<td align="right">27</td>
<td align="right">137</td>
<td align="right">164</td>
<td align="right">3</td>
<td align="right">65</td>
<td align="right">58</td>
<td align="right">428</td>
<td align="right">7</td>
<td align="right">33</td>
<td align="right">55</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="right">688</td>
<td align="right">497</td>
<td align="right">1202</td>
<td align="right">1051</td>
<td align="right">313</td>
<td align="right">502</td>
<td align="right">85</td>
<td align="right">154</td>
<td align="right">70</td>
<td align="right">121</td>
<td align="right">733</td>
<td align="right">63</td>
<td align="right">392</td>
<td align="right">204</td>
<td align="right">400</td>
<td align="right">574</td>
<td align="right">62</td>
<td align="right">523</td>
<td align="right">363</td>
<td align="right">78</td>
<td align="right">57</td>
<td align="right">232</td>
<td align="right">313</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="right">217</td>
<td align="right">114</td>
<td align="right">128</td>
<td align="right">37</td>
<td align="right">36</td>
<td align="right">36</td>
<td align="right">26</td>
<td align="right">17</td>
<td align="right">32</td>
<td align="right">5</td>
<td align="right">74</td>
<td align="right">3</td>
<td align="right">44</td>
<td align="right">17</td>
<td align="right">103</td>
<td align="right">356</td>
<td align="right">3</td>
<td align="right">18</td>
<td align="right">37</td>
<td align="right">76</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">58</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype percentages per patient
datLab &lt;- data.frame(table(seurat$label, seurat$ID))
colnames(datLab) &lt;- c(&quot;label&quot;, &quot;ID&quot;, &quot;cnt&quot;)
datPat &lt;- data.frame(table(seurat$ID))
colnames(datPat) &lt;- c(&quot;ID&quot;, &quot;total&quot;)
datFrac &lt;- datLab %&gt;% left_join(., datPat, by=&quot;ID&quot;) %&gt;% 
  mutate(percentage = cnt*100/total)
knitr::kable(datFrac)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">label</th>
<th align="left">ID</th>
<th align="right">cnt</th>
<th align="right">total</th>
<th align="right">percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ1</td>
<td align="right">207</td>
<td align="right">2740</td>
<td align="right">7.5547445</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ1</td>
<td align="right">446</td>
<td align="right">2740</td>
<td align="right">16.2773723</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ1</td>
<td align="right">760</td>
<td align="right">2740</td>
<td align="right">27.7372263</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ1</td>
<td align="right">291</td>
<td align="right">2740</td>
<td align="right">10.6204380</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ1</td>
<td align="right">131</td>
<td align="right">2740</td>
<td align="right">4.7810219</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ1</td>
<td align="right">688</td>
<td align="right">2740</td>
<td align="right">25.1094891</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ1</td>
<td align="right">217</td>
<td align="right">2740</td>
<td align="right">7.9197080</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ10</td>
<td align="right">1094</td>
<td align="right">3731</td>
<td align="right">29.3218976</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ10</td>
<td align="right">398</td>
<td align="right">3731</td>
<td align="right">10.6673814</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ10</td>
<td align="right">982</td>
<td align="right">3731</td>
<td align="right">26.3200214</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ10</td>
<td align="right">463</td>
<td align="right">3731</td>
<td align="right">12.4095417</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ10</td>
<td align="right">183</td>
<td align="right">3731</td>
<td align="right">4.9048512</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ10</td>
<td align="right">497</td>
<td align="right">3731</td>
<td align="right">13.3208255</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ10</td>
<td align="right">114</td>
<td align="right">3731</td>
<td align="right">3.0554811</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ11</td>
<td align="right">298</td>
<td align="right">3991</td>
<td align="right">7.4668003</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ11</td>
<td align="right">805</td>
<td align="right">3991</td>
<td align="right">20.1703834</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ11</td>
<td align="right">809</td>
<td align="right">3991</td>
<td align="right">20.2706089</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ11</td>
<td align="right">588</td>
<td align="right">3991</td>
<td align="right">14.7331496</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ11</td>
<td align="right">161</td>
<td align="right">3991</td>
<td align="right">4.0340767</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ11</td>
<td align="right">1202</td>
<td align="right">3991</td>
<td align="right">30.1177650</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ11</td>
<td align="right">128</td>
<td align="right">3991</td>
<td align="right">3.2072162</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ12</td>
<td align="right">114</td>
<td align="right">3818</td>
<td align="right">2.9858565</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ12</td>
<td align="right">1140</td>
<td align="right">3818</td>
<td align="right">29.8585647</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ12</td>
<td align="right">1106</td>
<td align="right">3818</td>
<td align="right">28.9680461</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ12</td>
<td align="right">219</td>
<td align="right">3818</td>
<td align="right">5.7359874</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ12</td>
<td align="right">151</td>
<td align="right">3818</td>
<td align="right">3.9549502</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ12</td>
<td align="right">1051</td>
<td align="right">3818</td>
<td align="right">27.5275013</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ12</td>
<td align="right">37</td>
<td align="right">3818</td>
<td align="right">0.9690938</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ2</td>
<td align="right">106</td>
<td align="right">1684</td>
<td align="right">6.2945368</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ2</td>
<td align="right">422</td>
<td align="right">1684</td>
<td align="right">25.0593824</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ2</td>
<td align="right">637</td>
<td align="right">1684</td>
<td align="right">37.8266033</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ2</td>
<td align="right">79</td>
<td align="right">1684</td>
<td align="right">4.6912114</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ2</td>
<td align="right">91</td>
<td align="right">1684</td>
<td align="right">5.4038005</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ2</td>
<td align="right">313</td>
<td align="right">1684</td>
<td align="right">18.5866983</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ2</td>
<td align="right">36</td>
<td align="right">1684</td>
<td align="right">2.1377672</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ3</td>
<td align="right">233</td>
<td align="right">2396</td>
<td align="right">9.7245409</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ3</td>
<td align="right">568</td>
<td align="right">2396</td>
<td align="right">23.7061770</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ3</td>
<td align="right">819</td>
<td align="right">2396</td>
<td align="right">34.1819699</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ3</td>
<td align="right">115</td>
<td align="right">2396</td>
<td align="right">4.7996661</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ3</td>
<td align="right">123</td>
<td align="right">2396</td>
<td align="right">5.1335559</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ3</td>
<td align="right">502</td>
<td align="right">2396</td>
<td align="right">20.9515860</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ3</td>
<td align="right">36</td>
<td align="right">2396</td>
<td align="right">1.5025042</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ4</td>
<td align="right">44</td>
<td align="right">545</td>
<td align="right">8.0733945</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ4</td>
<td align="right">183</td>
<td align="right">545</td>
<td align="right">33.5779817</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ4</td>
<td align="right">140</td>
<td align="right">545</td>
<td align="right">25.6880734</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ4</td>
<td align="right">56</td>
<td align="right">545</td>
<td align="right">10.2752294</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ4</td>
<td align="right">11</td>
<td align="right">545</td>
<td align="right">2.0183486</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ4</td>
<td align="right">85</td>
<td align="right">545</td>
<td align="right">15.5963303</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ4</td>
<td align="right">26</td>
<td align="right">545</td>
<td align="right">4.7706422</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ5</td>
<td align="right">38</td>
<td align="right">781</td>
<td align="right">4.8655570</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ5</td>
<td align="right">239</td>
<td align="right">781</td>
<td align="right">30.6017926</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ5</td>
<td align="right">260</td>
<td align="right">781</td>
<td align="right">33.2906530</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ5</td>
<td align="right">31</td>
<td align="right">781</td>
<td align="right">3.9692702</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ5</td>
<td align="right">42</td>
<td align="right">781</td>
<td align="right">5.3777209</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ5</td>
<td align="right">154</td>
<td align="right">781</td>
<td align="right">19.7183099</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ5</td>
<td align="right">17</td>
<td align="right">781</td>
<td align="right">2.1766965</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ6</td>
<td align="right">12</td>
<td align="right">491</td>
<td align="right">2.4439919</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ6</td>
<td align="right">153</td>
<td align="right">491</td>
<td align="right">31.1608961</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ6</td>
<td align="right">150</td>
<td align="right">491</td>
<td align="right">30.5498982</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ6</td>
<td align="right">61</td>
<td align="right">491</td>
<td align="right">12.4236253</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ6</td>
<td align="right">13</td>
<td align="right">491</td>
<td align="right">2.6476578</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ6</td>
<td align="right">70</td>
<td align="right">491</td>
<td align="right">14.2566191</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ6</td>
<td align="right">32</td>
<td align="right">491</td>
<td align="right">6.5173116</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ7</td>
<td align="right">88</td>
<td align="right">653</td>
<td align="right">13.4762634</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ7</td>
<td align="right">121</td>
<td align="right">653</td>
<td align="right">18.5298622</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ7</td>
<td align="right">231</td>
<td align="right">653</td>
<td align="right">35.3751914</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ7</td>
<td align="right">16</td>
<td align="right">653</td>
<td align="right">2.4502297</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ7</td>
<td align="right">71</td>
<td align="right">653</td>
<td align="right">10.8728943</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ7</td>
<td align="right">121</td>
<td align="right">653</td>
<td align="right">18.5298622</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ7</td>
<td align="right">5</td>
<td align="right">653</td>
<td align="right">0.7656968</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ8</td>
<td align="right">710</td>
<td align="right">3921</td>
<td align="right">18.1076256</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ8</td>
<td align="right">773</td>
<td align="right">3921</td>
<td align="right">19.7143586</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ8</td>
<td align="right">1136</td>
<td align="right">3921</td>
<td align="right">28.9722010</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ8</td>
<td align="right">342</td>
<td align="right">3921</td>
<td align="right">8.7222647</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ8</td>
<td align="right">153</td>
<td align="right">3921</td>
<td align="right">3.9020658</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ8</td>
<td align="right">733</td>
<td align="right">3921</td>
<td align="right">18.6942107</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ8</td>
<td align="right">74</td>
<td align="right">3921</td>
<td align="right">1.8872737</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG16</td>
<td align="right">21</td>
<td align="right">1321</td>
<td align="right">1.5897048</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">SG16</td>
<td align="right">103</td>
<td align="right">1321</td>
<td align="right">7.7971234</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">SG16</td>
<td align="right">142</td>
<td align="right">1321</td>
<td align="right">10.7494322</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG16</td>
<td align="right">13</td>
<td align="right">1321</td>
<td align="right">0.9841030</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">SG16</td>
<td align="right">976</td>
<td align="right">1321</td>
<td align="right">73.8834217</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG16</td>
<td align="right">63</td>
<td align="right">1321</td>
<td align="right">4.7691143</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG16</td>
<td align="right">3</td>
<td align="right">1321</td>
<td align="right">0.2271007</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG20</td>
<td align="right">96</td>
<td align="right">1536</td>
<td align="right">6.2500000</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG20</td>
<td align="right">223</td>
<td align="right">1536</td>
<td align="right">14.5182292</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG20</td>
<td align="right">606</td>
<td align="right">1536</td>
<td align="right">39.4531250</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG20</td>
<td align="right">153</td>
<td align="right">1536</td>
<td align="right">9.9609375</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG20</td>
<td align="right">22</td>
<td align="right">1536</td>
<td align="right">1.4322917</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG20</td>
<td align="right">392</td>
<td align="right">1536</td>
<td align="right">25.5208333</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG20</td>
<td align="right">44</td>
<td align="right">1536</td>
<td align="right">2.8645833</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG21</td>
<td align="right">41</td>
<td align="right">826</td>
<td align="right">4.9636804</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">SG21</td>
<td align="right">263</td>
<td align="right">826</td>
<td align="right">31.8401937</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">SG21</td>
<td align="right">234</td>
<td align="right">826</td>
<td align="right">28.3292978</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG21</td>
<td align="right">40</td>
<td align="right">826</td>
<td align="right">4.8426150</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">SG21</td>
<td align="right">27</td>
<td align="right">826</td>
<td align="right">3.2687651</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG21</td>
<td align="right">204</td>
<td align="right">826</td>
<td align="right">24.6973366</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG21</td>
<td align="right">17</td>
<td align="right">826</td>
<td align="right">2.0581114</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG2224</td>
<td align="right">295</td>
<td align="right">2381</td>
<td align="right">12.3897522</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG2224</td>
<td align="right">455</td>
<td align="right">2381</td>
<td align="right">19.1096178</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG2224</td>
<td align="right">829</td>
<td align="right">2381</td>
<td align="right">34.8173037</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG2224</td>
<td align="right">162</td>
<td align="right">2381</td>
<td align="right">6.8038639</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG2224</td>
<td align="right">137</td>
<td align="right">2381</td>
<td align="right">5.7538849</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG2224</td>
<td align="right">400</td>
<td align="right">2381</td>
<td align="right">16.7996640</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG2224</td>
<td align="right">103</td>
<td align="right">2381</td>
<td align="right">4.3259135</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG2325</td>
<td align="right">396</td>
<td align="right">3131</td>
<td align="right">12.6477164</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">SG2325</td>
<td align="right">637</td>
<td align="right">3131</td>
<td align="right">20.3449377</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">SG2325</td>
<td align="right">774</td>
<td align="right">3131</td>
<td align="right">24.7205366</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG2325</td>
<td align="right">230</td>
<td align="right">3131</td>
<td align="right">7.3458959</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">SG2325</td>
<td align="right">164</td>
<td align="right">3131</td>
<td align="right">5.2379431</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG2325</td>
<td align="right">574</td>
<td align="right">3131</td>
<td align="right">18.3328010</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG2325</td>
<td align="right">356</td>
<td align="right">3131</td>
<td align="right">11.3701693</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG26</td>
<td align="right">16</td>
<td align="right">238</td>
<td align="right">6.7226891</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG26</td>
<td align="right">68</td>
<td align="right">238</td>
<td align="right">28.5714286</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG26</td>
<td align="right">79</td>
<td align="right">238</td>
<td align="right">33.1932773</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG26</td>
<td align="right">7</td>
<td align="right">238</td>
<td align="right">2.9411765</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG26</td>
<td align="right">3</td>
<td align="right">238</td>
<td align="right">1.2605042</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG26</td>
<td align="right">62</td>
<td align="right">238</td>
<td align="right">26.0504202</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG26</td>
<td align="right">3</td>
<td align="right">238</td>
<td align="right">1.2605042</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG27</td>
<td align="right">70</td>
<td align="right">1545</td>
<td align="right">4.5307443</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">SG27</td>
<td align="right">552</td>
<td align="right">1545</td>
<td align="right">35.7281553</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">SG27</td>
<td align="right">304</td>
<td align="right">1545</td>
<td align="right">19.6763754</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG27</td>
<td align="right">13</td>
<td align="right">1545</td>
<td align="right">0.8414239</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">SG27</td>
<td align="right">65</td>
<td align="right">1545</td>
<td align="right">4.2071197</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG27</td>
<td align="right">523</td>
<td align="right">1545</td>
<td align="right">33.8511327</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG27</td>
<td align="right">18</td>
<td align="right">1545</td>
<td align="right">1.1650485</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG28</td>
<td align="right">78</td>
<td align="right">1254</td>
<td align="right">6.2200957</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG28</td>
<td align="right">424</td>
<td align="right">1254</td>
<td align="right">33.8118022</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG28</td>
<td align="right">256</td>
<td align="right">1254</td>
<td align="right">20.4146730</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG28</td>
<td align="right">38</td>
<td align="right">1254</td>
<td align="right">3.0303030</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG28</td>
<td align="right">58</td>
<td align="right">1254</td>
<td align="right">4.6251994</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG28</td>
<td align="right">363</td>
<td align="right">1254</td>
<td align="right">28.9473684</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG28</td>
<td align="right">37</td>
<td align="right">1254</td>
<td align="right">2.9505582</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG29</td>
<td align="right">93</td>
<td align="right">1242</td>
<td align="right">7.4879227</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">SG29</td>
<td align="right">163</td>
<td align="right">1242</td>
<td align="right">13.1239936</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">SG29</td>
<td align="right">145</td>
<td align="right">1242</td>
<td align="right">11.6747182</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG29</td>
<td align="right">259</td>
<td align="right">1242</td>
<td align="right">20.8534622</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">SG29</td>
<td align="right">428</td>
<td align="right">1242</td>
<td align="right">34.4605475</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG29</td>
<td align="right">78</td>
<td align="right">1242</td>
<td align="right">6.2801932</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG29</td>
<td align="right">76</td>
<td align="right">1242</td>
<td align="right">6.1191626</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG30</td>
<td align="right">46</td>
<td align="right">236</td>
<td align="right">19.4915254</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG30</td>
<td align="right">73</td>
<td align="right">236</td>
<td align="right">30.9322034</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG30</td>
<td align="right">39</td>
<td align="right">236</td>
<td align="right">16.5254237</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG30</td>
<td align="right">14</td>
<td align="right">236</td>
<td align="right">5.9322034</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG30</td>
<td align="right">7</td>
<td align="right">236</td>
<td align="right">2.9661017</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG30</td>
<td align="right">57</td>
<td align="right">236</td>
<td align="right">24.1525424</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG30</td>
<td align="right">0</td>
<td align="right">236</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG31</td>
<td align="right">178</td>
<td align="right">1192</td>
<td align="right">14.9328859</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">SG31</td>
<td align="right">253</td>
<td align="right">1192</td>
<td align="right">21.2248322</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">SG31</td>
<td align="right">347</td>
<td align="right">1192</td>
<td align="right">29.1107383</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG31</td>
<td align="right">126</td>
<td align="right">1192</td>
<td align="right">10.5704698</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">SG31</td>
<td align="right">33</td>
<td align="right">1192</td>
<td align="right">2.7684564</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG31</td>
<td align="right">232</td>
<td align="right">1192</td>
<td align="right">19.4630872</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG31</td>
<td align="right">23</td>
<td align="right">1192</td>
<td align="right">1.9295302</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG32</td>
<td align="right">40</td>
<td align="right">1428</td>
<td align="right">2.8011204</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG32</td>
<td align="right">584</td>
<td align="right">1428</td>
<td align="right">40.8963585</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG32</td>
<td align="right">313</td>
<td align="right">1428</td>
<td align="right">21.9187675</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG32</td>
<td align="right">65</td>
<td align="right">1428</td>
<td align="right">4.5518207</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG32</td>
<td align="right">55</td>
<td align="right">1428</td>
<td align="right">3.8515406</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG32</td>
<td align="right">313</td>
<td align="right">1428</td>
<td align="right">21.9187675</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG32</td>
<td align="right">58</td>
<td align="right">1428</td>
<td align="right">4.0616246</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ggbarplot(datFrac, x=&quot;ID&quot;, y=&quot;percentage&quot;,
          fill = &quot;label&quot;,
          palette = colLab) +
  rotate_x_text(angle = 90)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/cnt%20tables%20per%20pat-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="per-cond" class="section level3">
<h3>per cond</h3>
<pre class="r"><code>## total cells per cond
knitr::kable(table(seurat$cond))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HF</td>
<td align="right">1064</td>
</tr>
<tr class="even">
<td align="left">HH</td>
<td align="right">15461</td>
</tr>
<tr class="odd">
<td align="left">InfCardiomyopathy</td>
<td align="right">1428</td>
</tr>
<tr class="even">
<td align="left">inflamation</td>
<td align="right">236</td>
</tr>
<tr class="odd">
<td align="left">Myocarditis</td>
<td align="right">10532</td>
</tr>
<tr class="even">
<td align="left">Perimyocarditis</td>
<td align="right">1192</td>
</tr>
<tr class="odd">
<td align="left">transplant</td>
<td align="right">11168</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype per cond counts
knitr::kable(table(seurat$label, seurat$cond))</code></pre>
<table>
<colgroup>
<col width="19%" />
<col width="4%" />
<col width="5%" />
<col width="18%" />
<col width="12%" />
<col width="12%" />
<col width="16%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">HF</th>
<th align="right">HH</th>
<th align="right">InfCardiomyopathy</th>
<th align="right">inflamation</th>
<th align="right">Myocarditis</th>
<th align="right">Perimyocarditis</th>
<th align="right">transplant</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="right">57</td>
<td align="right">2216</td>
<td align="right">40</td>
<td align="right">46</td>
<td align="right">821</td>
<td align="right">178</td>
<td align="right">956</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="right">331</td>
<td align="right">3116</td>
<td align="right">584</td>
<td align="right">73</td>
<td align="right">2295</td>
<td align="right">253</td>
<td align="right">2394</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="right">313</td>
<td align="right">4033</td>
<td align="right">313</td>
<td align="right">39</td>
<td align="right">3142</td>
<td align="right">347</td>
<td align="right">2911</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="right">47</td>
<td align="right">1612</td>
<td align="right">65</td>
<td align="right">14</td>
<td align="right">908</td>
<td align="right">126</td>
<td align="right">609</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="right">30</td>
<td align="right">648</td>
<td align="right">55</td>
<td align="right">7</td>
<td align="right">910</td>
<td align="right">33</td>
<td align="right">1422</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="right">266</td>
<td align="right">3483</td>
<td align="right">313</td>
<td align="right">57</td>
<td align="right">2011</td>
<td align="right">232</td>
<td align="right">2315</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="right">20</td>
<td align="right">353</td>
<td align="right">58</td>
<td align="right">0</td>
<td align="right">445</td>
<td align="right">23</td>
<td align="right">561</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype percentages per cond
datLab &lt;- data.frame(table(seurat$label, seurat$cond))
colnames(datLab) &lt;- c(&quot;label&quot;, &quot;cond&quot;, &quot;cnt&quot;)
datPat &lt;- data.frame(table(seurat$cond))
colnames(datPat) &lt;- c(&quot;cond&quot;, &quot;total&quot;)
datFrac &lt;- datLab %&gt;% left_join(., datPat, by=&quot;cond&quot;) %&gt;% 
  mutate(percentage = cnt*100/total)
knitr::kable(datFrac)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">label</th>
<th align="left">cond</th>
<th align="right">cnt</th>
<th align="right">total</th>
<th align="right">percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">HF</td>
<td align="right">57</td>
<td align="right">1064</td>
<td align="right">5.357143</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">HF</td>
<td align="right">331</td>
<td align="right">1064</td>
<td align="right">31.109023</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">HF</td>
<td align="right">313</td>
<td align="right">1064</td>
<td align="right">29.417293</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">HF</td>
<td align="right">47</td>
<td align="right">1064</td>
<td align="right">4.417293</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">HF</td>
<td align="right">30</td>
<td align="right">1064</td>
<td align="right">2.819549</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">HF</td>
<td align="right">266</td>
<td align="right">1064</td>
<td align="right">25.000000</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">HF</td>
<td align="right">20</td>
<td align="right">1064</td>
<td align="right">1.879699</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">HH</td>
<td align="right">2216</td>
<td align="right">15461</td>
<td align="right">14.332838</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">HH</td>
<td align="right">3116</td>
<td align="right">15461</td>
<td align="right">20.153936</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">HH</td>
<td align="right">4033</td>
<td align="right">15461</td>
<td align="right">26.084988</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">HH</td>
<td align="right">1612</td>
<td align="right">15461</td>
<td align="right">10.426234</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">HH</td>
<td align="right">648</td>
<td align="right">15461</td>
<td align="right">4.191191</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">HH</td>
<td align="right">3483</td>
<td align="right">15461</td>
<td align="right">22.527650</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">HH</td>
<td align="right">353</td>
<td align="right">15461</td>
<td align="right">2.283164</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">40</td>
<td align="right">1428</td>
<td align="right">2.801120</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">584</td>
<td align="right">1428</td>
<td align="right">40.896358</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">313</td>
<td align="right">1428</td>
<td align="right">21.918768</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">65</td>
<td align="right">1428</td>
<td align="right">4.551821</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">55</td>
<td align="right">1428</td>
<td align="right">3.851541</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">313</td>
<td align="right">1428</td>
<td align="right">21.918768</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">58</td>
<td align="right">1428</td>
<td align="right">4.061625</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">inflamation</td>
<td align="right">46</td>
<td align="right">236</td>
<td align="right">19.491525</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">inflamation</td>
<td align="right">73</td>
<td align="right">236</td>
<td align="right">30.932203</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">inflamation</td>
<td align="right">39</td>
<td align="right">236</td>
<td align="right">16.525424</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">inflamation</td>
<td align="right">14</td>
<td align="right">236</td>
<td align="right">5.932203</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">inflamation</td>
<td align="right">7</td>
<td align="right">236</td>
<td align="right">2.966102</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">inflamation</td>
<td align="right">57</td>
<td align="right">236</td>
<td align="right">24.152542</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">inflamation</td>
<td align="right">0</td>
<td align="right">236</td>
<td align="right">0.000000</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">Myocarditis</td>
<td align="right">821</td>
<td align="right">10532</td>
<td align="right">7.795291</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">Myocarditis</td>
<td align="right">2295</td>
<td align="right">10532</td>
<td align="right">21.790733</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">Myocarditis</td>
<td align="right">3142</td>
<td align="right">10532</td>
<td align="right">29.832890</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">Myocarditis</td>
<td align="right">908</td>
<td align="right">10532</td>
<td align="right">8.621344</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">Myocarditis</td>
<td align="right">910</td>
<td align="right">10532</td>
<td align="right">8.640334</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">Myocarditis</td>
<td align="right">2011</td>
<td align="right">10532</td>
<td align="right">19.094189</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">Myocarditis</td>
<td align="right">445</td>
<td align="right">10532</td>
<td align="right">4.225218</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">Perimyocarditis</td>
<td align="right">178</td>
<td align="right">1192</td>
<td align="right">14.932886</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">Perimyocarditis</td>
<td align="right">253</td>
<td align="right">1192</td>
<td align="right">21.224832</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">Perimyocarditis</td>
<td align="right">347</td>
<td align="right">1192</td>
<td align="right">29.110738</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">Perimyocarditis</td>
<td align="right">126</td>
<td align="right">1192</td>
<td align="right">10.570470</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">Perimyocarditis</td>
<td align="right">33</td>
<td align="right">1192</td>
<td align="right">2.768456</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">Perimyocarditis</td>
<td align="right">232</td>
<td align="right">1192</td>
<td align="right">19.463087</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">Perimyocarditis</td>
<td align="right">23</td>
<td align="right">1192</td>
<td align="right">1.929530</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">transplant</td>
<td align="right">956</td>
<td align="right">11168</td>
<td align="right">8.560172</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">transplant</td>
<td align="right">2394</td>
<td align="right">11168</td>
<td align="right">21.436246</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">transplant</td>
<td align="right">2911</td>
<td align="right">11168</td>
<td align="right">26.065544</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">transplant</td>
<td align="right">609</td>
<td align="right">11168</td>
<td align="right">5.453080</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">transplant</td>
<td align="right">1422</td>
<td align="right">11168</td>
<td align="right">12.732808</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">transplant</td>
<td align="right">2315</td>
<td align="right">11168</td>
<td align="right">20.728868</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">transplant</td>
<td align="right">561</td>
<td align="right">11168</td>
<td align="right">5.023281</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ggbarplot(datFrac, x=&quot;cond&quot;, y=&quot;percentage&quot;,
          fill = &quot;label&quot;,
          palette = colLab) +
  rotate_x_text(angle = 90)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/cnt%20tables%20per%20cond-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="vis-marker-for-celltype-assignment" class="section level2">
<h2>vis marker for celltype assignment</h2>
<div id="heatmap" class="section level3">
<h3>heatmap</h3>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 

selGenesAll &lt;- read_tsv(file = paste0(basedir,
                                      &quot;/data/markerLabels.txt&quot;)) %&gt;% 
  left_join(., genes, by = &quot;geneID&quot;)

Idents(seurat) &lt;- seurat$seurat_clusters
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesAll,
                  colVecIdent = colPal, 
                  ordVec=levels(seurat),
                  gapVecR=NULL, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20sel%20marker-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$label
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesAll,
                  colVecIdent = colLab, 
                  ordVec=levels(seurat),
                  gapVecR=NULL, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20sel%20marker-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="dotplot" class="section level3">
<h3>Dotplot</h3>
<pre class="r"><code>DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenesAll$gene, scale =T,
        cluster.idents = T) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenesAll$gene, labels=selGenesAll$geneID) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/dotplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$seurat_clusters
DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenesAll$gene, scale =T,
        cluster.idents = T) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenesAll$gene, labels=selGenesAll$geneID) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/dotplot-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="save-seurat-object" class="section level2">
<h2>save seurat object</h2>
<pre class="r"><code>saveRDS(seurat, file = paste0(basedir, 
                              &quot;/data/humanHeartsPlusGraz_intPatients_merged&quot;, 
                              &quot;labeled_seurat.rds&quot;))</code></pre>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] sctransform_0.3.3           viridis_0.6.2              
 [3] viridisLite_0.4.0           pheatmap_1.0.12            
 [5] ggpubr_0.4.0                ggsci_2.9                  
 [7] runSeurat3_0.1.0            here_1.0.1                 
 [9] magrittr_2.0.3              sp_1.4-7                   
[11] SeuratObject_4.1.0          Seurat_4.1.1               
[13] forcats_0.5.1               stringr_1.4.0              
[15] dplyr_1.0.9                 purrr_0.3.4                
[17] readr_2.1.2                 tidyr_1.2.0                
[19] tibble_3.1.7                ggplot2_3.3.6              
[21] tidyverse_1.3.1             SingleCellExperiment_1.14.1
[23] SummarizedExperiment_1.22.0 Biobase_2.52.0             
[25] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[27] IRanges_2.26.0              S4Vectors_0.30.2           
[29] BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[31] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             reticulate_1.24        tidyselect_1.1.2      
  [4] htmlwidgets_1.5.4      grid_4.1.0             Rtsne_0.16            
  [7] munsell_0.5.0          codetools_0.2-18       ica_1.0-2             
 [10] future_1.25.0          miniUI_0.1.1.1         withr_2.5.0           
 [13] spatstat.random_2.2-0  colorspace_2.0-3       progressr_0.10.0      
 [16] highr_0.9              knitr_1.39             rstudioapi_0.13       
 [19] ROCR_1.0-11            ggsignif_0.6.3         tensor_1.5            
 [22] listenv_0.8.0          labeling_0.4.2         git2r_0.30.1          
 [25] GenomeInfoDbData_1.2.6 polyclip_1.10-0        bit64_4.0.5           
 [28] farver_2.1.0           rprojroot_2.0.3        parallelly_1.31.1     
 [31] vctrs_0.4.1            generics_0.1.2         xfun_0.30             
 [34] R6_2.5.1               bitops_1.0-7           spatstat.utils_2.3-1  
 [37] DelayedArray_0.18.0    assertthat_0.2.1       vroom_1.5.7           
 [40] promises_1.2.0.1       scales_1.2.0           rgeos_0.5-9           
 [43] gtable_0.3.0           globals_0.15.0         goftest_1.2-3         
 [46] workflowr_1.7.0        rlang_1.0.2            splines_4.1.0         
 [49] rstatix_0.7.0          lazyeval_0.2.2         spatstat.geom_2.4-0   
 [52] broom_0.8.0            yaml_2.3.5             reshape2_1.4.4        
 [55] abind_1.4-5            modelr_0.1.8           backports_1.4.1       
 [58] httpuv_1.6.5           tools_4.1.0            ellipsis_0.3.2        
 [61] spatstat.core_2.4-2    jquerylib_0.1.4        RColorBrewer_1.1-3    
 [64] ggridges_0.5.3         Rcpp_1.0.8.3           plyr_1.8.7            
 [67] zlibbioc_1.38.0        RCurl_1.98-1.6         rpart_4.1.16          
 [70] deldir_1.0-6           pbapply_1.5-0          cowplot_1.1.1         
 [73] zoo_1.8-10             haven_2.5.0            ggrepel_0.9.1         
 [76] cluster_2.1.3          fs_1.5.2               data.table_1.14.2     
 [79] scattermore_0.8        lmtest_0.9-40          reprex_2.0.1          
 [82] RANN_2.6.1             fitdistrplus_1.1-8     hms_1.1.1             
 [85] patchwork_1.1.1        mime_0.12              evaluate_0.15         
 [88] xtable_1.8-4           readxl_1.4.0           gridExtra_2.3         
 [91] compiler_4.1.0         KernSmooth_2.23-20     crayon_1.5.1          
 [94] htmltools_0.5.2        mgcv_1.8-40            later_1.3.0           
 [97] tzdb_0.3.0             lubridate_1.8.0        DBI_1.1.2             
[100] dbplyr_2.1.1           MASS_7.3-57            Matrix_1.4-1          
[103] car_3.0-13             cli_3.3.0              igraph_1.3.1          
[106] pkgconfig_2.0.3        plotly_4.10.0          spatstat.sparse_2.1-1 
[109] xml2_1.3.3             bslib_0.3.1            XVector_0.32.0        
[112] rvest_1.0.2            digest_0.6.29          RcppAnnoy_0.0.19      
[115] spatstat.data_2.2-0    rmarkdown_2.14         cellranger_1.1.0      
[118] leiden_0.3.10          uwot_0.1.11            shiny_1.7.1           
[121] lifecycle_1.0.1        nlme_3.1-157           jsonlite_1.8.0        
[124] carData_3.0-5          fansi_1.0.3            pillar_1.7.0          
[127] lattice_0.20-45        fastmap_1.1.0          httr_1.4.3            
[130] survival_3.3-1         glue_1.6.2             png_0.1-7             
[133] bit_4.0.4              stringi_1.7.6          sass_0.4.1            
[136] irlba_2.3.5            future.apply_1.9.0    </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Wed Jul  6 14:03:00 2022&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] sctransform_0.3.3           viridis_0.6.2              
 [3] viridisLite_0.4.0           pheatmap_1.0.12            
 [5] ggpubr_0.4.0                ggsci_2.9                  
 [7] runSeurat3_0.1.0            here_1.0.1                 
 [9] magrittr_2.0.3              sp_1.4-7                   
[11] SeuratObject_4.1.0          Seurat_4.1.1               
[13] forcats_0.5.1               stringr_1.4.0              
[15] dplyr_1.0.9                 purrr_0.3.4                
[17] readr_2.1.2                 tidyr_1.2.0                
[19] tibble_3.1.7                ggplot2_3.3.6              
[21] tidyverse_1.3.1             SingleCellExperiment_1.14.1
[23] SummarizedExperiment_1.22.0 Biobase_2.52.0             
[25] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[27] IRanges_2.26.0              S4Vectors_0.30.2           
[29] BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[31] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             reticulate_1.24        tidyselect_1.1.2      
  [4] htmlwidgets_1.5.4      grid_4.1.0             Rtsne_0.16            
  [7] munsell_0.5.0          codetools_0.2-18       ica_1.0-2             
 [10] future_1.25.0          miniUI_0.1.1.1         withr_2.5.0           
 [13] spatstat.random_2.2-0  colorspace_2.0-3       progressr_0.10.0      
 [16] highr_0.9              knitr_1.39             rstudioapi_0.13       
 [19] ROCR_1.0-11            ggsignif_0.6.3         tensor_1.5            
 [22] listenv_0.8.0          labeling_0.4.2         git2r_0.30.1          
 [25] GenomeInfoDbData_1.2.6 polyclip_1.10-0        bit64_4.0.5           
 [28] farver_2.1.0           rprojroot_2.0.3        parallelly_1.31.1     
 [31] vctrs_0.4.1            generics_0.1.2         xfun_0.30             
 [34] R6_2.5.1               bitops_1.0-7           spatstat.utils_2.3-1  
 [37] DelayedArray_0.18.0    assertthat_0.2.1       vroom_1.5.7           
 [40] promises_1.2.0.1       scales_1.2.0           rgeos_0.5-9           
 [43] gtable_0.3.0           globals_0.15.0         goftest_1.2-3         
 [46] workflowr_1.7.0        rlang_1.0.2            splines_4.1.0         
 [49] rstatix_0.7.0          lazyeval_0.2.2         spatstat.geom_2.4-0   
 [52] broom_0.8.0            yaml_2.3.5             reshape2_1.4.4        
 [55] abind_1.4-5            modelr_0.1.8           backports_1.4.1       
 [58] httpuv_1.6.5           tools_4.1.0            ellipsis_0.3.2        
 [61] spatstat.core_2.4-2    jquerylib_0.1.4        RColorBrewer_1.1-3    
 [64] ggridges_0.5.3         Rcpp_1.0.8.3           plyr_1.8.7            
 [67] zlibbioc_1.38.0        RCurl_1.98-1.6         rpart_4.1.16          
 [70] deldir_1.0-6           pbapply_1.5-0          cowplot_1.1.1         
 [73] zoo_1.8-10             haven_2.5.0            ggrepel_0.9.1         
 [76] cluster_2.1.3          fs_1.5.2               data.table_1.14.2     
 [79] scattermore_0.8        lmtest_0.9-40          reprex_2.0.1          
 [82] RANN_2.6.1             fitdistrplus_1.1-8     hms_1.1.1             
 [85] patchwork_1.1.1        mime_0.12              evaluate_0.15         
 [88] xtable_1.8-4           readxl_1.4.0           gridExtra_2.3         
 [91] compiler_4.1.0         KernSmooth_2.23-20     crayon_1.5.1          
 [94] htmltools_0.5.2        mgcv_1.8-40            later_1.3.0           
 [97] tzdb_0.3.0             lubridate_1.8.0        DBI_1.1.2             
[100] dbplyr_2.1.1           MASS_7.3-57            Matrix_1.4-1          
[103] car_3.0-13             cli_3.3.0              igraph_1.3.1          
[106] pkgconfig_2.0.3        plotly_4.10.0          spatstat.sparse_2.1-1 
[109] xml2_1.3.3             bslib_0.3.1            XVector_0.32.0        
[112] rvest_1.0.2            digest_0.6.29          RcppAnnoy_0.0.19      
[115] spatstat.data_2.2-0    rmarkdown_2.14         cellranger_1.1.0      
[118] leiden_0.3.10          uwot_0.1.11            shiny_1.7.1           
[121] lifecycle_1.0.1        nlme_3.1-157           jsonlite_1.8.0        
[124] carData_3.0-5          fansi_1.0.3            pillar_1.7.0          
[127] lattice_0.20-45        fastmap_1.1.0          httr_1.4.3            
[130] survival_3.3-1         glue_1.6.2             png_0.1-7             
[133] bit_4.0.4              stringi_1.7.6          sass_0.4.1            
[136] irlba_2.3.5            future.apply_1.9.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
