<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2020-05-14" />

<title>Merge samples</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">humanCardiacFibroblasts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Merge samples</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">14 May 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>humanCardiacFibroblasts/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210903code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210903)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210903code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210903)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststreecd902dd255038cafc3f45ae7b1a4883b11b6c78ctargetblankcd902dda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/cd902dd255038cafc3f45ae7b1a4883b11b6c78c" target="_blank">cd902dd</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststreecd902dd255038cafc3f45ae7b1a4883b11b6c78ctargetblankcd902dda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/cd902dd255038cafc3f45ae7b1a4883b11b6c78c" target="_blank">cd902dd</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSEA/
    Ignored:    data/humanFibroblast/
    Ignored:    figure/DEgenesGZplusSG_Groups.Rmd/.DS_Store

Unstaged changes:
    Modified:   analysis/assignLabelshumanHeartsPlusGrazInt.Rmd
    Modified:   analysis/integrateAcrossPatientsGZplusSG.Rmd
    Modified:   analysis/mergeHumanSamplesPlusGraz.Rmd
    Modified:   metadata2.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/assignLabelshumanHeartsPlusGrazInt.Rmd</code>) and HTML
(<code>docs/assignLabelshumanHeartsPlusGrazInt.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/analysis/assignLabelshumanHeartsPlusGrazInt.Rmd" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
<td>
add human cardiac samples GZ and SG
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/humanCardiacFibroblasts/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/assignLabelshumanHeartsPlusGrazInt.html" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
<td>
add human cardiac samples GZ and SG
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/analysis/assignLabelshumanHeartsPlusGrazInt.Rmd" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
<td>
assign labels
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/humanCardiacFibroblasts/141fae8969bec76e9e2d03e1ae2846382105928d/docs/assignLabelshumanHeartsPlusGrazInt.html" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
<td>
assign labels
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(SingleCellExperiment)
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(ggsci)
  library(ggpubr)
  library(pheatmap)
  library(viridis)
  library(sctransform)
})</code></pre>
</div>
<div id="load-data" class="section level2">
<h2>load data</h2>
<pre class="r"><code>basedir &lt;- here()
seurat &lt;- readRDS(file = paste0(basedir, 
                              &quot;/data/humanHeartsPlusGraz_intPatients_merged&quot;, 
                              &quot;_seurat.rds&quot;))
Idents(seurat) &lt;- seurat$seurat_clusters</code></pre>
</div>
<div id="assign-labels" class="section level2">
<h2>assign labels</h2>
<pre class="r"><code>seurat$label &lt;- &quot;other&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;2&quot;,&quot;6&quot;,&quot;7&quot;))] &lt;- &quot;Endothelial&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;5&quot;))] &lt;- &quot;Tcell&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;4&quot;))] &lt;- &quot;Cardiomyocyte&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;0&quot;))] &lt;- &quot;Fibroblast&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;1&quot;,&quot;8&quot;))] &lt;- &quot;Perivascular&quot;
seurat$label[which(seurat$seurat_clusters %in% c(&quot;3&quot;))] &lt;- &quot;MonocyteMacrophage&quot;</code></pre>
</div>
<div id="color-vectors" class="section level2">
<h2>color vectors</h2>
<pre class="r"><code>colPal &lt;- c(pal_igv()(12),
            pal_aaas()(10))[1:length(levels(seurat))]
colTec &lt;- pal_jama()(length(unique(seurat$technique)))
colSmp &lt;- c(pal_uchicago()(9), pal_npg()(10), pal_aaas()(10), 
            pal_jama()(7))[1:length(unique(seurat$dataset))]
colCond &lt;- pal_npg()(length(unique(seurat$cond)))
colID &lt;- c(pal_jco()(10), pal_npg()(10), pal_futurama()(10),
           pal_d3()(10))[1:length(unique(seurat$ID))]
colOrig &lt;- pal_aaas()(length(unique(seurat$origin)))
colIso &lt;- pal_nejm()(length(unique(seurat$isolation)))
colProc &lt;- pal_aaas()(length(unique(seurat$processing)))
colLab &lt;- pal_futurama()(length(unique(seurat$label)))

names(colPal) &lt;- levels(seurat)
names(colTec) &lt;- unique(seurat$technique)
names(colSmp) &lt;- unique(seurat$dataset)
names(colCond) &lt;- unique(seurat$cond)
names(colID) &lt;- unique(seurat$ID)
names(colOrig) &lt;- unique(seurat$origin)
names(colIso) &lt;- unique(seurat$isolation)
names(colProc) &lt;- unique(seurat$processing)
names(colLab) &lt;- unique(seurat$label)</code></pre>
</div>
<div id="vis-data" class="section level2 tabset">
<h2 class="tabset">vis data</h2>
<div id="clusters" class="section level3">
<h3>clusters</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cluster-1">
Past versions of "vis cluster-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-cluster-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis cluster-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis cluster-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="label" class="section level3">
<h3>label</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;,  group.by = &quot;label&quot;, cols=colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20label-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-label-1">
Past versions of "vis label-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-label-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis label-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis label-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="technique" class="section level3">
<h3>technique</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;technique&quot;, cols=colTec)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20tecnique-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-tecnique-1">
Past versions of "vis tecnique-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-tecnique-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis tecnique-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis tecnique-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample" class="section level3">
<h3>Sample</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;dataset&quot;, cols=colSmp)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20sample-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-sample-1">
Past versions of "vis sample-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis sample-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis sample-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="id" class="section level3">
<h3>ID</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;ID&quot;, cols=colID)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20ID-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-ID-1">
Past versions of "vis ID-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-ID-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis ID-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis ID-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="origin" class="section level3">
<h3>Origin</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;origin&quot;, cols=colOrig)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20origin-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-origin-1">
Past versions of "vis origin-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-origin-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis origin-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis origin-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="isolation" class="section level3">
<h3>isolation</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;isolation&quot;, cols=colIso)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20isolation-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-isolation-1">
Past versions of "vis isolation-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-isolation-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis isolation-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis isolation-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cond" class="section level3">
<h3>cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;cond&quot;, cols=colCond)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20cond-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cond-1">
Past versions of "vis cond-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-cond-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis cond-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis cond-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="processing" class="section level3">
<h3>processing</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;processing&quot;, cols=colProc)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20proc-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-proc-1">
Past versions of "vis proc-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-proc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis proc-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis proc-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cnt-tables" class="section level2 tabset">
<h2 class="tabset">cnt tables</h2>
<div id="per-patient" class="section level3">
<h3>per patient</h3>
<pre class="r"><code>## total cells per patient
knitr::kable(table(seurat$ID))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GZ1</td>
<td align="right">2740</td>
</tr>
<tr class="even">
<td align="left">GZ10</td>
<td align="right">3731</td>
</tr>
<tr class="odd">
<td align="left">GZ11</td>
<td align="right">3991</td>
</tr>
<tr class="even">
<td align="left">GZ12</td>
<td align="right">3818</td>
</tr>
<tr class="odd">
<td align="left">GZ13</td>
<td align="right">9882</td>
</tr>
<tr class="even">
<td align="left">GZ14</td>
<td align="right">1268</td>
</tr>
<tr class="odd">
<td align="left">GZ15</td>
<td align="right">4439</td>
</tr>
<tr class="even">
<td align="left">GZ16</td>
<td align="right">436</td>
</tr>
<tr class="odd">
<td align="left">GZ17</td>
<td align="right">1370</td>
</tr>
<tr class="even">
<td align="left">GZ18</td>
<td align="right">2280</td>
</tr>
<tr class="odd">
<td align="left">GZ19</td>
<td align="right">111</td>
</tr>
<tr class="even">
<td align="left">GZ2</td>
<td align="right">1684</td>
</tr>
<tr class="odd">
<td align="left">GZ20</td>
<td align="right">2706</td>
</tr>
<tr class="even">
<td align="left">GZ21</td>
<td align="right">1442</td>
</tr>
<tr class="odd">
<td align="left">GZ22</td>
<td align="right">1998</td>
</tr>
<tr class="even">
<td align="left">GZ23</td>
<td align="right">841</td>
</tr>
<tr class="odd">
<td align="left">GZ24</td>
<td align="right">1480</td>
</tr>
<tr class="even">
<td align="left">GZ3</td>
<td align="right">2396</td>
</tr>
<tr class="odd">
<td align="left">GZ4</td>
<td align="right">545</td>
</tr>
<tr class="even">
<td align="left">GZ5</td>
<td align="right">781</td>
</tr>
<tr class="odd">
<td align="left">GZ6</td>
<td align="right">491</td>
</tr>
<tr class="even">
<td align="left">GZ7</td>
<td align="right">653</td>
</tr>
<tr class="odd">
<td align="left">GZ8</td>
<td align="right">3921</td>
</tr>
<tr class="even">
<td align="left">GZ9</td>
<td align="right">4908</td>
</tr>
<tr class="odd">
<td align="left">SG29</td>
<td align="right">1242</td>
</tr>
<tr class="even">
<td align="left">SG30</td>
<td align="right">236</td>
</tr>
<tr class="odd">
<td align="left">SG31</td>
<td align="right">1192</td>
</tr>
<tr class="even">
<td align="left">SG32</td>
<td align="right">1428</td>
</tr>
<tr class="odd">
<td align="left">SG33</td>
<td align="right">6286</td>
</tr>
<tr class="even">
<td align="left">SG34</td>
<td align="right">620</td>
</tr>
<tr class="odd">
<td align="left">SG35</td>
<td align="right">2363</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype per patient counts
knitr::kable(table(seurat$label, seurat$ID))</code></pre>
<table>
<colgroup>
<col width="11%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">GZ1</th>
<th align="right">GZ10</th>
<th align="right">GZ11</th>
<th align="right">GZ12</th>
<th align="right">GZ13</th>
<th align="right">GZ14</th>
<th align="right">GZ15</th>
<th align="right">GZ16</th>
<th align="right">GZ17</th>
<th align="right">GZ18</th>
<th align="right">GZ19</th>
<th align="right">GZ2</th>
<th align="right">GZ20</th>
<th align="right">GZ21</th>
<th align="right">GZ22</th>
<th align="right">GZ23</th>
<th align="right">GZ24</th>
<th align="right">GZ3</th>
<th align="right">GZ4</th>
<th align="right">GZ5</th>
<th align="right">GZ6</th>
<th align="right">GZ7</th>
<th align="right">GZ8</th>
<th align="right">GZ9</th>
<th align="right">SG29</th>
<th align="right">SG30</th>
<th align="right">SG31</th>
<th align="right">SG32</th>
<th align="right">SG33</th>
<th align="right">SG34</th>
<th align="right">SG35</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="right">208</td>
<td align="right">1098</td>
<td align="right">299</td>
<td align="right">116</td>
<td align="right">2206</td>
<td align="right">50</td>
<td align="right">116</td>
<td align="right">25</td>
<td align="right">71</td>
<td align="right">328</td>
<td align="right">1</td>
<td align="right">108</td>
<td align="right">46</td>
<td align="right">58</td>
<td align="right">159</td>
<td align="right">54</td>
<td align="right">117</td>
<td align="right">234</td>
<td align="right">45</td>
<td align="right">39</td>
<td align="right">12</td>
<td align="right">88</td>
<td align="right">714</td>
<td align="right">650</td>
<td align="right">109</td>
<td align="right">47</td>
<td align="right">178</td>
<td align="right">40</td>
<td align="right">212</td>
<td align="right">173</td>
<td align="right">304</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="right">442</td>
<td align="right">411</td>
<td align="right">810</td>
<td align="right">1152</td>
<td align="right">2259</td>
<td align="right">323</td>
<td align="right">1067</td>
<td align="right">72</td>
<td align="right">193</td>
<td align="right">643</td>
<td align="right">30</td>
<td align="right">425</td>
<td align="right">1242</td>
<td align="right">258</td>
<td align="right">273</td>
<td align="right">226</td>
<td align="right">285</td>
<td align="right">592</td>
<td align="right">180</td>
<td align="right">244</td>
<td align="right">155</td>
<td align="right">131</td>
<td align="right">783</td>
<td align="right">1025</td>
<td align="right">165</td>
<td align="right">74</td>
<td align="right">257</td>
<td align="right">581</td>
<td align="right">257</td>
<td align="right">104</td>
<td align="right">581</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="right">718</td>
<td align="right">984</td>
<td align="right">841</td>
<td align="right">1110</td>
<td align="right">2222</td>
<td align="right">316</td>
<td align="right">1095</td>
<td align="right">90</td>
<td align="right">368</td>
<td align="right">624</td>
<td align="right">16</td>
<td align="right">640</td>
<td align="right">624</td>
<td align="right">99</td>
<td align="right">314</td>
<td align="right">68</td>
<td align="right">268</td>
<td align="right">815</td>
<td align="right">136</td>
<td align="right">261</td>
<td align="right">150</td>
<td align="right">220</td>
<td align="right">1146</td>
<td align="right">1468</td>
<td align="right">141</td>
<td align="right">42</td>
<td align="right">347</td>
<td align="right">307</td>
<td align="right">402</td>
<td align="right">115</td>
<td align="right">759</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="right">299</td>
<td align="right">487</td>
<td align="right">659</td>
<td align="right">221</td>
<td align="right">898</td>
<td align="right">91</td>
<td align="right">330</td>
<td align="right">56</td>
<td align="right">331</td>
<td align="right">93</td>
<td align="right">17</td>
<td align="right">85</td>
<td align="right">137</td>
<td align="right">184</td>
<td align="right">444</td>
<td align="right">123</td>
<td align="right">106</td>
<td align="right">118</td>
<td align="right">54</td>
<td align="right">31</td>
<td align="right">63</td>
<td align="right">18</td>
<td align="right">359</td>
<td align="right">638</td>
<td align="right">321</td>
<td align="right">14</td>
<td align="right">124</td>
<td align="right">65</td>
<td align="right">2170</td>
<td align="right">51</td>
<td align="right">204</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="right">59</td>
<td align="right">93</td>
<td align="right">27</td>
<td align="right">22</td>
<td align="right">205</td>
<td align="right">10</td>
<td align="right">77</td>
<td align="right">11</td>
<td align="right">28</td>
<td align="right">29</td>
<td align="right">1</td>
<td align="right">30</td>
<td align="right">20</td>
<td align="right">15</td>
<td align="right">30</td>
<td align="right">7</td>
<td align="right">11</td>
<td align="right">37</td>
<td align="right">8</td>
<td align="right">27</td>
<td align="right">3</td>
<td align="right">53</td>
<td align="right">69</td>
<td align="right">34</td>
<td align="right">15</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">33</td>
<td align="right">32</td>
<td align="right">6</td>
<td align="right">29</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="right">746</td>
<td align="right">523</td>
<td align="right">1201</td>
<td align="right">1096</td>
<td align="right">1813</td>
<td align="right">439</td>
<td align="right">1536</td>
<td align="right">80</td>
<td align="right">190</td>
<td align="right">517</td>
<td align="right">33</td>
<td align="right">322</td>
<td align="right">497</td>
<td align="right">803</td>
<td align="right">728</td>
<td align="right">337</td>
<td align="right">657</td>
<td align="right">523</td>
<td align="right">90</td>
<td align="right">156</td>
<td align="right">72</td>
<td align="right">123</td>
<td align="right">742</td>
<td align="right">999</td>
<td align="right">103</td>
<td align="right">55</td>
<td align="right">234</td>
<td align="right">320</td>
<td align="right">170</td>
<td align="right">145</td>
<td align="right">397</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="right">268</td>
<td align="right">135</td>
<td align="right">154</td>
<td align="right">101</td>
<td align="right">279</td>
<td align="right">39</td>
<td align="right">218</td>
<td align="right">102</td>
<td align="right">189</td>
<td align="right">46</td>
<td align="right">13</td>
<td align="right">74</td>
<td align="right">140</td>
<td align="right">25</td>
<td align="right">50</td>
<td align="right">26</td>
<td align="right">36</td>
<td align="right">77</td>
<td align="right">32</td>
<td align="right">23</td>
<td align="right">36</td>
<td align="right">20</td>
<td align="right">108</td>
<td align="right">94</td>
<td align="right">388</td>
<td align="right">4</td>
<td align="right">29</td>
<td align="right">82</td>
<td align="right">3043</td>
<td align="right">26</td>
<td align="right">89</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype percentages per patient
datLab &lt;- data.frame(table(seurat$label, seurat$ID))
colnames(datLab) &lt;- c(&quot;label&quot;, &quot;ID&quot;, &quot;cnt&quot;)
datPat &lt;- data.frame(table(seurat$ID))
colnames(datPat) &lt;- c(&quot;ID&quot;, &quot;total&quot;)
datFrac &lt;- datLab %&gt;% left_join(., datPat, by=&quot;ID&quot;) %&gt;% 
  mutate(percentage = cnt*100/total)
knitr::kable(datFrac)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">label</th>
<th align="left">ID</th>
<th align="right">cnt</th>
<th align="right">total</th>
<th align="right">percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ1</td>
<td align="right">208</td>
<td align="right">2740</td>
<td align="right">7.5912409</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ1</td>
<td align="right">442</td>
<td align="right">2740</td>
<td align="right">16.1313869</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ1</td>
<td align="right">718</td>
<td align="right">2740</td>
<td align="right">26.2043796</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ1</td>
<td align="right">299</td>
<td align="right">2740</td>
<td align="right">10.9124088</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ1</td>
<td align="right">59</td>
<td align="right">2740</td>
<td align="right">2.1532847</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ1</td>
<td align="right">746</td>
<td align="right">2740</td>
<td align="right">27.2262774</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ1</td>
<td align="right">268</td>
<td align="right">2740</td>
<td align="right">9.7810219</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ10</td>
<td align="right">1098</td>
<td align="right">3731</td>
<td align="right">29.4291075</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ10</td>
<td align="right">411</td>
<td align="right">3731</td>
<td align="right">11.0158135</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ10</td>
<td align="right">984</td>
<td align="right">3731</td>
<td align="right">26.3736264</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ10</td>
<td align="right">487</td>
<td align="right">3731</td>
<td align="right">13.0528009</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ10</td>
<td align="right">93</td>
<td align="right">3731</td>
<td align="right">2.4926293</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ10</td>
<td align="right">523</td>
<td align="right">3731</td>
<td align="right">14.0176896</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ10</td>
<td align="right">135</td>
<td align="right">3731</td>
<td align="right">3.6183329</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ11</td>
<td align="right">299</td>
<td align="right">3991</td>
<td align="right">7.4918567</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ11</td>
<td align="right">810</td>
<td align="right">3991</td>
<td align="right">20.2956652</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ11</td>
<td align="right">841</td>
<td align="right">3991</td>
<td align="right">21.0724129</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ11</td>
<td align="right">659</td>
<td align="right">3991</td>
<td align="right">16.5121523</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ11</td>
<td align="right">27</td>
<td align="right">3991</td>
<td align="right">0.6765222</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ11</td>
<td align="right">1201</td>
<td align="right">3991</td>
<td align="right">30.0927086</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ11</td>
<td align="right">154</td>
<td align="right">3991</td>
<td align="right">3.8586820</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ12</td>
<td align="right">116</td>
<td align="right">3818</td>
<td align="right">3.0382399</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ12</td>
<td align="right">1152</td>
<td align="right">3818</td>
<td align="right">30.1728654</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ12</td>
<td align="right">1110</td>
<td align="right">3818</td>
<td align="right">29.0728130</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ12</td>
<td align="right">221</td>
<td align="right">3818</td>
<td align="right">5.7883709</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ12</td>
<td align="right">22</td>
<td align="right">3818</td>
<td align="right">0.5762179</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ12</td>
<td align="right">1096</td>
<td align="right">3818</td>
<td align="right">28.7061289</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ12</td>
<td align="right">101</td>
<td align="right">3818</td>
<td align="right">2.6453641</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ13</td>
<td align="right">2206</td>
<td align="right">9882</td>
<td align="right">22.3234163</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ13</td>
<td align="right">2259</td>
<td align="right">9882</td>
<td align="right">22.8597450</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ13</td>
<td align="right">2222</td>
<td align="right">9882</td>
<td align="right">22.4853269</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ13</td>
<td align="right">898</td>
<td align="right">9882</td>
<td align="right">9.0872293</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ13</td>
<td align="right">205</td>
<td align="right">9882</td>
<td align="right">2.0744789</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ13</td>
<td align="right">1813</td>
<td align="right">9882</td>
<td align="right">18.3464886</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ13</td>
<td align="right">279</td>
<td align="right">9882</td>
<td align="right">2.8233151</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ14</td>
<td align="right">50</td>
<td align="right">1268</td>
<td align="right">3.9432177</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ14</td>
<td align="right">323</td>
<td align="right">1268</td>
<td align="right">25.4731861</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ14</td>
<td align="right">316</td>
<td align="right">1268</td>
<td align="right">24.9211356</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ14</td>
<td align="right">91</td>
<td align="right">1268</td>
<td align="right">7.1766562</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ14</td>
<td align="right">10</td>
<td align="right">1268</td>
<td align="right">0.7886435</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ14</td>
<td align="right">439</td>
<td align="right">1268</td>
<td align="right">34.6214511</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ14</td>
<td align="right">39</td>
<td align="right">1268</td>
<td align="right">3.0757098</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ15</td>
<td align="right">116</td>
<td align="right">4439</td>
<td align="right">2.6132012</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ15</td>
<td align="right">1067</td>
<td align="right">4439</td>
<td align="right">24.0369453</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ15</td>
<td align="right">1095</td>
<td align="right">4439</td>
<td align="right">24.6677180</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ15</td>
<td align="right">330</td>
<td align="right">4439</td>
<td align="right">7.4341068</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ15</td>
<td align="right">77</td>
<td align="right">4439</td>
<td align="right">1.7346249</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ15</td>
<td align="right">1536</td>
<td align="right">4439</td>
<td align="right">34.6023879</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ15</td>
<td align="right">218</td>
<td align="right">4439</td>
<td align="right">4.9110160</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ16</td>
<td align="right">25</td>
<td align="right">436</td>
<td align="right">5.7339450</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ16</td>
<td align="right">72</td>
<td align="right">436</td>
<td align="right">16.5137615</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ16</td>
<td align="right">90</td>
<td align="right">436</td>
<td align="right">20.6422018</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ16</td>
<td align="right">56</td>
<td align="right">436</td>
<td align="right">12.8440367</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ16</td>
<td align="right">11</td>
<td align="right">436</td>
<td align="right">2.5229358</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ16</td>
<td align="right">80</td>
<td align="right">436</td>
<td align="right">18.3486239</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ16</td>
<td align="right">102</td>
<td align="right">436</td>
<td align="right">23.3944954</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ17</td>
<td align="right">71</td>
<td align="right">1370</td>
<td align="right">5.1824818</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ17</td>
<td align="right">193</td>
<td align="right">1370</td>
<td align="right">14.0875912</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ17</td>
<td align="right">368</td>
<td align="right">1370</td>
<td align="right">26.8613139</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ17</td>
<td align="right">331</td>
<td align="right">1370</td>
<td align="right">24.1605839</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ17</td>
<td align="right">28</td>
<td align="right">1370</td>
<td align="right">2.0437956</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ17</td>
<td align="right">190</td>
<td align="right">1370</td>
<td align="right">13.8686131</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ17</td>
<td align="right">189</td>
<td align="right">1370</td>
<td align="right">13.7956204</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ18</td>
<td align="right">328</td>
<td align="right">2280</td>
<td align="right">14.3859649</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ18</td>
<td align="right">643</td>
<td align="right">2280</td>
<td align="right">28.2017544</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ18</td>
<td align="right">624</td>
<td align="right">2280</td>
<td align="right">27.3684211</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ18</td>
<td align="right">93</td>
<td align="right">2280</td>
<td align="right">4.0789474</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ18</td>
<td align="right">29</td>
<td align="right">2280</td>
<td align="right">1.2719298</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ18</td>
<td align="right">517</td>
<td align="right">2280</td>
<td align="right">22.6754386</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ18</td>
<td align="right">46</td>
<td align="right">2280</td>
<td align="right">2.0175439</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ19</td>
<td align="right">1</td>
<td align="right">111</td>
<td align="right">0.9009009</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ19</td>
<td align="right">30</td>
<td align="right">111</td>
<td align="right">27.0270270</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ19</td>
<td align="right">16</td>
<td align="right">111</td>
<td align="right">14.4144144</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ19</td>
<td align="right">17</td>
<td align="right">111</td>
<td align="right">15.3153153</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ19</td>
<td align="right">1</td>
<td align="right">111</td>
<td align="right">0.9009009</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ19</td>
<td align="right">33</td>
<td align="right">111</td>
<td align="right">29.7297297</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ19</td>
<td align="right">13</td>
<td align="right">111</td>
<td align="right">11.7117117</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ2</td>
<td align="right">108</td>
<td align="right">1684</td>
<td align="right">6.4133017</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ2</td>
<td align="right">425</td>
<td align="right">1684</td>
<td align="right">25.2375297</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ2</td>
<td align="right">640</td>
<td align="right">1684</td>
<td align="right">38.0047506</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ2</td>
<td align="right">85</td>
<td align="right">1684</td>
<td align="right">5.0475059</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ2</td>
<td align="right">30</td>
<td align="right">1684</td>
<td align="right">1.7814727</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ2</td>
<td align="right">322</td>
<td align="right">1684</td>
<td align="right">19.1211401</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ2</td>
<td align="right">74</td>
<td align="right">1684</td>
<td align="right">4.3942993</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ20</td>
<td align="right">46</td>
<td align="right">2706</td>
<td align="right">1.6999261</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ20</td>
<td align="right">1242</td>
<td align="right">2706</td>
<td align="right">45.8980044</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ20</td>
<td align="right">624</td>
<td align="right">2706</td>
<td align="right">23.0598670</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ20</td>
<td align="right">137</td>
<td align="right">2706</td>
<td align="right">5.0628234</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ20</td>
<td align="right">20</td>
<td align="right">2706</td>
<td align="right">0.7390983</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ20</td>
<td align="right">497</td>
<td align="right">2706</td>
<td align="right">18.3665928</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ20</td>
<td align="right">140</td>
<td align="right">2706</td>
<td align="right">5.1736881</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ21</td>
<td align="right">58</td>
<td align="right">1442</td>
<td align="right">4.0221914</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ21</td>
<td align="right">258</td>
<td align="right">1442</td>
<td align="right">17.8918169</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ21</td>
<td align="right">99</td>
<td align="right">1442</td>
<td align="right">6.8654646</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ21</td>
<td align="right">184</td>
<td align="right">1442</td>
<td align="right">12.7600555</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ21</td>
<td align="right">15</td>
<td align="right">1442</td>
<td align="right">1.0402219</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ21</td>
<td align="right">803</td>
<td align="right">1442</td>
<td align="right">55.6865465</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ21</td>
<td align="right">25</td>
<td align="right">1442</td>
<td align="right">1.7337032</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ22</td>
<td align="right">159</td>
<td align="right">1998</td>
<td align="right">7.9579580</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ22</td>
<td align="right">273</td>
<td align="right">1998</td>
<td align="right">13.6636637</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ22</td>
<td align="right">314</td>
<td align="right">1998</td>
<td align="right">15.7157157</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ22</td>
<td align="right">444</td>
<td align="right">1998</td>
<td align="right">22.2222222</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ22</td>
<td align="right">30</td>
<td align="right">1998</td>
<td align="right">1.5015015</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ22</td>
<td align="right">728</td>
<td align="right">1998</td>
<td align="right">36.4364364</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ22</td>
<td align="right">50</td>
<td align="right">1998</td>
<td align="right">2.5025025</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ23</td>
<td align="right">54</td>
<td align="right">841</td>
<td align="right">6.4209275</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ23</td>
<td align="right">226</td>
<td align="right">841</td>
<td align="right">26.8727705</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ23</td>
<td align="right">68</td>
<td align="right">841</td>
<td align="right">8.0856124</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ23</td>
<td align="right">123</td>
<td align="right">841</td>
<td align="right">14.6254459</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ23</td>
<td align="right">7</td>
<td align="right">841</td>
<td align="right">0.8323424</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ23</td>
<td align="right">337</td>
<td align="right">841</td>
<td align="right">40.0713436</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ23</td>
<td align="right">26</td>
<td align="right">841</td>
<td align="right">3.0915577</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ24</td>
<td align="right">117</td>
<td align="right">1480</td>
<td align="right">7.9054054</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ24</td>
<td align="right">285</td>
<td align="right">1480</td>
<td align="right">19.2567568</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ24</td>
<td align="right">268</td>
<td align="right">1480</td>
<td align="right">18.1081081</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ24</td>
<td align="right">106</td>
<td align="right">1480</td>
<td align="right">7.1621622</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ24</td>
<td align="right">11</td>
<td align="right">1480</td>
<td align="right">0.7432432</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ24</td>
<td align="right">657</td>
<td align="right">1480</td>
<td align="right">44.3918919</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ24</td>
<td align="right">36</td>
<td align="right">1480</td>
<td align="right">2.4324324</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ3</td>
<td align="right">234</td>
<td align="right">2396</td>
<td align="right">9.7662771</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ3</td>
<td align="right">592</td>
<td align="right">2396</td>
<td align="right">24.7078464</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ3</td>
<td align="right">815</td>
<td align="right">2396</td>
<td align="right">34.0150250</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ3</td>
<td align="right">118</td>
<td align="right">2396</td>
<td align="right">4.9248748</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ3</td>
<td align="right">37</td>
<td align="right">2396</td>
<td align="right">1.5442404</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ3</td>
<td align="right">523</td>
<td align="right">2396</td>
<td align="right">21.8280467</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ3</td>
<td align="right">77</td>
<td align="right">2396</td>
<td align="right">3.2136895</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ4</td>
<td align="right">45</td>
<td align="right">545</td>
<td align="right">8.2568807</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ4</td>
<td align="right">180</td>
<td align="right">545</td>
<td align="right">33.0275229</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ4</td>
<td align="right">136</td>
<td align="right">545</td>
<td align="right">24.9541284</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ4</td>
<td align="right">54</td>
<td align="right">545</td>
<td align="right">9.9082569</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ4</td>
<td align="right">8</td>
<td align="right">545</td>
<td align="right">1.4678899</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ4</td>
<td align="right">90</td>
<td align="right">545</td>
<td align="right">16.5137615</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ4</td>
<td align="right">32</td>
<td align="right">545</td>
<td align="right">5.8715596</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ5</td>
<td align="right">39</td>
<td align="right">781</td>
<td align="right">4.9935980</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ5</td>
<td align="right">244</td>
<td align="right">781</td>
<td align="right">31.2419974</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ5</td>
<td align="right">261</td>
<td align="right">781</td>
<td align="right">33.4186940</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ5</td>
<td align="right">31</td>
<td align="right">781</td>
<td align="right">3.9692702</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ5</td>
<td align="right">27</td>
<td align="right">781</td>
<td align="right">3.4571063</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ5</td>
<td align="right">156</td>
<td align="right">781</td>
<td align="right">19.9743918</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ5</td>
<td align="right">23</td>
<td align="right">781</td>
<td align="right">2.9449424</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ6</td>
<td align="right">12</td>
<td align="right">491</td>
<td align="right">2.4439919</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ6</td>
<td align="right">155</td>
<td align="right">491</td>
<td align="right">31.5682281</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ6</td>
<td align="right">150</td>
<td align="right">491</td>
<td align="right">30.5498982</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ6</td>
<td align="right">63</td>
<td align="right">491</td>
<td align="right">12.8309572</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ6</td>
<td align="right">3</td>
<td align="right">491</td>
<td align="right">0.6109980</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ6</td>
<td align="right">72</td>
<td align="right">491</td>
<td align="right">14.6639511</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ6</td>
<td align="right">36</td>
<td align="right">491</td>
<td align="right">7.3319756</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ7</td>
<td align="right">88</td>
<td align="right">653</td>
<td align="right">13.4762634</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ7</td>
<td align="right">131</td>
<td align="right">653</td>
<td align="right">20.0612557</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ7</td>
<td align="right">220</td>
<td align="right">653</td>
<td align="right">33.6906585</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ7</td>
<td align="right">18</td>
<td align="right">653</td>
<td align="right">2.7565084</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ7</td>
<td align="right">53</td>
<td align="right">653</td>
<td align="right">8.1163859</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ7</td>
<td align="right">123</td>
<td align="right">653</td>
<td align="right">18.8361409</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ7</td>
<td align="right">20</td>
<td align="right">653</td>
<td align="right">3.0627871</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ8</td>
<td align="right">714</td>
<td align="right">3921</td>
<td align="right">18.2096404</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ8</td>
<td align="right">783</td>
<td align="right">3921</td>
<td align="right">19.9693956</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ8</td>
<td align="right">1146</td>
<td align="right">3921</td>
<td align="right">29.2272379</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ8</td>
<td align="right">359</td>
<td align="right">3921</td>
<td align="right">9.1558276</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">GZ8</td>
<td align="right">69</td>
<td align="right">3921</td>
<td align="right">1.7597552</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">GZ8</td>
<td align="right">742</td>
<td align="right">3921</td>
<td align="right">18.9237439</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">GZ8</td>
<td align="right">108</td>
<td align="right">3921</td>
<td align="right">2.7543994</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ9</td>
<td align="right">650</td>
<td align="right">4908</td>
<td align="right">13.2436838</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">GZ9</td>
<td align="right">1025</td>
<td align="right">4908</td>
<td align="right">20.8842706</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">GZ9</td>
<td align="right">1468</td>
<td align="right">4908</td>
<td align="right">29.9103504</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">GZ9</td>
<td align="right">638</td>
<td align="right">4908</td>
<td align="right">12.9991850</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">GZ9</td>
<td align="right">34</td>
<td align="right">4908</td>
<td align="right">0.6927465</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ9</td>
<td align="right">999</td>
<td align="right">4908</td>
<td align="right">20.3545232</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ9</td>
<td align="right">94</td>
<td align="right">4908</td>
<td align="right">1.9152404</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG29</td>
<td align="right">109</td>
<td align="right">1242</td>
<td align="right">8.7761675</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG29</td>
<td align="right">165</td>
<td align="right">1242</td>
<td align="right">13.2850242</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG29</td>
<td align="right">141</td>
<td align="right">1242</td>
<td align="right">11.3526570</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG29</td>
<td align="right">321</td>
<td align="right">1242</td>
<td align="right">25.8454106</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG29</td>
<td align="right">15</td>
<td align="right">1242</td>
<td align="right">1.2077295</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG29</td>
<td align="right">103</td>
<td align="right">1242</td>
<td align="right">8.2930757</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG29</td>
<td align="right">388</td>
<td align="right">1242</td>
<td align="right">31.2399356</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG30</td>
<td align="right">47</td>
<td align="right">236</td>
<td align="right">19.9152542</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">SG30</td>
<td align="right">74</td>
<td align="right">236</td>
<td align="right">31.3559322</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">SG30</td>
<td align="right">42</td>
<td align="right">236</td>
<td align="right">17.7966102</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG30</td>
<td align="right">14</td>
<td align="right">236</td>
<td align="right">5.9322034</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">SG30</td>
<td align="right">0</td>
<td align="right">236</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG30</td>
<td align="right">55</td>
<td align="right">236</td>
<td align="right">23.3050847</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG30</td>
<td align="right">4</td>
<td align="right">236</td>
<td align="right">1.6949153</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG31</td>
<td align="right">178</td>
<td align="right">1192</td>
<td align="right">14.9328859</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG31</td>
<td align="right">257</td>
<td align="right">1192</td>
<td align="right">21.5604027</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG31</td>
<td align="right">347</td>
<td align="right">1192</td>
<td align="right">29.1107383</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG31</td>
<td align="right">124</td>
<td align="right">1192</td>
<td align="right">10.4026846</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG31</td>
<td align="right">23</td>
<td align="right">1192</td>
<td align="right">1.9295302</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG31</td>
<td align="right">234</td>
<td align="right">1192</td>
<td align="right">19.6308725</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG31</td>
<td align="right">29</td>
<td align="right">1192</td>
<td align="right">2.4328859</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG32</td>
<td align="right">40</td>
<td align="right">1428</td>
<td align="right">2.8011204</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">SG32</td>
<td align="right">581</td>
<td align="right">1428</td>
<td align="right">40.6862745</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">SG32</td>
<td align="right">307</td>
<td align="right">1428</td>
<td align="right">21.4985994</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG32</td>
<td align="right">65</td>
<td align="right">1428</td>
<td align="right">4.5518207</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">SG32</td>
<td align="right">33</td>
<td align="right">1428</td>
<td align="right">2.3109244</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG32</td>
<td align="right">320</td>
<td align="right">1428</td>
<td align="right">22.4089636</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG32</td>
<td align="right">82</td>
<td align="right">1428</td>
<td align="right">5.7422969</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG33</td>
<td align="right">212</td>
<td align="right">6286</td>
<td align="right">3.3725740</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG33</td>
<td align="right">257</td>
<td align="right">6286</td>
<td align="right">4.0884505</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG33</td>
<td align="right">402</td>
<td align="right">6286</td>
<td align="right">6.3951639</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG33</td>
<td align="right">2170</td>
<td align="right">6286</td>
<td align="right">34.5211581</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG33</td>
<td align="right">32</td>
<td align="right">6286</td>
<td align="right">0.5090678</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG33</td>
<td align="right">170</td>
<td align="right">6286</td>
<td align="right">2.7044225</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG33</td>
<td align="right">3043</td>
<td align="right">6286</td>
<td align="right">48.4091632</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG34</td>
<td align="right">173</td>
<td align="right">620</td>
<td align="right">27.9032258</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">SG34</td>
<td align="right">104</td>
<td align="right">620</td>
<td align="right">16.7741935</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">SG34</td>
<td align="right">115</td>
<td align="right">620</td>
<td align="right">18.5483871</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG34</td>
<td align="right">51</td>
<td align="right">620</td>
<td align="right">8.2258065</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">SG34</td>
<td align="right">6</td>
<td align="right">620</td>
<td align="right">0.9677419</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG34</td>
<td align="right">145</td>
<td align="right">620</td>
<td align="right">23.3870968</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG34</td>
<td align="right">26</td>
<td align="right">620</td>
<td align="right">4.1935484</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">SG35</td>
<td align="right">304</td>
<td align="right">2363</td>
<td align="right">12.8650021</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG35</td>
<td align="right">581</td>
<td align="right">2363</td>
<td align="right">24.5873889</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG35</td>
<td align="right">759</td>
<td align="right">2363</td>
<td align="right">32.1201862</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">SG35</td>
<td align="right">204</td>
<td align="right">2363</td>
<td align="right">8.6330935</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">SG35</td>
<td align="right">29</td>
<td align="right">2363</td>
<td align="right">1.2272535</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">SG35</td>
<td align="right">397</td>
<td align="right">2363</td>
<td align="right">16.8006771</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">SG35</td>
<td align="right">89</td>
<td align="right">2363</td>
<td align="right">3.7663986</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ggbarplot(datFrac, x=&quot;ID&quot;, y=&quot;percentage&quot;,
          fill = &quot;label&quot;,
          palette = colLab) +
  rotate_x_text(angle = 90)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/cnt%20tables%20per%20pat-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-cnt-tables-per-pat-1">
Past versions of "cnt tables per pat-1.png"
</button>
</p>
<div id="fig-no-spaces-cnt-tables-per-pat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/cnt tables per pat-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/cnt tables per pat-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="per-cond" class="section level3">
<h3>per cond</h3>
<pre class="r"><code>## total cells per cond
knitr::kable(table(seurat$cond))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HH</td>
<td align="right">36012</td>
</tr>
<tr class="even">
<td align="left">InfCardiomyopathy</td>
<td align="right">1428</td>
</tr>
<tr class="odd">
<td align="left">inflamation</td>
<td align="right">236</td>
</tr>
<tr class="even">
<td align="left">Myocarditis</td>
<td align="right">32411</td>
</tr>
<tr class="odd">
<td align="left">Perimyocarditis</td>
<td align="right">1192</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype per cond counts
knitr::kable(table(seurat$label, seurat$cond))</code></pre>
<table>
<colgroup>
<col width="23%" />
<col width="6%" />
<col width="21%" />
<col width="14%" />
<col width="14%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">HH</th>
<th align="right">InfCardiomyopathy</th>
<th align="right">inflamation</th>
<th align="right">Myocarditis</th>
<th align="right">Perimyocarditis</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="right">5471</td>
<td align="right">40</td>
<td align="right">47</td>
<td align="right">2169</td>
<td align="right">178</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="right">7482</td>
<td align="right">581</td>
<td align="right">74</td>
<td align="right">6846</td>
<td align="right">257</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="right">8520</td>
<td align="right">307</td>
<td align="right">42</td>
<td align="right">7490</td>
<td align="right">347</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="right">4119</td>
<td align="right">65</td>
<td align="right">14</td>
<td align="right">4469</td>
<td align="right">124</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="right">513</td>
<td align="right">33</td>
<td align="right">0</td>
<td align="right">475</td>
<td align="right">23</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="right">8899</td>
<td align="right">320</td>
<td align="right">55</td>
<td align="right">6139</td>
<td align="right">234</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="right">1008</td>
<td align="right">82</td>
<td align="right">4</td>
<td align="right">4823</td>
<td align="right">29</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype percentages per cond
datLab &lt;- data.frame(table(seurat$label, seurat$cond))
colnames(datLab) &lt;- c(&quot;label&quot;, &quot;cond&quot;, &quot;cnt&quot;)
datPat &lt;- data.frame(table(seurat$cond))
colnames(datPat) &lt;- c(&quot;cond&quot;, &quot;total&quot;)
datFrac &lt;- datLab %&gt;% left_join(., datPat, by=&quot;cond&quot;) %&gt;% 
  mutate(percentage = cnt*100/total)
knitr::kable(datFrac)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">label</th>
<th align="left">cond</th>
<th align="right">cnt</th>
<th align="right">total</th>
<th align="right">percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">HH</td>
<td align="right">5471</td>
<td align="right">36012</td>
<td align="right">15.192158</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">HH</td>
<td align="right">7482</td>
<td align="right">36012</td>
<td align="right">20.776408</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">HH</td>
<td align="right">8520</td>
<td align="right">36012</td>
<td align="right">23.658780</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">HH</td>
<td align="right">4119</td>
<td align="right">36012</td>
<td align="right">11.437854</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">HH</td>
<td align="right">513</td>
<td align="right">36012</td>
<td align="right">1.424525</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">HH</td>
<td align="right">8899</td>
<td align="right">36012</td>
<td align="right">24.711207</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">HH</td>
<td align="right">1008</td>
<td align="right">36012</td>
<td align="right">2.799067</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">40</td>
<td align="right">1428</td>
<td align="right">2.801120</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">581</td>
<td align="right">1428</td>
<td align="right">40.686275</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">307</td>
<td align="right">1428</td>
<td align="right">21.498599</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">65</td>
<td align="right">1428</td>
<td align="right">4.551821</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">33</td>
<td align="right">1428</td>
<td align="right">2.310924</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">320</td>
<td align="right">1428</td>
<td align="right">22.408964</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">InfCardiomyopathy</td>
<td align="right">82</td>
<td align="right">1428</td>
<td align="right">5.742297</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">inflamation</td>
<td align="right">47</td>
<td align="right">236</td>
<td align="right">19.915254</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">inflamation</td>
<td align="right">74</td>
<td align="right">236</td>
<td align="right">31.355932</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">inflamation</td>
<td align="right">42</td>
<td align="right">236</td>
<td align="right">17.796610</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">inflamation</td>
<td align="right">14</td>
<td align="right">236</td>
<td align="right">5.932203</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">inflamation</td>
<td align="right">0</td>
<td align="right">236</td>
<td align="right">0.000000</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">inflamation</td>
<td align="right">55</td>
<td align="right">236</td>
<td align="right">23.305085</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">inflamation</td>
<td align="right">4</td>
<td align="right">236</td>
<td align="right">1.694915</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">Myocarditis</td>
<td align="right">2169</td>
<td align="right">32411</td>
<td align="right">6.692172</td>
</tr>
<tr class="odd">
<td align="left">Endothelial</td>
<td align="left">Myocarditis</td>
<td align="right">6846</td>
<td align="right">32411</td>
<td align="right">21.122458</td>
</tr>
<tr class="even">
<td align="left">Fibroblast</td>
<td align="left">Myocarditis</td>
<td align="right">7490</td>
<td align="right">32411</td>
<td align="right">23.109438</td>
</tr>
<tr class="odd">
<td align="left">MonocyteMacrophage</td>
<td align="left">Myocarditis</td>
<td align="right">4469</td>
<td align="right">32411</td>
<td align="right">13.788529</td>
</tr>
<tr class="even">
<td align="left">other</td>
<td align="left">Myocarditis</td>
<td align="right">475</td>
<td align="right">32411</td>
<td align="right">1.465552</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">Myocarditis</td>
<td align="right">6139</td>
<td align="right">32411</td>
<td align="right">18.941100</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">Myocarditis</td>
<td align="right">4823</td>
<td align="right">32411</td>
<td align="right">14.880750</td>
</tr>
<tr class="odd">
<td align="left">Cardiomyocyte</td>
<td align="left">Perimyocarditis</td>
<td align="right">178</td>
<td align="right">1192</td>
<td align="right">14.932886</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">Perimyocarditis</td>
<td align="right">257</td>
<td align="right">1192</td>
<td align="right">21.560403</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">Perimyocarditis</td>
<td align="right">347</td>
<td align="right">1192</td>
<td align="right">29.110738</td>
</tr>
<tr class="even">
<td align="left">MonocyteMacrophage</td>
<td align="left">Perimyocarditis</td>
<td align="right">124</td>
<td align="right">1192</td>
<td align="right">10.402685</td>
</tr>
<tr class="odd">
<td align="left">other</td>
<td align="left">Perimyocarditis</td>
<td align="right">23</td>
<td align="right">1192</td>
<td align="right">1.929530</td>
</tr>
<tr class="even">
<td align="left">Perivascular</td>
<td align="left">Perimyocarditis</td>
<td align="right">234</td>
<td align="right">1192</td>
<td align="right">19.630872</td>
</tr>
<tr class="odd">
<td align="left">Tcell</td>
<td align="left">Perimyocarditis</td>
<td align="right">29</td>
<td align="right">1192</td>
<td align="right">2.432886</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ggbarplot(datFrac, x=&quot;cond&quot;, y=&quot;percentage&quot;,
          fill = &quot;label&quot;,
          palette = colLab) +
  rotate_x_text(angle = 90)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/cnt%20tables%20per%20cond-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-cnt-tables-per-cond-1">
Past versions of "cnt tables per cond-1.png"
</button>
</p>
<div id="fig-no-spaces-cnt-tables-per-cond-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/cnt tables per cond-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/cnt tables per cond-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="vis-marker-for-celltype-assignment" class="section level2">
<h2>vis marker for celltype assignment</h2>
<div id="heatmap" class="section level3">
<h3>heatmap</h3>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 

selGenesAll &lt;- read_tsv(file = paste0(basedir,
                                      &quot;/data/markerLabels.txt&quot;)) %&gt;% 
  left_join(., genes, by = &quot;geneID&quot;)

Idents(seurat) &lt;- seurat$seurat_clusters
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesAll,
                  colVecIdent = colPal, 
                  ordVec=levels(seurat),
                  gapVecR=NULL, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20sel%20marker-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-sel-marker-1">
Past versions of "vis sel marker-1.png"
</button>
</p>
<div id="fig-no-spaces-vis-sel-marker-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis sel marker-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis sel marker-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seurat) &lt;- seurat$label
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesAll,
                  colVecIdent = colLab, 
                  ordVec=levels(seurat),
                  gapVecR=NULL, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis%20sel%20marker-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-sel-marker-2">
Past versions of "vis sel marker-2.png"
</button>
</p>
<div id="fig-no-spaces-vis-sel-marker-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis sel marker-2.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/vis sel marker-2.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dotplot" class="section level3">
<h3>Dotplot</h3>
<pre class="r"><code>DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenesAll$gene, scale =T,
        cluster.idents = T) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenesAll$gene, labels=selGenesAll$geneID) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/dotplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dotplot-1">
Past versions of dotplot-1.png
</button>
</p>
<div id="fig-dotplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/dotplot-1.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/dotplot-1.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seurat) &lt;- seurat$seurat_clusters
DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenesAll$gene, scale =T,
        cluster.idents = T) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenesAll$gene, labels=selGenesAll$geneID) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazInt.Rmd/dotplot-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dotplot-2">
Past versions of dotplot-2.png
</button>
</p>
<div id="fig-dotplot-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/cd902dd255038cafc3f45ae7b1a4883b11b6c78c/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/dotplot-2.png" target="_blank">cd902dd</a>
</td>
<td>
mluetge
</td>
<td>
2022-09-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/141fae8969bec76e9e2d03e1ae2846382105928d/docs/figure/assignLabelshumanHeartsPlusGrazInt.Rmd/dotplot-2.png" target="_blank">141fae8</a>
</td>
<td>
mluetge
</td>
<td>
2022-07-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="save-seurat-object" class="section level2">
<h2>save seurat object</h2>
<pre class="r"><code>saveRDS(seurat, file = paste0(basedir, 
                              &quot;/data/humanHeartsPlusGraz_intPatients_merged&quot;, 
                              &quot;labeled_seurat.rds&quot;))</code></pre>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] sctransform_0.3.4           viridis_0.6.2              
 [3] viridisLite_0.4.1           pheatmap_1.0.12            
 [5] ggpubr_0.4.0                ggsci_2.9                  
 [7] runSeurat3_0.1.0            here_1.0.1                 
 [9] magrittr_2.0.3              sp_1.5-0                   
[11] SeuratObject_4.1.1          Seurat_4.1.1               
[13] forcats_0.5.2               stringr_1.4.1              
[15] dplyr_1.0.10                purrr_0.3.4                
[17] readr_2.1.2                 tidyr_1.2.1                
[19] tibble_3.1.8                ggplot2_3.3.6              
[21] tidyverse_1.3.2             SingleCellExperiment_1.18.0
[23] SummarizedExperiment_1.26.1 Biobase_2.56.0             
[25] GenomicRanges_1.48.0        GenomeInfoDb_1.32.4        
[27] IRanges_2.30.1              S4Vectors_0.34.0           
[29] BiocGenerics_0.42.0         MatrixGenerics_1.8.1       
[31] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             reticulate_1.26        tidyselect_1.1.2      
  [4] htmlwidgets_1.5.4      grid_4.2.1             Rtsne_0.16            
  [7] munsell_0.5.0          codetools_0.2-18       ica_1.0-3             
 [10] future_1.28.0          miniUI_0.1.1.1         withr_2.5.0           
 [13] spatstat.random_2.2-0  colorspace_2.0-3       progressr_0.11.0      
 [16] highr_0.9              knitr_1.40             rstudioapi_0.14       
 [19] ROCR_1.0-11            ggsignif_0.6.3         tensor_1.5            
 [22] listenv_0.8.0          labeling_0.4.2         git2r_0.30.1          
 [25] GenomeInfoDbData_1.2.8 polyclip_1.10-0        bit64_4.0.5           
 [28] farver_2.1.1           rprojroot_2.0.3        parallelly_1.32.1     
 [31] vctrs_0.4.1            generics_0.1.3         xfun_0.32             
 [34] R6_2.5.1               bitops_1.0-7           spatstat.utils_2.3-1  
 [37] cachem_1.0.6           DelayedArray_0.22.0    assertthat_0.2.1      
 [40] vroom_1.5.7            promises_1.2.0.1       scales_1.2.1          
 [43] googlesheets4_1.0.1    rgeos_0.5-9            gtable_0.3.1          
 [46] globals_0.16.1         goftest_1.2-3          workflowr_1.7.0       
 [49] rlang_1.0.5            splines_4.2.1          rstatix_0.7.0         
 [52] lazyeval_0.2.2         gargle_1.2.0           spatstat.geom_2.4-0   
 [55] broom_1.0.1            yaml_2.3.5             reshape2_1.4.4        
 [58] abind_1.4-5            modelr_0.1.9           backports_1.4.1       
 [61] httpuv_1.6.6           tools_4.2.1            ellipsis_0.3.2        
 [64] spatstat.core_2.4-4    jquerylib_0.1.4        RColorBrewer_1.1-3    
 [67] ggridges_0.5.3         Rcpp_1.0.9             plyr_1.8.7            
 [70] zlibbioc_1.42.0        RCurl_1.98-1.8         rpart_4.1.16          
 [73] deldir_1.0-6           pbapply_1.5-0          cowplot_1.1.1         
 [76] zoo_1.8-10             haven_2.5.1            ggrepel_0.9.1         
 [79] cluster_2.1.4          fs_1.5.2               data.table_1.14.2     
 [82] scattermore_0.8        lmtest_0.9-40          reprex_2.0.2          
 [85] RANN_2.6.1             googledrive_2.0.0      whisker_0.4           
 [88] fitdistrplus_1.1-8     hms_1.1.2              patchwork_1.1.2       
 [91] mime_0.12              evaluate_0.16          xtable_1.8-4          
 [94] readxl_1.4.1           gridExtra_2.3          compiler_4.2.1        
 [97] KernSmooth_2.23-20     crayon_1.5.1           htmltools_0.5.3       
[100] mgcv_1.8-40            later_1.3.0            tzdb_0.3.0            
[103] lubridate_1.8.0        DBI_1.1.3              dbplyr_2.2.1          
[106] MASS_7.3-58.1          Matrix_1.4-1           car_3.1-0             
[109] cli_3.4.0              parallel_4.2.1         igraph_1.3.4          
[112] pkgconfig_2.0.3        plotly_4.10.0          spatstat.sparse_2.1-1 
[115] xml2_1.3.3             bslib_0.4.0            XVector_0.36.0        
[118] rvest_1.0.3            digest_0.6.29          RcppAnnoy_0.0.19      
[121] spatstat.data_2.2-0    rmarkdown_2.16         cellranger_1.1.0      
[124] leiden_0.4.2           uwot_0.1.14            shiny_1.7.2           
[127] lifecycle_1.0.2        nlme_3.1-159           jsonlite_1.8.0        
[130] carData_3.0-5          fansi_1.0.3            pillar_1.8.1          
[133] lattice_0.20-45        fastmap_1.1.0          httr_1.4.4            
[136] survival_3.4-0         glue_1.6.2             png_0.1-7             
[139] bit_4.0.4              stringi_1.7.8          sass_0.4.2            
[142] irlba_2.3.5            future.apply_1.9.0    </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Mon Sep 26 17:13:41 2022&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] sctransform_0.3.4           viridis_0.6.2              
 [3] viridisLite_0.4.1           pheatmap_1.0.12            
 [5] ggpubr_0.4.0                ggsci_2.9                  
 [7] runSeurat3_0.1.0            here_1.0.1                 
 [9] magrittr_2.0.3              sp_1.5-0                   
[11] SeuratObject_4.1.1          Seurat_4.1.1               
[13] forcats_0.5.2               stringr_1.4.1              
[15] dplyr_1.0.10                purrr_0.3.4                
[17] readr_2.1.2                 tidyr_1.2.1                
[19] tibble_3.1.8                ggplot2_3.3.6              
[21] tidyverse_1.3.2             SingleCellExperiment_1.18.0
[23] SummarizedExperiment_1.26.1 Biobase_2.56.0             
[25] GenomicRanges_1.48.0        GenomeInfoDb_1.32.4        
[27] IRanges_2.30.1              S4Vectors_0.34.0           
[29] BiocGenerics_0.42.0         MatrixGenerics_1.8.1       
[31] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             reticulate_1.26        tidyselect_1.1.2      
  [4] htmlwidgets_1.5.4      grid_4.2.1             Rtsne_0.16            
  [7] munsell_0.5.0          codetools_0.2-18       ica_1.0-3             
 [10] future_1.28.0          miniUI_0.1.1.1         withr_2.5.0           
 [13] spatstat.random_2.2-0  colorspace_2.0-3       progressr_0.11.0      
 [16] highr_0.9              knitr_1.40             rstudioapi_0.14       
 [19] ROCR_1.0-11            ggsignif_0.6.3         tensor_1.5            
 [22] listenv_0.8.0          labeling_0.4.2         git2r_0.30.1          
 [25] GenomeInfoDbData_1.2.8 polyclip_1.10-0        bit64_4.0.5           
 [28] farver_2.1.1           rprojroot_2.0.3        parallelly_1.32.1     
 [31] vctrs_0.4.1            generics_0.1.3         xfun_0.32             
 [34] R6_2.5.1               bitops_1.0-7           spatstat.utils_2.3-1  
 [37] cachem_1.0.6           DelayedArray_0.22.0    assertthat_0.2.1      
 [40] vroom_1.5.7            promises_1.2.0.1       scales_1.2.1          
 [43] googlesheets4_1.0.1    rgeos_0.5-9            gtable_0.3.1          
 [46] globals_0.16.1         goftest_1.2-3          workflowr_1.7.0       
 [49] rlang_1.0.5            splines_4.2.1          rstatix_0.7.0         
 [52] lazyeval_0.2.2         gargle_1.2.0           spatstat.geom_2.4-0   
 [55] broom_1.0.1            yaml_2.3.5             reshape2_1.4.4        
 [58] abind_1.4-5            modelr_0.1.9           backports_1.4.1       
 [61] httpuv_1.6.6           tools_4.2.1            ellipsis_0.3.2        
 [64] spatstat.core_2.4-4    jquerylib_0.1.4        RColorBrewer_1.1-3    
 [67] ggridges_0.5.3         Rcpp_1.0.9             plyr_1.8.7            
 [70] zlibbioc_1.42.0        RCurl_1.98-1.8         rpart_4.1.16          
 [73] deldir_1.0-6           pbapply_1.5-0          cowplot_1.1.1         
 [76] zoo_1.8-10             haven_2.5.1            ggrepel_0.9.1         
 [79] cluster_2.1.4          fs_1.5.2               data.table_1.14.2     
 [82] scattermore_0.8        lmtest_0.9-40          reprex_2.0.2          
 [85] RANN_2.6.1             googledrive_2.0.0      whisker_0.4           
 [88] fitdistrplus_1.1-8     hms_1.1.2              patchwork_1.1.2       
 [91] mime_0.12              evaluate_0.16          xtable_1.8-4          
 [94] readxl_1.4.1           gridExtra_2.3          compiler_4.2.1        
 [97] KernSmooth_2.23-20     crayon_1.5.1           htmltools_0.5.3       
[100] mgcv_1.8-40            later_1.3.0            tzdb_0.3.0            
[103] lubridate_1.8.0        DBI_1.1.3              dbplyr_2.2.1          
[106] MASS_7.3-58.1          Matrix_1.4-1           car_3.1-0             
[109] cli_3.4.0              parallel_4.2.1         igraph_1.3.4          
[112] pkgconfig_2.0.3        plotly_4.10.0          spatstat.sparse_2.1-1 
[115] xml2_1.3.3             bslib_0.4.0            XVector_0.36.0        
[118] rvest_1.0.3            digest_0.6.29          RcppAnnoy_0.0.19      
[121] spatstat.data_2.2-0    rmarkdown_2.16         cellranger_1.1.0      
[124] leiden_0.4.2           uwot_0.1.14            shiny_1.7.2           
[127] lifecycle_1.0.2        nlme_3.1-159           jsonlite_1.8.0        
[130] carData_3.0-5          fansi_1.0.3            pillar_1.8.1          
[133] lattice_0.20-45        fastmap_1.1.0          httr_1.4.4            
[136] survival_3.4-0         glue_1.6.2             png_0.1-7             
[139] bit_4.0.4              stringi_1.7.8          sass_0.4.2            
[142] irlba_2.3.5            future.apply_1.9.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
