<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2022-07-31" />

<title>assign labels</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">humanCardiacFibroblasts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">assign labels</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">31 July 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>humanCardiacFibroblasts/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210903code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210903)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210903code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210903)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststree2a11889f57e3ea1855c1d33f0311ac59e38f6f29targetblank2a11889a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/2a11889f57e3ea1855c1d33f0311ac59e38f6f29" target="_blank">2a11889</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststree2a11889f57e3ea1855c1d33f0311ac59e38f6f29targetblank2a11889a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/2a11889f57e3ea1855c1d33f0311ac59e38f6f29" target="_blank">2a11889</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSEA/
    Ignored:    data/humanFibroblast/
    Ignored:    figure/DEgenesGZplusSG_Groups.Rmd/.DS_Store

Untracked files:
    Untracked:  analysis/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd

Unstaged changes:
    Modified:   analysis/assignLabelshumanHeartsPlusGrazIntWoGZ24.Rmd
    Modified:   analysis/integrateAcrossPatientsGZplusSG.Rmd
    Modified:   analysis/mergeHumanSamplesPlusGraz.Rmd
    Modified:   metadata2.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(SingleCellExperiment)
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(ggsci)
  library(ggpubr)
  library(pheatmap)
  library(viridis)
  library(sctransform)
})</code></pre>
</div>
<div id="load-data" class="section level2">
<h2>load data</h2>
<pre class="r"><code>basedir &lt;- here()
seurat &lt;- readRDS(file = paste0(basedir, 
                              &quot;/data/humanHeartsPlusGraz_intPatients_merged&quot;, 
                              &quot;_seurat.rds&quot;))
Idents(seurat) &lt;- seurat$seurat_clusters
table(seurat$ID)</code></pre>
<pre><code>
 GZ1 GZ14 GZ15 GZ16 GZ17 GZ18  GZ2 GZ20  GZ3  GZ4  GZ5  GZ6  GZ7 SG17 SG21 SG24 
2740 1268 4439  436 1370 2280 1684 2706 2396  545  781  491  653 1321 1536 2381 
SG25 SG28 SG29 SG31 SG32 SG33 SG34 SG35 
3131 1545 2496 1192 1428 6286  620 2363 </code></pre>
</div>
<div id="assign-labels" class="section level2">
<h2>assign labels</h2>
<pre class="r"><code>seurat$label &lt;- &quot;other&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;2&quot;,&quot;9&quot;))] &lt;- &quot;Endothelial&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;8&quot;))] &lt;- &quot;EndoEC&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;6&quot;))] &lt;- &quot;Tcell&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;5&quot;))] &lt;- &quot;Cardiomyocyte&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;0&quot;))] &lt;- &quot;Fibroblast&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;1&quot;))] &lt;- &quot;Perivascular&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;7&quot;))] &lt;- &quot;SMC&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;3&quot;))] &lt;- &quot;resMacrophage&quot;
seurat$label[which(seurat$integrated_snn_res.0.6 %in% c(&quot;8&quot;))] &lt;- &quot;infMacrophage&quot;
seurat$label[which(seurat$integrated_snn_res.0.4 %in% c(&quot;10&quot;))] &lt;- &quot;NeuralCells&quot;

## reembedding of cluster 4 (mix of diff cell typpes)
seuratSub &lt;- subset(seurat, integrated_snn_res.0.4==&quot;4&quot;)

seuratSub &lt;- rerunSeurat3(seuratSub)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3471
Number of edges: 127742

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9495
Number of communities: 12
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3471
Number of edges: 127742

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9033
Number of communities: 15
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3471
Number of edges: 127742

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8787
Number of communities: 17
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3471
Number of edges: 127742

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9290
Number of communities: 14
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>Tcell &lt;- colnames(seuratSub)[which(seuratSub$RNA_snn_res.0.4 %in% c(&quot;0&quot;, &quot;3&quot;,&quot;6&quot;, &quot;5&quot;, &quot;11&quot;))]
endo1 &lt;- colnames(seuratSub)[which(seuratSub$RNA_snn_res.0.4 %in% c(&quot;1&quot;))]
fibro &lt;- colnames(seuratSub)[which(seuratSub$RNA_snn_res.0.4 %in% c(&quot;2&quot;))]
peri &lt;- colnames(seuratSub)[which(seuratSub$RNA_snn_res.0.4 %in% c(&quot;9&quot;, &quot;12&quot;))]
Adipo &lt;- colnames(seuratSub)[which(seuratSub$RNA_snn_res.0.4 %in% c(&quot;7&quot;))]
resmacro &lt;- colnames(seuratSub)[which(seuratSub$RNA_snn_res.0.4 %in% c(&quot;4&quot;, &quot;13&quot;))]
infmacro &lt;- colnames(seuratSub)[which(seuratSub$RNA_snn_res.0.4 %in% c(&quot;8&quot;))]
LEC &lt;- colnames(seuratSub)[which(seuratSub$RNA_snn_res.0.4 %in% c(&quot;10&quot;))]


seurat$label[which(colnames(seurat) %in% Tcell)] &lt;- &quot;Tcell&quot;
seurat$label[which(colnames(seurat) %in% peri)] &lt;- &quot;Perivascular&quot;
seurat$label[which(colnames(seurat) %in% endo1)] &lt;- &quot;Endothelial&quot;
seurat$label[which(colnames(seurat) %in% fibro)] &lt;- &quot;Fibroblast&quot;
seurat$label[which(colnames(seurat) %in% Adipo)] &lt;- &quot;Adipocytes&quot;
seurat$label[which(colnames(seurat) %in% resmacro)] &lt;- &quot;resMacrophage&quot;
seurat$label[which(colnames(seurat) %in% infmacro)] &lt;- &quot;infMacrophage&quot;
seurat$label[which(colnames(seurat) %in% LEC)] &lt;- &quot;LEC&quot;

unique(seurat$integrated_snn_res.0.4)</code></pre>
<pre><code> [1] 8  2  4  3  6  0  1  5  7  9  10
Levels: 0 1 2 3 4 5 6 7 8 9 10</code></pre>
<pre class="r"><code>seurat$clust &lt;- as.character(seurat$integrated_snn_res.0.4)
seurat$clust[which(seurat$integrated_snn_res.0.6 == &quot;8&quot;)] &lt;- &quot;11&quot;
seurat$clust[which(colnames(seurat) %in% Tcell)] &lt;- &quot;12&quot;
seurat$clust[which(colnames(seurat) %in% peri)] &lt;- &quot;13&quot;
seurat$clust[which(colnames(seurat) %in% endo1)] &lt;- &quot;14&quot;
seurat$clust[which(colnames(seurat) %in% fibro)] &lt;- &quot;15&quot;
seurat$clust[which(colnames(seurat) %in% Adipo)] &lt;- &quot;16&quot;
seurat$clust[which(colnames(seurat) %in% resmacro)] &lt;- &quot;17&quot;
seurat$clust[which(colnames(seurat) %in% infmacro)] &lt;- &quot;18&quot;
seurat$clust[which(colnames(seurat) %in% LEC)] &lt;- &quot;19&quot;</code></pre>
</div>
<div id="color-vectors" class="section level2">
<h2>color vectors</h2>
<pre class="r"><code>colPal &lt;- c(pal_igv()(12),
            pal_aaas()(10))[1:length(levels(seurat))]
colTec &lt;- pal_jama()(length(unique(seurat$technique)))
colSmp &lt;- c(pal_uchicago()(9), pal_npg()(10), pal_aaas()(10), 
            pal_jama()(7))[1:length(unique(seurat$dataset))]
colCond &lt;- pal_npg()(length(unique(seurat$cond)))
colID &lt;- c(pal_jco()(10), pal_npg()(10), pal_futurama()(10),
           pal_d3()(10))[1:length(unique(seurat$ID))]
colOrig &lt;- pal_aaas()(length(unique(seurat$origin)))
colIso &lt;- pal_nejm()(length(unique(seurat$isolation)))
colProc &lt;- pal_aaas()(length(unique(seurat$processing)))
colLab &lt;- c(pal_futurama()(8), pal_uchicago()(6))[1:length(unique(seurat$label))]
colClust &lt;- c(pal_igv()(12),
            pal_aaas()(10))[1:length(unique(seurat$clust))]

names(colPal) &lt;- levels(seurat)
names(colTec) &lt;- unique(seurat$technique)
names(colSmp) &lt;- unique(seurat$dataset)
names(colCond) &lt;- unique(seurat$cond)
names(colID) &lt;- unique(seurat$ID)
names(colOrig) &lt;- unique(seurat$origin)
names(colIso) &lt;- unique(seurat$isolation)
names(colProc) &lt;- unique(seurat$processing)
names(colLab) &lt;- unique(seurat$label)
names(colClust) &lt;- unique(seurat$clust)</code></pre>
</div>
<div id="vis-data" class="section level2 tabset">
<h2 class="tabset">vis data</h2>
<div id="clusters" class="section level3">
<h3>clusters</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal,
        shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20cluster-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cluster-plus-re-embedding" class="section level3">
<h3>cluster plus re-embedding</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colClust, group.by = &quot;clust&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20cluster%20plus-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colClust, group.by = &quot;clust&quot;,
        shuffle = F)+
  theme_void()</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20cluster%20plus-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="label" class="section level3">
<h3>label</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;,  group.by = &quot;label&quot;, cols=colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20label-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20label-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="technique" class="section level3">
<h3>technique</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;technique&quot;, cols=colTec)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20tecnique-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sample" class="section level3">
<h3>Sample</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;dataset&quot;, cols=colSmp)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20sample-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="id" class="section level3">
<h3>ID</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;ID&quot;, cols=colID, shuffle = T)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20ID-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;ID&quot;, cols=colID,
        shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20ID-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="origin" class="section level3">
<h3>Origin</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;origin&quot;, cols=colOrig,
        shuffle = T)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20origin-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;origin&quot;, cols=colOrig,
        shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20origin-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="isolation" class="section level3">
<h3>isolation</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;isolation&quot;, cols=colIso)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20isolation-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cond" class="section level3">
<h3>cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;cond&quot;, cols=colCond)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20cond-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="processing" class="section level3">
<h3>processing</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;processing&quot;, cols=colProc)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20proc-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="cnt-tables" class="section level2 tabset">
<h2 class="tabset">cnt tables</h2>
<div id="per-patient" class="section level3">
<h3>per patient</h3>
<pre class="r"><code>## total cells per patient
knitr::kable(table(seurat$ID))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GZ1</td>
<td align="right">2740</td>
</tr>
<tr class="even">
<td align="left">GZ14</td>
<td align="right">1268</td>
</tr>
<tr class="odd">
<td align="left">GZ15</td>
<td align="right">4439</td>
</tr>
<tr class="even">
<td align="left">GZ16</td>
<td align="right">436</td>
</tr>
<tr class="odd">
<td align="left">GZ17</td>
<td align="right">1370</td>
</tr>
<tr class="even">
<td align="left">GZ18</td>
<td align="right">2280</td>
</tr>
<tr class="odd">
<td align="left">GZ2</td>
<td align="right">1684</td>
</tr>
<tr class="even">
<td align="left">GZ20</td>
<td align="right">2706</td>
</tr>
<tr class="odd">
<td align="left">GZ3</td>
<td align="right">2396</td>
</tr>
<tr class="even">
<td align="left">GZ4</td>
<td align="right">545</td>
</tr>
<tr class="odd">
<td align="left">GZ5</td>
<td align="right">781</td>
</tr>
<tr class="even">
<td align="left">GZ6</td>
<td align="right">491</td>
</tr>
<tr class="odd">
<td align="left">GZ7</td>
<td align="right">653</td>
</tr>
<tr class="even">
<td align="left">SG17</td>
<td align="right">1321</td>
</tr>
<tr class="odd">
<td align="left">SG21</td>
<td align="right">1536</td>
</tr>
<tr class="even">
<td align="left">SG24</td>
<td align="right">2381</td>
</tr>
<tr class="odd">
<td align="left">SG25</td>
<td align="right">3131</td>
</tr>
<tr class="even">
<td align="left">SG28</td>
<td align="right">1545</td>
</tr>
<tr class="odd">
<td align="left">SG29</td>
<td align="right">2496</td>
</tr>
<tr class="even">
<td align="left">SG31</td>
<td align="right">1192</td>
</tr>
<tr class="odd">
<td align="left">SG32</td>
<td align="right">1428</td>
</tr>
<tr class="even">
<td align="left">SG33</td>
<td align="right">6286</td>
</tr>
<tr class="odd">
<td align="left">SG34</td>
<td align="right">620</td>
</tr>
<tr class="even">
<td align="left">SG35</td>
<td align="right">2363</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype per patient counts
knitr::kable(table(seurat$label, seurat$ID))</code></pre>
<table>
<colgroup>
<col width="11%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
<col width="3%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">GZ1</th>
<th align="right">GZ14</th>
<th align="right">GZ15</th>
<th align="right">GZ16</th>
<th align="right">GZ17</th>
<th align="right">GZ18</th>
<th align="right">GZ2</th>
<th align="right">GZ20</th>
<th align="right">GZ3</th>
<th align="right">GZ4</th>
<th align="right">GZ5</th>
<th align="right">GZ6</th>
<th align="right">GZ7</th>
<th align="right">SG17</th>
<th align="right">SG21</th>
<th align="right">SG24</th>
<th align="right">SG25</th>
<th align="right">SG28</th>
<th align="right">SG29</th>
<th align="right">SG31</th>
<th align="right">SG32</th>
<th align="right">SG33</th>
<th align="right">SG34</th>
<th align="right">SG35</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">45</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">49</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="right">208</td>
<td align="right">51</td>
<td align="right">115</td>
<td align="right">25</td>
<td align="right">71</td>
<td align="right">329</td>
<td align="right">107</td>
<td align="right">46</td>
<td align="right">238</td>
<td align="right">45</td>
<td align="right">39</td>
<td align="right">12</td>
<td align="right">90</td>
<td align="right">56</td>
<td align="right">96</td>
<td align="right">299</td>
<td align="right">397</td>
<td align="right">71</td>
<td align="right">182</td>
<td align="right">178</td>
<td align="right">40</td>
<td align="right">216</td>
<td align="right">177</td>
<td align="right">303</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="right">173</td>
<td align="right">32</td>
<td align="right">78</td>
<td align="right">21</td>
<td align="right">20</td>
<td align="right">182</td>
<td align="right">127</td>
<td align="right">301</td>
<td align="right">192</td>
<td align="right">64</td>
<td align="right">39</td>
<td align="right">46</td>
<td align="right">12</td>
<td align="right">23</td>
<td align="right">39</td>
<td align="right">92</td>
<td align="right">98</td>
<td align="right">48</td>
<td align="right">99</td>
<td align="right">60</td>
<td align="right">73</td>
<td align="right">82</td>
<td align="right">18</td>
<td align="right">61</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="right">290</td>
<td align="right">290</td>
<td align="right">961</td>
<td align="right">54</td>
<td align="right">167</td>
<td align="right">456</td>
<td align="right">312</td>
<td align="right">933</td>
<td align="right">374</td>
<td align="right">119</td>
<td align="right">207</td>
<td align="right">108</td>
<td align="right">111</td>
<td align="right">826</td>
<td align="right">186</td>
<td align="right">376</td>
<td align="right">548</td>
<td align="right">506</td>
<td align="right">461</td>
<td align="right">197</td>
<td align="right">511</td>
<td align="right">189</td>
<td align="right">82</td>
<td align="right">522</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="right">813</td>
<td align="right">321</td>
<td align="right">1185</td>
<td align="right">126</td>
<td align="right">377</td>
<td align="right">630</td>
<td align="right">667</td>
<td align="right">652</td>
<td align="right">867</td>
<td align="right">143</td>
<td align="right">263</td>
<td align="right">156</td>
<td align="right">240</td>
<td align="right">165</td>
<td align="right">613</td>
<td align="right">882</td>
<td align="right">871</td>
<td align="right">316</td>
<td align="right">422</td>
<td align="right">354</td>
<td align="right">324</td>
<td align="right">414</td>
<td align="right">120</td>
<td align="right">794</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="right">76</td>
<td align="right">22</td>
<td align="right">149</td>
<td align="right">32</td>
<td align="right">135</td>
<td align="right">23</td>
<td align="right">20</td>
<td align="right">38</td>
<td align="right">39</td>
<td align="right">20</td>
<td align="right">10</td>
<td align="right">15</td>
<td align="right">5</td>
<td align="right">114</td>
<td align="right">29</td>
<td align="right">53</td>
<td align="right">102</td>
<td align="right">10</td>
<td align="right">379</td>
<td align="right">35</td>
<td align="right">15</td>
<td align="right">1393</td>
<td align="right">8</td>
<td align="right">55</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">30</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="right">58</td>
<td align="right">10</td>
<td align="right">76</td>
<td align="right">6</td>
<td align="right">28</td>
<td align="right">28</td>
<td align="right">28</td>
<td align="right">20</td>
<td align="right">33</td>
<td align="right">7</td>
<td align="right">26</td>
<td align="right">2</td>
<td align="right">8</td>
<td align="right">19</td>
<td align="right">7</td>
<td align="right">27</td>
<td align="right">27</td>
<td align="right">26</td>
<td align="right">21</td>
<td align="right">22</td>
<td align="right">27</td>
<td align="right">38</td>
<td align="right">5</td>
<td align="right">28</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="right">473</td>
<td align="right">337</td>
<td align="right">1299</td>
<td align="right">18</td>
<td align="right">144</td>
<td align="right">380</td>
<td align="right">264</td>
<td align="right">337</td>
<td align="right">401</td>
<td align="right">55</td>
<td align="right">117</td>
<td align="right">54</td>
<td align="right">63</td>
<td align="right">46</td>
<td align="right">302</td>
<td align="right">285</td>
<td align="right">437</td>
<td align="right">406</td>
<td align="right">297</td>
<td align="right">179</td>
<td align="right">240</td>
<td align="right">178</td>
<td align="right">122</td>
<td align="right">322</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="right">225</td>
<td align="right">72</td>
<td align="right">174</td>
<td align="right">24</td>
<td align="right">200</td>
<td align="right">73</td>
<td align="right">63</td>
<td align="right">104</td>
<td align="right">80</td>
<td align="right">34</td>
<td align="right">21</td>
<td align="right">47</td>
<td align="right">12</td>
<td align="right">13</td>
<td align="right">126</td>
<td align="right">118</td>
<td align="right">150</td>
<td align="right">8</td>
<td align="right">391</td>
<td align="right">91</td>
<td align="right">55</td>
<td align="right">651</td>
<td align="right">42</td>
<td align="right">145</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="right">224</td>
<td align="right">102</td>
<td align="right">256</td>
<td align="right">22</td>
<td align="right">45</td>
<td align="right">144</td>
<td align="right">57</td>
<td align="right">149</td>
<td align="right">108</td>
<td align="right">31</td>
<td align="right">41</td>
<td align="right">19</td>
<td align="right">58</td>
<td align="right">40</td>
<td align="right">93</td>
<td align="right">139</td>
<td align="right">139</td>
<td align="right">134</td>
<td align="right">113</td>
<td align="right">51</td>
<td align="right">82</td>
<td align="right">35</td>
<td align="right">22</td>
<td align="right">73</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="right">195</td>
<td align="right">31</td>
<td align="right">146</td>
<td align="right">108</td>
<td align="right">183</td>
<td align="right">29</td>
<td align="right">33</td>
<td align="right">124</td>
<td align="right">32</td>
<td align="right">26</td>
<td align="right">18</td>
<td align="right">32</td>
<td align="right">6</td>
<td align="right">17</td>
<td align="right">43</td>
<td align="right">98</td>
<td align="right">313</td>
<td align="right">17</td>
<td align="right">121</td>
<td align="right">25</td>
<td align="right">54</td>
<td align="right">3082</td>
<td align="right">24</td>
<td align="right">60</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## celltype percentages per patient
datLab &lt;- data.frame(table(seurat$label, seurat$ID))
colnames(datLab) &lt;- c(&quot;label&quot;, &quot;ID&quot;, &quot;cnt&quot;)
datPat &lt;- data.frame(table(seurat$ID))
colnames(datPat) &lt;- c(&quot;ID&quot;, &quot;total&quot;)
datFrac &lt;- datLab %&gt;% left_join(., datPat, by=&quot;ID&quot;) %&gt;% 
  mutate(percentage = cnt*100/total)
knitr::kable(datFrac)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">label</th>
<th align="left">ID</th>
<th align="right">cnt</th>
<th align="right">total</th>
<th align="right">percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ1</td>
<td align="right">0</td>
<td align="right">2740</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ1</td>
<td align="right">208</td>
<td align="right">2740</td>
<td align="right">7.5912409</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ1</td>
<td align="right">173</td>
<td align="right">2740</td>
<td align="right">6.3138686</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ1</td>
<td align="right">290</td>
<td align="right">2740</td>
<td align="right">10.5839416</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ1</td>
<td align="right">813</td>
<td align="right">2740</td>
<td align="right">29.6715328</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ1</td>
<td align="right">76</td>
<td align="right">2740</td>
<td align="right">2.7737226</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ1</td>
<td align="right">5</td>
<td align="right">2740</td>
<td align="right">0.1824818</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ1</td>
<td align="right">58</td>
<td align="right">2740</td>
<td align="right">2.1167883</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ1</td>
<td align="right">473</td>
<td align="right">2740</td>
<td align="right">17.2627737</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ1</td>
<td align="right">225</td>
<td align="right">2740</td>
<td align="right">8.2116788</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ1</td>
<td align="right">224</td>
<td align="right">2740</td>
<td align="right">8.1751825</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ1</td>
<td align="right">195</td>
<td align="right">2740</td>
<td align="right">7.1167883</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ14</td>
<td align="right">0</td>
<td align="right">1268</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ14</td>
<td align="right">51</td>
<td align="right">1268</td>
<td align="right">4.0220820</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ14</td>
<td align="right">32</td>
<td align="right">1268</td>
<td align="right">2.5236593</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ14</td>
<td align="right">290</td>
<td align="right">1268</td>
<td align="right">22.8706625</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ14</td>
<td align="right">321</td>
<td align="right">1268</td>
<td align="right">25.3154574</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ14</td>
<td align="right">22</td>
<td align="right">1268</td>
<td align="right">1.7350158</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ14</td>
<td align="right">0</td>
<td align="right">1268</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ14</td>
<td align="right">10</td>
<td align="right">1268</td>
<td align="right">0.7886435</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ14</td>
<td align="right">337</td>
<td align="right">1268</td>
<td align="right">26.5772871</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ14</td>
<td align="right">72</td>
<td align="right">1268</td>
<td align="right">5.6782334</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ14</td>
<td align="right">102</td>
<td align="right">1268</td>
<td align="right">8.0441640</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ14</td>
<td align="right">31</td>
<td align="right">1268</td>
<td align="right">2.4447950</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ15</td>
<td align="right">0</td>
<td align="right">4439</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ15</td>
<td align="right">115</td>
<td align="right">4439</td>
<td align="right">2.5906736</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ15</td>
<td align="right">78</td>
<td align="right">4439</td>
<td align="right">1.7571525</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ15</td>
<td align="right">961</td>
<td align="right">4439</td>
<td align="right">21.6490200</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ15</td>
<td align="right">1185</td>
<td align="right">4439</td>
<td align="right">26.6952016</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ15</td>
<td align="right">149</td>
<td align="right">4439</td>
<td align="right">3.3566118</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ15</td>
<td align="right">0</td>
<td align="right">4439</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ15</td>
<td align="right">76</td>
<td align="right">4439</td>
<td align="right">1.7120973</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ15</td>
<td align="right">1299</td>
<td align="right">4439</td>
<td align="right">29.2633476</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ15</td>
<td align="right">174</td>
<td align="right">4439</td>
<td align="right">3.9198018</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ15</td>
<td align="right">256</td>
<td align="right">4439</td>
<td align="right">5.7670647</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ15</td>
<td align="right">146</td>
<td align="right">4439</td>
<td align="right">3.2890291</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ16</td>
<td align="right">0</td>
<td align="right">436</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ16</td>
<td align="right">25</td>
<td align="right">436</td>
<td align="right">5.7339450</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ16</td>
<td align="right">21</td>
<td align="right">436</td>
<td align="right">4.8165138</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ16</td>
<td align="right">54</td>
<td align="right">436</td>
<td align="right">12.3853211</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ16</td>
<td align="right">126</td>
<td align="right">436</td>
<td align="right">28.8990826</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ16</td>
<td align="right">32</td>
<td align="right">436</td>
<td align="right">7.3394495</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ16</td>
<td align="right">0</td>
<td align="right">436</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ16</td>
<td align="right">6</td>
<td align="right">436</td>
<td align="right">1.3761468</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ16</td>
<td align="right">18</td>
<td align="right">436</td>
<td align="right">4.1284404</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ16</td>
<td align="right">24</td>
<td align="right">436</td>
<td align="right">5.5045872</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ16</td>
<td align="right">22</td>
<td align="right">436</td>
<td align="right">5.0458716</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ16</td>
<td align="right">108</td>
<td align="right">436</td>
<td align="right">24.7706422</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ17</td>
<td align="right">0</td>
<td align="right">1370</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ17</td>
<td align="right">71</td>
<td align="right">1370</td>
<td align="right">5.1824818</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ17</td>
<td align="right">20</td>
<td align="right">1370</td>
<td align="right">1.4598540</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ17</td>
<td align="right">167</td>
<td align="right">1370</td>
<td align="right">12.1897810</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ17</td>
<td align="right">377</td>
<td align="right">1370</td>
<td align="right">27.5182482</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ17</td>
<td align="right">135</td>
<td align="right">1370</td>
<td align="right">9.8540146</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ17</td>
<td align="right">0</td>
<td align="right">1370</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ17</td>
<td align="right">28</td>
<td align="right">1370</td>
<td align="right">2.0437956</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ17</td>
<td align="right">144</td>
<td align="right">1370</td>
<td align="right">10.5109489</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ17</td>
<td align="right">200</td>
<td align="right">1370</td>
<td align="right">14.5985401</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ17</td>
<td align="right">45</td>
<td align="right">1370</td>
<td align="right">3.2846715</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ17</td>
<td align="right">183</td>
<td align="right">1370</td>
<td align="right">13.3576642</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ18</td>
<td align="right">0</td>
<td align="right">2280</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ18</td>
<td align="right">329</td>
<td align="right">2280</td>
<td align="right">14.4298246</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ18</td>
<td align="right">182</td>
<td align="right">2280</td>
<td align="right">7.9824561</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ18</td>
<td align="right">456</td>
<td align="right">2280</td>
<td align="right">20.0000000</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ18</td>
<td align="right">630</td>
<td align="right">2280</td>
<td align="right">27.6315789</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ18</td>
<td align="right">23</td>
<td align="right">2280</td>
<td align="right">1.0087719</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ18</td>
<td align="right">6</td>
<td align="right">2280</td>
<td align="right">0.2631579</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ18</td>
<td align="right">28</td>
<td align="right">2280</td>
<td align="right">1.2280702</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ18</td>
<td align="right">380</td>
<td align="right">2280</td>
<td align="right">16.6666667</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ18</td>
<td align="right">73</td>
<td align="right">2280</td>
<td align="right">3.2017544</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ18</td>
<td align="right">144</td>
<td align="right">2280</td>
<td align="right">6.3157895</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ18</td>
<td align="right">29</td>
<td align="right">2280</td>
<td align="right">1.2719298</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ2</td>
<td align="right">1</td>
<td align="right">1684</td>
<td align="right">0.0593824</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ2</td>
<td align="right">107</td>
<td align="right">1684</td>
<td align="right">6.3539192</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ2</td>
<td align="right">127</td>
<td align="right">1684</td>
<td align="right">7.5415677</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ2</td>
<td align="right">312</td>
<td align="right">1684</td>
<td align="right">18.5273159</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ2</td>
<td align="right">667</td>
<td align="right">1684</td>
<td align="right">39.6080760</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ2</td>
<td align="right">20</td>
<td align="right">1684</td>
<td align="right">1.1876485</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ2</td>
<td align="right">5</td>
<td align="right">1684</td>
<td align="right">0.2969121</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ2</td>
<td align="right">28</td>
<td align="right">1684</td>
<td align="right">1.6627078</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ2</td>
<td align="right">264</td>
<td align="right">1684</td>
<td align="right">15.6769596</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ2</td>
<td align="right">63</td>
<td align="right">1684</td>
<td align="right">3.7410926</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ2</td>
<td align="right">57</td>
<td align="right">1684</td>
<td align="right">3.3847981</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ2</td>
<td align="right">33</td>
<td align="right">1684</td>
<td align="right">1.9596200</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ20</td>
<td align="right">0</td>
<td align="right">2706</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ20</td>
<td align="right">46</td>
<td align="right">2706</td>
<td align="right">1.6999261</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ20</td>
<td align="right">301</td>
<td align="right">2706</td>
<td align="right">11.1234294</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ20</td>
<td align="right">933</td>
<td align="right">2706</td>
<td align="right">34.4789357</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ20</td>
<td align="right">652</td>
<td align="right">2706</td>
<td align="right">24.0946046</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ20</td>
<td align="right">38</td>
<td align="right">2706</td>
<td align="right">1.4042868</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ20</td>
<td align="right">2</td>
<td align="right">2706</td>
<td align="right">0.0739098</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ20</td>
<td align="right">20</td>
<td align="right">2706</td>
<td align="right">0.7390983</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ20</td>
<td align="right">337</td>
<td align="right">2706</td>
<td align="right">12.4538064</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ20</td>
<td align="right">104</td>
<td align="right">2706</td>
<td align="right">3.8433112</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ20</td>
<td align="right">149</td>
<td align="right">2706</td>
<td align="right">5.5062823</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ20</td>
<td align="right">124</td>
<td align="right">2706</td>
<td align="right">4.5824095</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ3</td>
<td align="right">2</td>
<td align="right">2396</td>
<td align="right">0.0834725</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ3</td>
<td align="right">238</td>
<td align="right">2396</td>
<td align="right">9.9332220</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ3</td>
<td align="right">192</td>
<td align="right">2396</td>
<td align="right">8.0133556</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ3</td>
<td align="right">374</td>
<td align="right">2396</td>
<td align="right">15.6093489</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ3</td>
<td align="right">867</td>
<td align="right">2396</td>
<td align="right">36.1853088</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ3</td>
<td align="right">39</td>
<td align="right">2396</td>
<td align="right">1.6277129</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ3</td>
<td align="right">30</td>
<td align="right">2396</td>
<td align="right">1.2520868</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ3</td>
<td align="right">33</td>
<td align="right">2396</td>
<td align="right">1.3772955</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ3</td>
<td align="right">401</td>
<td align="right">2396</td>
<td align="right">16.7362270</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ3</td>
<td align="right">80</td>
<td align="right">2396</td>
<td align="right">3.3388982</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ3</td>
<td align="right">108</td>
<td align="right">2396</td>
<td align="right">4.5075125</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ3</td>
<td align="right">32</td>
<td align="right">2396</td>
<td align="right">1.3355593</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ4</td>
<td align="right">1</td>
<td align="right">545</td>
<td align="right">0.1834862</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ4</td>
<td align="right">45</td>
<td align="right">545</td>
<td align="right">8.2568807</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ4</td>
<td align="right">64</td>
<td align="right">545</td>
<td align="right">11.7431193</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ4</td>
<td align="right">119</td>
<td align="right">545</td>
<td align="right">21.8348624</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ4</td>
<td align="right">143</td>
<td align="right">545</td>
<td align="right">26.2385321</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ4</td>
<td align="right">20</td>
<td align="right">545</td>
<td align="right">3.6697248</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ4</td>
<td align="right">0</td>
<td align="right">545</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ4</td>
<td align="right">7</td>
<td align="right">545</td>
<td align="right">1.2844037</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ4</td>
<td align="right">55</td>
<td align="right">545</td>
<td align="right">10.0917431</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ4</td>
<td align="right">34</td>
<td align="right">545</td>
<td align="right">6.2385321</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ4</td>
<td align="right">31</td>
<td align="right">545</td>
<td align="right">5.6880734</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ4</td>
<td align="right">26</td>
<td align="right">545</td>
<td align="right">4.7706422</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ5</td>
<td align="right">0</td>
<td align="right">781</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ5</td>
<td align="right">39</td>
<td align="right">781</td>
<td align="right">4.9935980</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ5</td>
<td align="right">39</td>
<td align="right">781</td>
<td align="right">4.9935980</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ5</td>
<td align="right">207</td>
<td align="right">781</td>
<td align="right">26.5044814</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ5</td>
<td align="right">263</td>
<td align="right">781</td>
<td align="right">33.6747759</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ5</td>
<td align="right">10</td>
<td align="right">781</td>
<td align="right">1.2804097</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ5</td>
<td align="right">0</td>
<td align="right">781</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ5</td>
<td align="right">26</td>
<td align="right">781</td>
<td align="right">3.3290653</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ5</td>
<td align="right">117</td>
<td align="right">781</td>
<td align="right">14.9807939</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ5</td>
<td align="right">21</td>
<td align="right">781</td>
<td align="right">2.6888604</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ5</td>
<td align="right">41</td>
<td align="right">781</td>
<td align="right">5.2496799</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ5</td>
<td align="right">18</td>
<td align="right">781</td>
<td align="right">2.3047375</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ6</td>
<td align="right">0</td>
<td align="right">491</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ6</td>
<td align="right">12</td>
<td align="right">491</td>
<td align="right">2.4439919</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ6</td>
<td align="right">46</td>
<td align="right">491</td>
<td align="right">9.3686354</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ6</td>
<td align="right">108</td>
<td align="right">491</td>
<td align="right">21.9959267</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ6</td>
<td align="right">156</td>
<td align="right">491</td>
<td align="right">31.7718941</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ6</td>
<td align="right">15</td>
<td align="right">491</td>
<td align="right">3.0549898</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ6</td>
<td align="right">0</td>
<td align="right">491</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ6</td>
<td align="right">2</td>
<td align="right">491</td>
<td align="right">0.4073320</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ6</td>
<td align="right">54</td>
<td align="right">491</td>
<td align="right">10.9979633</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ6</td>
<td align="right">47</td>
<td align="right">491</td>
<td align="right">9.5723014</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ6</td>
<td align="right">19</td>
<td align="right">491</td>
<td align="right">3.8696538</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ6</td>
<td align="right">32</td>
<td align="right">491</td>
<td align="right">6.5173116</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">GZ7</td>
<td align="right">45</td>
<td align="right">653</td>
<td align="right">6.8912711</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">GZ7</td>
<td align="right">90</td>
<td align="right">653</td>
<td align="right">13.7825421</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">GZ7</td>
<td align="right">12</td>
<td align="right">653</td>
<td align="right">1.8376723</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">GZ7</td>
<td align="right">111</td>
<td align="right">653</td>
<td align="right">16.9984686</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">GZ7</td>
<td align="right">240</td>
<td align="right">653</td>
<td align="right">36.7534456</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">GZ7</td>
<td align="right">5</td>
<td align="right">653</td>
<td align="right">0.7656968</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">GZ7</td>
<td align="right">3</td>
<td align="right">653</td>
<td align="right">0.4594181</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">GZ7</td>
<td align="right">8</td>
<td align="right">653</td>
<td align="right">1.2251149</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">GZ7</td>
<td align="right">63</td>
<td align="right">653</td>
<td align="right">9.6477795</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">GZ7</td>
<td align="right">12</td>
<td align="right">653</td>
<td align="right">1.8376723</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">GZ7</td>
<td align="right">58</td>
<td align="right">653</td>
<td align="right">8.8820827</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">GZ7</td>
<td align="right">6</td>
<td align="right">653</td>
<td align="right">0.9188361</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG17</td>
<td align="right">1</td>
<td align="right">1321</td>
<td align="right">0.0757002</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG17</td>
<td align="right">56</td>
<td align="right">1321</td>
<td align="right">4.2392127</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG17</td>
<td align="right">23</td>
<td align="right">1321</td>
<td align="right">1.7411052</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG17</td>
<td align="right">826</td>
<td align="right">1321</td>
<td align="right">62.5283876</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG17</td>
<td align="right">165</td>
<td align="right">1321</td>
<td align="right">12.4905375</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG17</td>
<td align="right">114</td>
<td align="right">1321</td>
<td align="right">8.6298259</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG17</td>
<td align="right">1</td>
<td align="right">1321</td>
<td align="right">0.0757002</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG17</td>
<td align="right">19</td>
<td align="right">1321</td>
<td align="right">1.4383043</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG17</td>
<td align="right">46</td>
<td align="right">1321</td>
<td align="right">3.4822104</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG17</td>
<td align="right">13</td>
<td align="right">1321</td>
<td align="right">0.9841030</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG17</td>
<td align="right">40</td>
<td align="right">1321</td>
<td align="right">3.0280091</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG17</td>
<td align="right">17</td>
<td align="right">1321</td>
<td align="right">1.2869039</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG21</td>
<td align="right">1</td>
<td align="right">1536</td>
<td align="right">0.0651042</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG21</td>
<td align="right">96</td>
<td align="right">1536</td>
<td align="right">6.2500000</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG21</td>
<td align="right">39</td>
<td align="right">1536</td>
<td align="right">2.5390625</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG21</td>
<td align="right">186</td>
<td align="right">1536</td>
<td align="right">12.1093750</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG21</td>
<td align="right">613</td>
<td align="right">1536</td>
<td align="right">39.9088542</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG21</td>
<td align="right">29</td>
<td align="right">1536</td>
<td align="right">1.8880208</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG21</td>
<td align="right">1</td>
<td align="right">1536</td>
<td align="right">0.0651042</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG21</td>
<td align="right">7</td>
<td align="right">1536</td>
<td align="right">0.4557292</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG21</td>
<td align="right">302</td>
<td align="right">1536</td>
<td align="right">19.6614583</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG21</td>
<td align="right">126</td>
<td align="right">1536</td>
<td align="right">8.2031250</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG21</td>
<td align="right">93</td>
<td align="right">1536</td>
<td align="right">6.0546875</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG21</td>
<td align="right">43</td>
<td align="right">1536</td>
<td align="right">2.7994792</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG24</td>
<td align="right">3</td>
<td align="right">2381</td>
<td align="right">0.1259975</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG24</td>
<td align="right">299</td>
<td align="right">2381</td>
<td align="right">12.5577488</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG24</td>
<td align="right">92</td>
<td align="right">2381</td>
<td align="right">3.8639227</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG24</td>
<td align="right">376</td>
<td align="right">2381</td>
<td align="right">15.7916842</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG24</td>
<td align="right">882</td>
<td align="right">2381</td>
<td align="right">37.0432591</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG24</td>
<td align="right">53</td>
<td align="right">2381</td>
<td align="right">2.2259555</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG24</td>
<td align="right">9</td>
<td align="right">2381</td>
<td align="right">0.3779924</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG24</td>
<td align="right">27</td>
<td align="right">2381</td>
<td align="right">1.1339773</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG24</td>
<td align="right">285</td>
<td align="right">2381</td>
<td align="right">11.9697606</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG24</td>
<td align="right">118</td>
<td align="right">2381</td>
<td align="right">4.9559009</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG24</td>
<td align="right">139</td>
<td align="right">2381</td>
<td align="right">5.8378832</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG24</td>
<td align="right">98</td>
<td align="right">2381</td>
<td align="right">4.1159177</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG25</td>
<td align="right">49</td>
<td align="right">3131</td>
<td align="right">1.5649952</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG25</td>
<td align="right">397</td>
<td align="right">3131</td>
<td align="right">12.6796551</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG25</td>
<td align="right">98</td>
<td align="right">3131</td>
<td align="right">3.1299904</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG25</td>
<td align="right">548</td>
<td align="right">3131</td>
<td align="right">17.5023954</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG25</td>
<td align="right">871</td>
<td align="right">3131</td>
<td align="right">27.8185883</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG25</td>
<td align="right">102</td>
<td align="right">3131</td>
<td align="right">3.2577451</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG25</td>
<td align="right">0</td>
<td align="right">3131</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG25</td>
<td align="right">27</td>
<td align="right">3131</td>
<td align="right">0.8623443</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG25</td>
<td align="right">437</td>
<td align="right">3131</td>
<td align="right">13.9572022</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG25</td>
<td align="right">150</td>
<td align="right">3131</td>
<td align="right">4.7908017</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG25</td>
<td align="right">139</td>
<td align="right">3131</td>
<td align="right">4.4394762</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG25</td>
<td align="right">313</td>
<td align="right">3131</td>
<td align="right">9.9968061</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG28</td>
<td align="right">3</td>
<td align="right">1545</td>
<td align="right">0.1941748</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG28</td>
<td align="right">71</td>
<td align="right">1545</td>
<td align="right">4.5954693</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG28</td>
<td align="right">48</td>
<td align="right">1545</td>
<td align="right">3.1067961</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG28</td>
<td align="right">506</td>
<td align="right">1545</td>
<td align="right">32.7508091</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG28</td>
<td align="right">316</td>
<td align="right">1545</td>
<td align="right">20.4530744</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG28</td>
<td align="right">10</td>
<td align="right">1545</td>
<td align="right">0.6472492</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG28</td>
<td align="right">0</td>
<td align="right">1545</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG28</td>
<td align="right">26</td>
<td align="right">1545</td>
<td align="right">1.6828479</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG28</td>
<td align="right">406</td>
<td align="right">1545</td>
<td align="right">26.2783172</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG28</td>
<td align="right">8</td>
<td align="right">1545</td>
<td align="right">0.5177994</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG28</td>
<td align="right">134</td>
<td align="right">1545</td>
<td align="right">8.6731392</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG28</td>
<td align="right">17</td>
<td align="right">1545</td>
<td align="right">1.1003236</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG29</td>
<td align="right">1</td>
<td align="right">2496</td>
<td align="right">0.0400641</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG29</td>
<td align="right">182</td>
<td align="right">2496</td>
<td align="right">7.2916667</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG29</td>
<td align="right">99</td>
<td align="right">2496</td>
<td align="right">3.9663462</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG29</td>
<td align="right">461</td>
<td align="right">2496</td>
<td align="right">18.4695513</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG29</td>
<td align="right">422</td>
<td align="right">2496</td>
<td align="right">16.9070513</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG29</td>
<td align="right">379</td>
<td align="right">2496</td>
<td align="right">15.1842949</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG29</td>
<td align="right">9</td>
<td align="right">2496</td>
<td align="right">0.3605769</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG29</td>
<td align="right">21</td>
<td align="right">2496</td>
<td align="right">0.8413462</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG29</td>
<td align="right">297</td>
<td align="right">2496</td>
<td align="right">11.8990385</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG29</td>
<td align="right">391</td>
<td align="right">2496</td>
<td align="right">15.6650641</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG29</td>
<td align="right">113</td>
<td align="right">2496</td>
<td align="right">4.5272436</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG29</td>
<td align="right">121</td>
<td align="right">2496</td>
<td align="right">4.8477564</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG31</td>
<td align="right">0</td>
<td align="right">1192</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG31</td>
<td align="right">178</td>
<td align="right">1192</td>
<td align="right">14.9328859</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG31</td>
<td align="right">60</td>
<td align="right">1192</td>
<td align="right">5.0335570</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG31</td>
<td align="right">197</td>
<td align="right">1192</td>
<td align="right">16.5268456</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG31</td>
<td align="right">354</td>
<td align="right">1192</td>
<td align="right">29.6979866</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG31</td>
<td align="right">35</td>
<td align="right">1192</td>
<td align="right">2.9362416</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG31</td>
<td align="right">0</td>
<td align="right">1192</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG31</td>
<td align="right">22</td>
<td align="right">1192</td>
<td align="right">1.8456376</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG31</td>
<td align="right">179</td>
<td align="right">1192</td>
<td align="right">15.0167785</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG31</td>
<td align="right">91</td>
<td align="right">1192</td>
<td align="right">7.6342282</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG31</td>
<td align="right">51</td>
<td align="right">1192</td>
<td align="right">4.2785235</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG31</td>
<td align="right">25</td>
<td align="right">1192</td>
<td align="right">2.0973154</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG32</td>
<td align="right">5</td>
<td align="right">1428</td>
<td align="right">0.3501401</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG32</td>
<td align="right">40</td>
<td align="right">1428</td>
<td align="right">2.8011204</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG32</td>
<td align="right">73</td>
<td align="right">1428</td>
<td align="right">5.1120448</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG32</td>
<td align="right">511</td>
<td align="right">1428</td>
<td align="right">35.7843137</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG32</td>
<td align="right">324</td>
<td align="right">1428</td>
<td align="right">22.6890756</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG32</td>
<td align="right">15</td>
<td align="right">1428</td>
<td align="right">1.0504202</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG32</td>
<td align="right">2</td>
<td align="right">1428</td>
<td align="right">0.1400560</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG32</td>
<td align="right">27</td>
<td align="right">1428</td>
<td align="right">1.8907563</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG32</td>
<td align="right">240</td>
<td align="right">1428</td>
<td align="right">16.8067227</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG32</td>
<td align="right">55</td>
<td align="right">1428</td>
<td align="right">3.8515406</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG32</td>
<td align="right">82</td>
<td align="right">1428</td>
<td align="right">5.7422969</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG32</td>
<td align="right">54</td>
<td align="right">1428</td>
<td align="right">3.7815126</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG33</td>
<td align="right">8</td>
<td align="right">6286</td>
<td align="right">0.1272669</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG33</td>
<td align="right">216</td>
<td align="right">6286</td>
<td align="right">3.4362074</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG33</td>
<td align="right">82</td>
<td align="right">6286</td>
<td align="right">1.3044862</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG33</td>
<td align="right">189</td>
<td align="right">6286</td>
<td align="right">3.0066815</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG33</td>
<td align="right">414</td>
<td align="right">6286</td>
<td align="right">6.5860643</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG33</td>
<td align="right">1393</td>
<td align="right">6286</td>
<td align="right">22.1603563</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG33</td>
<td align="right">0</td>
<td align="right">6286</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG33</td>
<td align="right">38</td>
<td align="right">6286</td>
<td align="right">0.6045180</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG33</td>
<td align="right">178</td>
<td align="right">6286</td>
<td align="right">2.8316895</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG33</td>
<td align="right">651</td>
<td align="right">6286</td>
<td align="right">10.3563474</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG33</td>
<td align="right">35</td>
<td align="right">6286</td>
<td align="right">0.5567929</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG33</td>
<td align="right">3082</td>
<td align="right">6286</td>
<td align="right">49.0295896</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG34</td>
<td align="right">0</td>
<td align="right">620</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG34</td>
<td align="right">177</td>
<td align="right">620</td>
<td align="right">28.5483871</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG34</td>
<td align="right">18</td>
<td align="right">620</td>
<td align="right">2.9032258</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG34</td>
<td align="right">82</td>
<td align="right">620</td>
<td align="right">13.2258065</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG34</td>
<td align="right">120</td>
<td align="right">620</td>
<td align="right">19.3548387</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG34</td>
<td align="right">8</td>
<td align="right">620</td>
<td align="right">1.2903226</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG34</td>
<td align="right">0</td>
<td align="right">620</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG34</td>
<td align="right">5</td>
<td align="right">620</td>
<td align="right">0.8064516</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG34</td>
<td align="right">122</td>
<td align="right">620</td>
<td align="right">19.6774194</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG34</td>
<td align="right">42</td>
<td align="right">620</td>
<td align="right">6.7741935</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG34</td>
<td align="right">22</td>
<td align="right">620</td>
<td align="right">3.5483871</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG34</td>
<td align="right">24</td>
<td align="right">620</td>
<td align="right">3.8709677</td>
</tr>
<tr class="odd">
<td align="left">Adipocytes</td>
<td align="left">SG35</td>
<td align="right">0</td>
<td align="right">2363</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Cardiomyocyte</td>
<td align="left">SG35</td>
<td align="right">303</td>
<td align="right">2363</td>
<td align="right">12.8226830</td>
</tr>
<tr class="odd">
<td align="left">EndoEC</td>
<td align="left">SG35</td>
<td align="right">61</td>
<td align="right">2363</td>
<td align="right">2.5814642</td>
</tr>
<tr class="even">
<td align="left">Endothelial</td>
<td align="left">SG35</td>
<td align="right">522</td>
<td align="right">2363</td>
<td align="right">22.0905628</td>
</tr>
<tr class="odd">
<td align="left">Fibroblast</td>
<td align="left">SG35</td>
<td align="right">794</td>
<td align="right">2363</td>
<td align="right">33.6013542</td>
</tr>
<tr class="even">
<td align="left">infMacrophage</td>
<td align="left">SG35</td>
<td align="right">55</td>
<td align="right">2363</td>
<td align="right">2.3275497</td>
</tr>
<tr class="odd">
<td align="left">LEC</td>
<td align="left">SG35</td>
<td align="right">0</td>
<td align="right">2363</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">NeuralCells</td>
<td align="left">SG35</td>
<td align="right">28</td>
<td align="right">2363</td>
<td align="right">1.1849344</td>
</tr>
<tr class="odd">
<td align="left">Perivascular</td>
<td align="left">SG35</td>
<td align="right">322</td>
<td align="right">2363</td>
<td align="right">13.6267457</td>
</tr>
<tr class="even">
<td align="left">resMacrophage</td>
<td align="left">SG35</td>
<td align="right">145</td>
<td align="right">2363</td>
<td align="right">6.1362675</td>
</tr>
<tr class="odd">
<td align="left">SMC</td>
<td align="left">SG35</td>
<td align="right">73</td>
<td align="right">2363</td>
<td align="right">3.0892933</td>
</tr>
<tr class="even">
<td align="left">Tcell</td>
<td align="left">SG35</td>
<td align="right">60</td>
<td align="right">2363</td>
<td align="right">2.5391452</td>
</tr>
</tbody>
</table>
<pre class="r"><code>ordVec &lt;- datFrac %&gt;% dplyr::filter(label==&quot;Tcell&quot;) %&gt;% 
  arrange(., percentage)
ordBar &lt;- c(&quot;Tcell&quot;,&quot;infMacrophage&quot;,&quot;resMacrophage&quot;,&quot;Fibroblast&quot;,&quot;Perivascular&quot;,
            &quot;SMC&quot;,&quot;Endothelial&quot;,&quot;EndoEC&quot;,&quot;LEC&quot;,&quot;Cardiomyocyte&quot;,&quot;Adipocytes&quot;,
            &quot;NeuralCells&quot;)
datFrac &lt;-datFrac %&gt;% mutate(labelFac=factor(label, levels = ordBar))
ggbarplot(datFrac, x=&quot;ID&quot;, y=&quot;percentage&quot;,
          fill = &quot;labelFac&quot;,
          palette = colLab,
          order= ordVec$ID) +
  rotate_x_text(angle = 90)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/cnt%20tables%20per%20pat-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## only Myocarditis patients
selMyo &lt;- unique(seurat$ID[which(seurat$cond != &quot;HH&quot;)])
datFracSel &lt;- datFrac %&gt;% filter(ID %in% selMyo)
ggbarplot(datFracSel, x=&quot;ID&quot;, y=&quot;percentage&quot;,
          fill = &quot;labelFac&quot;,
          palette = colLab,
          order= ordVec$ID) +
  rotate_x_text(angle = 90)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/cnt%20tables%20per%20pat-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="assign-groups-based-on-t-cell-fraction" class="section level2">
<h2>assign groups based on T cell fraction</h2>
</div>
<div id="vis-marker-for-celltype-assignment" class="section level2">
<h2>vis marker for celltype assignment</h2>
<div id="heatmap" class="section level3">
<h3>heatmap</h3>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 

selGenesAll &lt;- read_tsv(file = paste0(basedir,
                                      &quot;/data/markerLabels.txt&quot;)) %&gt;% 
  left_join(., genes, by = &quot;geneID&quot;)

Idents(seurat) &lt;- seurat$seurat_clusters
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesAll,
                  colVecIdent = colPal, 
                  ordVec=levels(seurat),
                  gapVecR=NULL, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20sel%20marker-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$label
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesAll,
                  colVecIdent = colLab, 
                  ordVec=levels(seurat),
                  gapVecR=NULL, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vis%20sel%20marker-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="dotplot" class="section level3">
<h3>Dotplot</h3>
<pre class="r"><code>DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenesAll$gene, scale =T,
        cluster.idents = T) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenesAll$gene, labels=selGenesAll$geneID) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/dotplot-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$seurat_clusters
DotPlot(seurat, assay=&quot;RNA&quot;, features = selGenesAll$gene, scale =T,
        cluster.idents = T) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenesAll$gene, labels=selGenesAll$geneID) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/dotplot-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="violin-plots-selected-marker" class="section level2">
<h2>violin plots selected marker</h2>
<pre class="r"><code>genesDat &lt;- data.frame(EnsID=rownames(seurat)) %&gt;% 
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;, EnsID))
selGenes &lt;- data.frame(gene=c(&quot;CD2&quot;, &quot;TNNT2&quot;, &quot;PECAM1&quot;, &quot;NRG1&quot;, &quot;PROX1&quot;,
                              &quot;PDGFRA&quot;, &quot;RGS5&quot;, &quot;MYH11&quot;, &quot;C1QA&quot;, &quot;NRXN1&quot;,
                              &quot;PLIN1&quot;)) %&gt;% 
  left_join(., genesDat, by=&quot;gene&quot;)

pList &lt;- sapply(selGenes$EnsID, function(x){
  p &lt;- VlnPlot(object = seurat, features = x,
               group.by = &quot;label&quot;,
               cols = colLab, pt.size = 0
               ) + 
    theme(legend.position = &quot;none&quot;)
    
  plot(p)
})</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pList &lt;- sapply(selGenes$EnsID, function(x){
  p &lt;- VlnPlot(object = seurat, features = x,
               group.by = &quot;label&quot;,
               cols = colLab, pt.size = 0.3
               ) + 
    theme(legend.position = &quot;none&quot;)
    
  plot(p)
})</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-16.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-18.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-20.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-21.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-22.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-23.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/vlnplt%20sel%20marker-24.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="feature-plots-bmps" class="section level2">
<h2>Feature plots BMPs</h2>
<pre class="r"><code>## list with all gene names for mapping of EnsIDs
genesDat &lt;- data.frame(EnsID=rownames(seurat)) %&gt;% 
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;, EnsID))

## selected genes to plot
selGenes &lt;- data.frame(gene=c(&quot;BMP2&quot;, &quot;BMP4&quot;, &quot;BMPR1A&quot;, &quot;BMPR2&quot;)) %&gt;% 
  left_join(., genesDat, by=&quot;gene&quot;)

## plotting loop order=F
pList &lt;- sapply(selGenes$EnsID, function(x){
p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = x,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = F)+
  theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/feautreplots%20BMPs-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/feautreplots%20BMPs-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/feautreplots%20BMPs-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/feautreplots%20BMPs-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## plotting loop order=T
pList &lt;- sapply(selGenes$EnsID, function(x){
p &lt;- FeaturePlot(seurat, reduction = &quot;umap&quot;, 
            features = x,
            cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
<p><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/feautreplots%20BMPs-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/feautreplots%20BMPs-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/feautreplots%20BMPs-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd/feautreplots%20BMPs-8.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="average-counts-bmps-in-fibroblasts" class="section level2">
<h2>average counts BMPs in Fibroblasts</h2>
<pre class="r"><code>seuratSub &lt;- subset(seurat, label==&quot;Fibroblast&quot;)

## assay data
clusterAssigned &lt;- as.data.frame(seuratSub$ID) %&gt;%
  dplyr::mutate(cell=rownames(.))
colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
seuratDat &lt;- GetAssayData(seuratSub)
  
  ## genes of interest
genes &lt;- data.frame(gene=rownames(seuratSub)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes$gene)

  ## matrix with averaged cnts per ident
logNormExpres &lt;- as.data.frame(t(as.matrix(
  seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
  dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
  dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
  dplyr::summarise_all(mean)

write.table(logNormExpres, 
            file=paste0(basedir, &quot;/data/BmpCntsFibroblastsPerPatient_woHH.txt&quot;),
            row.names = F, col.names = T, sep = &quot;\t&quot;, quote = F)</code></pre>
</div>
<div id="save-seurat-object" class="section level2">
<h2>save seurat object</h2>
<pre class="r"><code>saveRDS(seurat, file = paste0(basedir, 
                              &quot;/data/humanHeartsPlusGraz_intPatients_merged&quot;, 
                              &quot;labeled_woHH_seurat.rds&quot;))</code></pre>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] sctransform_0.3.5           viridis_0.6.2              
 [3] viridisLite_0.4.1           pheatmap_1.0.12            
 [5] ggpubr_0.4.0                ggsci_2.9                  
 [7] runSeurat3_0.1.0            here_1.0.1                 
 [9] magrittr_2.0.3              sp_1.5-0                   
[11] SeuratObject_4.1.2          Seurat_4.2.0               
[13] forcats_0.5.2               stringr_1.4.1              
[15] dplyr_1.0.10                purrr_0.3.5                
[17] readr_2.1.3                 tidyr_1.2.1                
[19] tibble_3.1.8                ggplot2_3.3.6              
[21] tidyverse_1.3.2             SingleCellExperiment_1.18.1
[23] SummarizedExperiment_1.26.1 Biobase_2.56.0             
[25] GenomicRanges_1.48.0        GenomeInfoDb_1.32.4        
[27] IRanges_2.30.1              S4Vectors_0.34.0           
[29] BiocGenerics_0.42.0         MatrixGenerics_1.8.1       
[31] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             reticulate_1.26        tidyselect_1.2.0      
  [4] htmlwidgets_1.5.4      grid_4.2.1             Rtsne_0.16            
  [7] munsell_0.5.0          codetools_0.2-18       ica_1.0-3             
 [10] future_1.28.0          miniUI_0.1.1.1         withr_2.5.0           
 [13] spatstat.random_2.2-0  colorspace_2.0-3       progressr_0.11.0      
 [16] highr_0.9              knitr_1.40             rstudioapi_0.14       
 [19] ROCR_1.0-11            ggsignif_0.6.4         tensor_1.5            
 [22] listenv_0.8.0          labeling_0.4.2         git2r_0.30.1          
 [25] GenomeInfoDbData_1.2.8 polyclip_1.10-4        bit64_4.0.5           
 [28] farver_2.1.1           rprojroot_2.0.3        parallelly_1.32.1     
 [31] vctrs_0.5.0            generics_0.1.3         xfun_0.34             
 [34] R6_2.5.1               ggbeeswarm_0.6.0       bitops_1.0-7          
 [37] spatstat.utils_3.0-1   cachem_1.0.6           DelayedArray_0.22.0   
 [40] assertthat_0.2.1       vroom_1.6.0            promises_1.2.0.1      
 [43] scales_1.2.1           googlesheets4_1.0.1    beeswarm_0.4.0        
 [46] rgeos_0.5-9            gtable_0.3.1           globals_0.16.1        
 [49] goftest_1.2-3          workflowr_1.7.0        rlang_1.0.6           
 [52] splines_4.2.1          rstatix_0.7.0          lazyeval_0.2.2        
 [55] gargle_1.2.1           spatstat.geom_2.4-0    broom_1.0.1           
 [58] yaml_2.3.6             reshape2_1.4.4         abind_1.4-5           
 [61] modelr_0.1.9           backports_1.4.1        httpuv_1.6.6          
 [64] tools_4.2.1            ellipsis_0.3.2         spatstat.core_2.4-4   
 [67] jquerylib_0.1.4        RColorBrewer_1.1-3     ggridges_0.5.4        
 [70] Rcpp_1.0.9             plyr_1.8.7             zlibbioc_1.42.0       
 [73] RCurl_1.98-1.9         rpart_4.1.19           deldir_1.0-6          
 [76] pbapply_1.5-0          cowplot_1.1.1          zoo_1.8-11            
 [79] haven_2.5.1            ggrepel_0.9.1          cluster_2.1.4         
 [82] fs_1.5.2               data.table_1.14.4      scattermore_0.8       
 [85] lmtest_0.9-40          reprex_2.0.2           RANN_2.6.1            
 [88] googledrive_2.0.0      fitdistrplus_1.1-8     hms_1.1.2             
 [91] patchwork_1.1.2        mime_0.12              evaluate_0.17         
 [94] xtable_1.8-4           readxl_1.4.1           gridExtra_2.3         
 [97] compiler_4.2.1         KernSmooth_2.23-20     crayon_1.5.2          
[100] htmltools_0.5.3        mgcv_1.8-41            later_1.3.0           
[103] tzdb_0.3.0             lubridate_1.8.0        DBI_1.1.3             
[106] dbplyr_2.2.1           MASS_7.3-58.1          Matrix_1.5-1          
[109] car_3.1-1              cli_3.4.1              parallel_4.2.1        
[112] igraph_1.3.5           pkgconfig_2.0.3        plotly_4.10.0         
[115] spatstat.sparse_3.0-0  xml2_1.3.3             vipor_0.4.5           
[118] bslib_0.4.0            XVector_0.36.0         rvest_1.0.3           
[121] digest_0.6.30          RcppAnnoy_0.0.19       spatstat.data_3.0-0   
[124] rmarkdown_2.17         cellranger_1.1.0       leiden_0.4.3          
[127] uwot_0.1.14            shiny_1.7.2            lifecycle_1.0.3       
[130] nlme_3.1-160           jsonlite_1.8.3         carData_3.0-5         
[133] fansi_1.0.3            pillar_1.8.1           lattice_0.20-45       
[136] ggrastr_1.0.1          fastmap_1.1.0          httr_1.4.4            
[139] survival_3.4-0         glue_1.6.2             png_0.1-7             
[142] bit_4.0.4              stringi_1.7.8          sass_0.4.2            
[145] irlba_2.3.5.1          future.apply_1.9.1    </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Tue Nov 29 15:51:21 2022&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] sctransform_0.3.5           viridis_0.6.2              
 [3] viridisLite_0.4.1           pheatmap_1.0.12            
 [5] ggpubr_0.4.0                ggsci_2.9                  
 [7] runSeurat3_0.1.0            here_1.0.1                 
 [9] magrittr_2.0.3              sp_1.5-0                   
[11] SeuratObject_4.1.2          Seurat_4.2.0               
[13] forcats_0.5.2               stringr_1.4.1              
[15] dplyr_1.0.10                purrr_0.3.5                
[17] readr_2.1.3                 tidyr_1.2.1                
[19] tibble_3.1.8                ggplot2_3.3.6              
[21] tidyverse_1.3.2             SingleCellExperiment_1.18.1
[23] SummarizedExperiment_1.26.1 Biobase_2.56.0             
[25] GenomicRanges_1.48.0        GenomeInfoDb_1.32.4        
[27] IRanges_2.30.1              S4Vectors_0.34.0           
[29] BiocGenerics_0.42.0         MatrixGenerics_1.8.1       
[31] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             reticulate_1.26        tidyselect_1.2.0      
  [4] htmlwidgets_1.5.4      grid_4.2.1             Rtsne_0.16            
  [7] munsell_0.5.0          codetools_0.2-18       ica_1.0-3             
 [10] future_1.28.0          miniUI_0.1.1.1         withr_2.5.0           
 [13] spatstat.random_2.2-0  colorspace_2.0-3       progressr_0.11.0      
 [16] highr_0.9              knitr_1.40             rstudioapi_0.14       
 [19] ROCR_1.0-11            ggsignif_0.6.4         tensor_1.5            
 [22] listenv_0.8.0          labeling_0.4.2         git2r_0.30.1          
 [25] GenomeInfoDbData_1.2.8 polyclip_1.10-4        bit64_4.0.5           
 [28] farver_2.1.1           rprojroot_2.0.3        parallelly_1.32.1     
 [31] vctrs_0.5.0            generics_0.1.3         xfun_0.34             
 [34] R6_2.5.1               ggbeeswarm_0.6.0       bitops_1.0-7          
 [37] spatstat.utils_3.0-1   cachem_1.0.6           DelayedArray_0.22.0   
 [40] assertthat_0.2.1       vroom_1.6.0            promises_1.2.0.1      
 [43] scales_1.2.1           googlesheets4_1.0.1    beeswarm_0.4.0        
 [46] rgeos_0.5-9            gtable_0.3.1           globals_0.16.1        
 [49] goftest_1.2-3          workflowr_1.7.0        rlang_1.0.6           
 [52] splines_4.2.1          rstatix_0.7.0          lazyeval_0.2.2        
 [55] gargle_1.2.1           spatstat.geom_2.4-0    broom_1.0.1           
 [58] yaml_2.3.6             reshape2_1.4.4         abind_1.4-5           
 [61] modelr_0.1.9           backports_1.4.1        httpuv_1.6.6          
 [64] tools_4.2.1            ellipsis_0.3.2         spatstat.core_2.4-4   
 [67] jquerylib_0.1.4        RColorBrewer_1.1-3     ggridges_0.5.4        
 [70] Rcpp_1.0.9             plyr_1.8.7             zlibbioc_1.42.0       
 [73] RCurl_1.98-1.9         rpart_4.1.19           deldir_1.0-6          
 [76] pbapply_1.5-0          cowplot_1.1.1          zoo_1.8-11            
 [79] haven_2.5.1            ggrepel_0.9.1          cluster_2.1.4         
 [82] fs_1.5.2               data.table_1.14.4      scattermore_0.8       
 [85] lmtest_0.9-40          reprex_2.0.2           RANN_2.6.1            
 [88] googledrive_2.0.0      fitdistrplus_1.1-8     hms_1.1.2             
 [91] patchwork_1.1.2        mime_0.12              evaluate_0.17         
 [94] xtable_1.8-4           readxl_1.4.1           gridExtra_2.3         
 [97] compiler_4.2.1         KernSmooth_2.23-20     crayon_1.5.2          
[100] htmltools_0.5.3        mgcv_1.8-41            later_1.3.0           
[103] tzdb_0.3.0             lubridate_1.8.0        DBI_1.1.3             
[106] dbplyr_2.2.1           MASS_7.3-58.1          Matrix_1.5-1          
[109] car_3.1-1              cli_3.4.1              parallel_4.2.1        
[112] igraph_1.3.5           pkgconfig_2.0.3        plotly_4.10.0         
[115] spatstat.sparse_3.0-0  xml2_1.3.3             vipor_0.4.5           
[118] bslib_0.4.0            XVector_0.36.0         rvest_1.0.3           
[121] digest_0.6.30          RcppAnnoy_0.0.19       spatstat.data_3.0-0   
[124] rmarkdown_2.17         cellranger_1.1.0       leiden_0.4.3          
[127] uwot_0.1.14            shiny_1.7.2            lifecycle_1.0.3       
[130] nlme_3.1-160           jsonlite_1.8.3         carData_3.0-5         
[133] fansi_1.0.3            pillar_1.8.1           lattice_0.20-45       
[136] ggrastr_1.0.1          fastmap_1.1.0          httr_1.4.4            
[139] survival_3.4-0         glue_1.6.2             png_0.1-7             
[142] bit_4.0.4              stringi_1.7.8          sass_0.4.2            
[145] irlba_2.3.5.1          future.apply_1.9.1    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
