<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />


<title>DE genes between T cell groups</title>

<script src="site_libs/header-attrs-2.18/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">humanCardiacFibroblasts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DE genes between T cell groups</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">Dec 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-12-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>humanCardiacFibroblasts/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210903code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210903)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210903code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210903)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststree1e723809868eb6a738ab7c8aa6a0bf244ea60da5targetblank1e72380a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/1e723809868eb6a738ab7c8aa6a0bf244ea60da5" target="_blank">1e72380</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststree1e723809868eb6a738ab7c8aa6a0bf244ea60da5targetblank1e72380a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/1e723809868eb6a738ab7c8aa6a0bf244ea60da5" target="_blank">1e72380</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSEA/
    Ignored:    data/humanFibroblast/
    Ignored:    figure/DEgenesGZplusSG_Groups.Rmd/.DS_Store

Untracked files:
    Untracked:  analysis/DEgenesTcellgrps.Rmd
    Untracked:  figure/DEgenesTcellgrps.Rmd/

Unstaged changes:
    Modified:   analysis/assignLabelshumanHeartsPlusGrazIntWoHH.Rmd
    Modified:   analysis/integrateAcrossPatientsGZplusSG.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(SingleCellExperiment)
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(ggsci)
  library(pheatmap)
  library(ggpubr)
  library(clusterProfiler)
  library(org.Hs.eg.db)
  library(DOSE)
  library(enrichplot)
  library(viridis)
})</code></pre>
</div>
<div id="heatmap-function" class="section level2">
<h2>heatmap function</h2>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  selGenes &lt;- selGenes$gene
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;.*_&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colVecIdent)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#2166AC&quot;, &quot;#F7F7F7&quot;, &quot;#B2182B&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}</code></pre>
</div>
<div id="set-dir" class="section level2">
<h2>set dir</h2>
<pre class="r"><code>basedir &lt;- here()
seurat &lt;- readRDS(file = paste0(basedir, 
                              &quot;/data/humanHeartsPlusGraz_intPatients_merged&quot;, 
                              &quot;labeled_woHH_seurat.rds&quot;))
Idents(seurat) &lt;- seurat$integrated_snn_res.0.6</code></pre>
</div>
<div id="color-vectors" class="section level2">
<h2>color vectors</h2>
<pre class="r"><code>colPal &lt;- c(pal_igv()(12),
            pal_aaas()(10))[1:length(levels(seurat))]
colTec &lt;- pal_jama()(length(unique(seurat$technique)))
colSmp &lt;- c(pal_uchicago()(9), pal_npg()(10), pal_aaas()(10), 
            pal_jama()(7))[1:length(unique(seurat$dataset))]
colCond &lt;- pal_npg()(length(unique(seurat$cond)))
colID &lt;- c(pal_jco()(10), pal_npg()(10), pal_futurama()(10),
           pal_d3()(10))[1:length(unique(seurat$ID))]
colOrig &lt;- pal_aaas()(length(unique(seurat$origin)))
colIso &lt;- pal_nejm()(length(unique(seurat$isolation)))
colProc &lt;- pal_aaas()(length(unique(seurat$processing)))
colTgrp &lt;- c(&quot;#9f2238&quot;, &quot;#48557e&quot;, &quot;#71737a&quot;)
colLab &lt;- c(&quot;#1b805d&quot;, &quot;#C71000FF&quot;, &quot;#344080&quot;, &quot;#803467&quot;, &quot;#5A9599FF&quot;,
            &quot;#FF6348FF&quot;, &quot;#84D7E1FF&quot;, &quot;#8A9045FF&quot;, &quot;#800000FF&quot;, &quot;#767676FF&quot;,
            &quot;#FFA319FF&quot;, &quot;#FF95A8FF&quot;) 

names(colLab) &lt;- c(&quot;EndoEC&quot;, &quot;Tcell&quot;,&quot;resMacrophage&quot;, &quot;Fibroblast&quot;,
                   &quot;infMacrophage&quot;, &quot;Perivascular&quot;,&quot;Cardiomyocyte&quot;,
                   &quot;Endothelial&quot;,&quot;Adipocytes&quot;,&quot;NeuralCells&quot;,&quot;SMC&quot;,&quot;LEC&quot;)
names(colTgrp) &lt;- c(&quot;TcellHigh&quot;, &quot;TcellLow&quot;, &quot;TcellInt&quot; )
names(colPal) &lt;- levels(seurat)
names(colTec) &lt;- unique(seurat$technique)
names(colSmp) &lt;- unique(seurat$dataset)
names(colCond) &lt;- unique(seurat$cond)
names(colID) &lt;- unique(seurat$ID)
names(colOrig) &lt;- unique(seurat$origin)
names(colIso) &lt;- unique(seurat$isolation)
names(colProc) &lt;- unique(seurat$processing)</code></pre>
</div>
<div id="vis-data" class="section level2 tabset">
<h2 class="tabset">vis data</h2>
<div id="clusters" class="section level3">
<h3>clusters</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal,
        shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20cluster-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="label" class="section level3">
<h3>label</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;,  group.by = &quot;label&quot;, cols=colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20label-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20label-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="id" class="section level3">
<h3>ID</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;ID&quot;, cols=colID, shuffle = T)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20ID-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;ID&quot;, cols=colID,
        shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20ID-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cond" class="section level3">
<h3>cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;cond&quot;, cols=colCond)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20cond-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="t-cell-grp" class="section level3">
<h3>T cell grp</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;TcellGrp&quot;, cols=colTgrp)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20T%20cell%20group-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;TcellGrp&quot;, cols=colTgrp,
        shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20T%20cell%20group-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="label-split-by-t-cell-grp" class="section level3">
<h3>label split by T cell Grp</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols=colLab,
        split.by = &quot;TcellGrp&quot;, shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/vis%20label%20split%20seq-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="overall-de-genes" class="section level2 tabset">
<h2 class="tabset">overall DE genes</h2>
<pre class="r"><code>## downsample to equal number of cells per celltype (min 300 cells as 
## T cell low grp has around 300 T cells)
Idents(seurat) &lt;- seurat$TcellGrp
seurat$label_plus_TcellGrp &lt;- paste0(seurat$label, &quot;_&quot;, seurat$TcellGrp)
Idents(seurat) &lt;- seurat$label_plus_TcellGrp
seuratSub &lt;- subset(seurat, downsample = 300)
table(seuratSub$label, seuratSub$TcellGrp)</code></pre>
<pre><code>               
                TcellHigh TcellInt TcellLow
  Adipocytes           63       11        5
  Cardiomyocyte       300      300      300
  EndoEC              300      300      300
  Endothelial         300      300      300
  Fibroblast          300      300      300
  infMacrophage       300      300      174
  LEC                  21       45       14
  NeuralCells         169      220      147
  Perivascular        300      300      300
  resMacrophage       300      300      300
  SMC                 300      300      300
  Tcell               300      300      231</code></pre>
<div id="overall-de-genes-all-grps" class="section level3">
<h3>overall DE genes all grps</h3>
<pre class="r"><code>seuratSub$label &lt;- factor(seuratSub$label)
Idents(seuratSub) &lt;- seuratSub$TcellGrp
overallDE &lt;- FindAllMarkers(object = seuratSub, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE &lt;- overallDE %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub$TcellGrp)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_allGrps.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_allGrps&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
</div>
<div id="t-cell-high-vs-t-cell-low" class="section level3">
<h3>T cell high vs T cell low</h3>
<pre class="r"><code>seuratSub2 &lt;- subset(seuratSub, TcellGrp %in% c(&quot;TcellHigh&quot;, &quot;TcellLow&quot;))
seuratSub2$label &lt;- factor(seuratSub2$label)
Idents(seuratSub2) &lt;- seuratSub2$TcellGrp
overallDE_ThighLow &lt;- FindAllMarkers(object = seuratSub2, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE_ThighLow &lt;- overallDE_ThighLow %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub2$TcellGrp)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE_ThighLow %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE_ThighLow, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_TcellHigh_vs_TcellLow.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_TcellHigh_vs_TcellLow&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
</div>
<div id="t-cell-high-vs-t-cell-int" class="section level3">
<h3>T cell high vs T cell int</h3>
<pre class="r"><code>seuratSub2 &lt;- subset(seuratSub, TcellGrp %in% c(&quot;TcellHigh&quot;, &quot;TcellInt&quot;))
seuratSub2$label &lt;- factor(seuratSub2$label)
Idents(seuratSub2) &lt;- seuratSub2$TcellGrp
overallDE_ThighInt &lt;- FindAllMarkers(object = seuratSub2, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE_ThighInt &lt;- overallDE_ThighInt %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub2$TcellGrp)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE_ThighInt %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE_ThighInt, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_TcellHigh_vs_TcellInt.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_TcellHigh_vs_TcellInt&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
</div>
<div id="t-cell-int-vs-t-cell-low" class="section level3">
<h3>T cell int vs T cell low</h3>
<pre class="r"><code>seuratSub2 &lt;- subset(seuratSub, TcellGrp %in% c(&quot;TcellLow&quot;, &quot;TcellInt&quot;))
seuratSub2$label &lt;- factor(seuratSub2$label)
Idents(seuratSub2) &lt;- seuratSub2$TcellGrp
overallDE_TLowInt &lt;- FindAllMarkers(object = seuratSub2, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE_TLowInt &lt;- overallDE_TLowInt %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub2$TcellGrp)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE_TLowInt %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE_TLowInt, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_TcellLow_vs_TcellInt.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_TcellLow_vs_TcellInt&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
</div>
<div id="t-cell-high-vs-t-cell-low-plus-int" class="section level3">
<h3>T cell high vs T cell low plus int</h3>
<pre class="r"><code>seuratSub2 &lt;- seuratSub
seuratSub2$TcellGrp2 &lt;- seuratSub2$TcellGrp
seuratSub2$TcellGrp2[which(seuratSub2$TcellGrp %in% c(&quot;TcellLow&quot;, &quot;TcellInt&quot;))] &lt;- &quot;TcellLowInt&quot;
Idents(seuratSub2) &lt;- seuratSub2$TcellGrp2
overallDE_T2 &lt;- FindAllMarkers(object = seuratSub2, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE_T2 &lt;- overallDE_T2 %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub2$TcellGrp2)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE_T2 %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE_T2, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_TcellHigh_vs_TcellLowPlusInt.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;downsampOverallDE_TcellHigh_vs_TcellLowPlusInt&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
</div>
</div>
<div id="cw-de-genes-all-grps" class="section level2">
<h2>cw DE genes all grps</h2>
<pre class="r"><code>## no downsampling..
Idents(seurat) &lt;- seurat$TcellGrp
grpVec &lt;- unique(seurat$label)

clustDE &lt;- lapply(grpVec, function(grp){
  grpSub &lt;- unique(seurat$label)[which(
    unique(seurat$label)==grp)]
  seuratSub2 &lt;- subset(seurat, label == grpSub)
  DEgenes &lt;-FindAllMarkers(seuratSub2, only.pos=T, logfc.threshold = 0.25,
                           min.pct = 0.1)
  if(nrow(DEgenes)&gt;1){
    DEgenes &lt;- DEgenes %&gt;% filter(p_val_adj&lt;0.01) %&gt;%
      mutate(group=paste0(grp, &quot;_&quot;, cluster)) %&gt;% 
      mutate(geneID=gene) %&gt;% mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) %&gt;% 
      filter(nchar(gene)&gt;1)
    }
})

names(clustDE) &lt;- grpVec

clustDE_Dat &lt;- data.frame(do.call(&quot;rbind&quot;, clustDE))
write.table(clustDE_Dat, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;CWDE_allGrps.txt&quot;))</code></pre>
<div id="avg-heatmap-top-cwde-genes" class="section level3 tabset">
<h3 class="tabset">avg Heatmap top cwDE genes</h3>
<div id="endoec" class="section level4">
<h4>EndoEC</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE EndoEC-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="tcell" class="section level4">
<h4>Tcell</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE Tcell-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="resmacrophage" class="section level4">
<h4>resMacrophage</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE resMacrophage-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="fibroblast" class="section level4">
<h4>Fibroblast</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE Fibroblast-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="infmacrophage" class="section level4">
<h4>infMacrophage</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE infMacrophage-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="perivascular" class="section level4">
<h4>Perivascular</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE Perivascular-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cardiomyocyte" class="section level4">
<h4>Cardiomyocyte</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE Cardiomyocyte-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="endothelial" class="section level4">
<h4>Endothelial</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE Endothelial-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="adipocytes" class="section level4">
<h4>Adipocytes</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE Adipocytes-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="neuralcells" class="section level4">
<h4>NeuralCells</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE NeuralCells-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="smc" class="section level4">
<h4>SMC</h4>
<p><img src="figure/DEgenesTcellgrps.Rmd/avgHeat cwDE SMC-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="cw-de-genes-t-cell-high-vs-t-cell-low" class="section level2">
<h2>cw DE genes T cell high vs T cell low</h2>
<pre class="r"><code>## no downsampling..
seuratSub3 &lt;- subset(seurat, TcellGrp %in% c(&quot;TcellLow&quot;, &quot;TcellHigh&quot;))

Idents(seuratSub3) &lt;- seuratSub3$TcellGrp
grpVec &lt;- unique(seuratSub3$label)

clustDE &lt;- lapply(grpVec, function(grp){
  grpSub &lt;- unique(seuratSub3$label)[which(
    unique(seuratSub3$label)==grp)]
  seuratSub2 &lt;- subset(seuratSub3, label == grpSub)
  DEgenes &lt;-FindAllMarkers(seuratSub2, only.pos=T, logfc.threshold = 0.25,
                           min.pct = 0.1)
  if(nrow(DEgenes)&gt;1){
    DEgenes &lt;- DEgenes %&gt;% filter(p_val_adj&lt;0.01) %&gt;%
      mutate(group=paste0(grp, &quot;_&quot;, cluster)) %&gt;% 
      mutate(geneID=gene) %&gt;% mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) %&gt;% 
      filter(nchar(gene)&gt;1)
    }
})

names(clustDE) &lt;- grpVec

clustDE_Dat &lt;- data.frame(do.call(&quot;rbind&quot;, clustDE))
write.table(clustDE_Dat, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;CWDE_TcellHigh_vs_TcellLow.txt&quot;))</code></pre>
</div>
<div id="fibroblast-de-genes" class="section level2 tabset">
<h2 class="tabset">Fibroblast DE genes</h2>
<pre class="r"><code>table(seurat$label, seurat$TcellGrp)</code></pre>
<pre><code>               
                TcellHigh TcellInt TcellLow
  Adipocytes           63       11        5
  Cardiomyocyte      1101     1260      871
  EndoEC              634     1119      634
  Endothelial        1721     3698     2043
  Fibroblast         3165     4849     3162
  infMacrophage      2078      397      174
  LEC                  21       45       14
  NeuralCells         169      220      147
  Perivascular       1612     3145     2066
  resMacrophage      1493      724      430
  SMC                 484      682      509
  Tcell              4409      728      231</code></pre>
<pre class="r"><code>## all grps &gt; 3000 cells --&gt; no need to downsample

Idents(seurat) &lt;- seurat$label_plus_TcellGrp
seuratSub &lt;- subset(seurat, label==&quot;Fibroblast&quot;)</code></pre>
<div id="fibroblast-de-genes-all-grps" class="section level3">
<h3>Fibroblast DE genes all grps</h3>
<pre class="r"><code>Idents(seuratSub) &lt;- seuratSub$TcellGrp
overallDE &lt;- FindAllMarkers(object = seuratSub, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE &lt;- overallDE %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub$TcellGrp)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_allGrps.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_allGrps&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
</div>
<div id="t-cell-high-vs-t-cell-low-1" class="section level3">
<h3>T cell high vs T cell low</h3>
<pre class="r"><code>seuratSub2 &lt;- subset(seuratSub, TcellGrp %in% c(&quot;TcellHigh&quot;, &quot;TcellLow&quot;))
seuratSub2$label &lt;- factor(seuratSub2$label)
Idents(seuratSub2) &lt;- seuratSub2$TcellGrp
overallDE_ThighLow &lt;- FindAllMarkers(object = seuratSub2, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE_ThighLow &lt;- overallDE_ThighLow %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub2$TcellGrp)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE_ThighLow %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE_ThighLow, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_TcellHigh_vs_TcellLow.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_TcellHigh_vs_TcellLow&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
<div id="vis-enrichment-analysis-go-terms-barplots"
class="section level4">
<h4>vis enrichment analysis GO terms barplots</h4>
<pre class="r"><code>selGO &lt;- read_tsv(paste0(basedir,&quot;/data/GSEA/selGO.txt&quot;)) %&gt;% 
  mutate(ID_plus_grp=paste0(GOterm, &quot;_&quot;, Grp))
GOconsDat &lt;- GOconsDat %&gt;% mutate(ID_plus_grp=paste0(ID, &quot;_&quot;, subset)) %&gt;% 
  filter(ID_plus_grp %in% selGO$ID_plus_grp) 

grpVec &lt;- unique(selGO$Grp)
lapply(grpVec, function(grp){
  selGODat &lt;- GOconsDat %&gt;% filter(subset == grp)
  selGODat &lt;- selGODat %&gt;% mutate(qscore=-log(p.adjust, base=10)) 
  p &lt;- ggbarplot(selGODat, x = &quot;Description&quot;, y = &quot;qscore&quot;,
          fill = &quot;subset&quot;,               
          color = &quot;subset&quot;,            
          palette = colTgrp,            
          sort.val = &quot;asc&quot;,           
          sort.by.groups = TRUE      
          #x.text.angle = 90           
          ) + 
  rotate()
p
})</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/barplot-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/barplot-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="vis-enrichment-analysis-go-terms-network"
class="section level4">
<h4>vis enrichment analysis GO terms network</h4>
<pre class="r"><code>grpVec &lt;- unique(selGO$Grp)
colTgrpGen &lt;- c(&quot;#c96778&quot;, &quot;#8391ba&quot;)
names(colTgrpGen) &lt;- grpVec

lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE_ThighLow %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    filter(ID %in% selGO$GOterm)
  egoSS@result &lt;- egoSSres
  colNode &lt;- colTgrp[cl]
  colGene &lt;- colTgrpGen[cl]
  p &lt;- cnetplot(egoSS, showCategory=10, node_label=&quot;all&quot;, 
        color_category=colNode, 
        color_gene=colGene,
        shadowtext=&quot;none&quot;) 
  p
})</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/network-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/network-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="vis-enrichment-analysis-go-terms-combined-network"
class="section level4">
<h4>vis enrichment analysis GO terms combined network</h4>
<pre class="r"><code>grpVec &lt;- unique(selGO$Grp)

orgMarkerList &lt;- lapply(grpVec, function(grp){
  selMarker &lt;- overallDE_ThighLow %&gt;% filter(cluster == grp) %&gt;%
    mutate(EnsID=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  out &lt;- selMarker$EnsID
})
names(orgMarkerList) &lt;- grpVec

ck &lt;- compareCluster(geneCluster = orgMarkerList,
                     fun = enrichGO,
                     OrgDb = org.Hs.eg.db,
                     keyType = &#39;ENSEMBL&#39;,
                     ont = &quot;BP&quot;,
                     pvalueCutoff  = 0.1,
                     qvalueCutoff  = 0.1
)
ck &lt;- setReadable(ck, OrgDb = org.Hs.eg.db)

cnetplot(ck, showCategory = 10, shadowtext=&quot;none&quot;) +
  scale_fill_manual(values = colTgrp)</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/combined%20network-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="vis-enrichment-analysis-go-terms-heatmap"
class="section level4">
<h4>vis enrichment analysis GO terms heatmap</h4>
<pre class="r"><code>logFCdat &lt;- overallDE_ThighLow %&gt;%
  mutate(log2FC=ifelse(cluster==&quot;TcellLow&quot;, -1*avg_log2FC, avg_log2FC)) %&gt;% 
  mutate(geneID=gsub(&quot;.*\\.&quot;, &quot;&quot;, gene)) %&gt;% 
  dplyr::select(log2FC, geneID)

IDtoDesc &lt;- GOconsDat %&gt;% dplyr::select(ID, Description)
geneIDList &lt;- lapply(GOconsDat$ID, function(id){
  selGenes &lt;- data.frame(strsplit(GOconsDat$geneID[which(GOconsDat$ID == id)],
                               split = &quot;/&quot;)) 
  colnames(selGenes) &lt;- &quot;geneID&quot;
  selGenes &lt;- selGenes %&gt;% mutate(ID=id)
})
geneIDdat &lt;- do.call(&quot;rbind&quot;, geneIDList) %&gt;%
  left_join(., IDtoDesc, by=&quot;ID&quot;) %&gt;% 
  left_join(., logFCdat, by=&quot;geneID&quot;) %&gt;% 
  dplyr::select(-ID) %&gt;% 
  spread(., Description, log2FC)

geneIDmat &lt;- t(as.matrix(geneIDdat[,-1]))
colnames(geneIDmat) &lt;- geneIDdat$geneID
ordDat &lt;- do.call(&quot;rbind&quot;, geneIDList) %&gt;%
  left_join(., IDtoDesc, by=&quot;ID&quot;)
geneIDmat &lt;- geneIDmat[unique(ordDat$Description),ordDat$geneID]

pheatmap(geneIDmat, scale=&quot;none&quot; ,treeheight_row = 0, cluster_rows = F, 
         cluster_cols = F, na_col = &quot;#FFFFFF&quot;,
         color = colorRampPalette(c(&quot;#005083&quot;, &quot;#0064ab&quot;,&quot;#6876a4&quot;,&quot;#9f8a89&quot;,
                                    &quot;#c99f6e&quot;, &quot;#f0b64d&quot;,&quot;#ffd28f&quot;))(50),
         cellwidth=15, cellheight=10,
         main = &quot;log2FC T cell high vs T cell low&quot;)</code></pre>
<p><img src="figure/DEgenesTcellgrps.Rmd/heatmap%20GO%20terms-1.png" width="3360" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="t-cell-high-vs-t-cell-int-1" class="section level3">
<h3>T cell high vs T cell int</h3>
<pre class="r"><code>seuratSub2 &lt;- subset(seuratSub, TcellGrp %in% c(&quot;TcellHigh&quot;, &quot;TcellInt&quot;))
seuratSub2$label &lt;- factor(seuratSub2$label)
Idents(seuratSub2) &lt;- seuratSub2$TcellGrp
overallDE_ThighInt &lt;- FindAllMarkers(object = seuratSub2, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE_ThighInt &lt;- overallDE_ThighInt %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub2$TcellGrp)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE_ThighInt %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE_ThighInt, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_TcellHigh_vs_TcellInt.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_TcellHigh_vs_TcellInt&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
</div>
<div id="t-cell-int-vs-t-cell-low-1" class="section level3">
<h3>T cell int vs T cell low</h3>
<pre class="r"><code>seuratSub2 &lt;- subset(seuratSub, TcellGrp %in% c(&quot;TcellLow&quot;, &quot;TcellInt&quot;))
seuratSub2$label &lt;- factor(seuratSub2$label)
Idents(seuratSub2) &lt;- seuratSub2$TcellGrp
overallDE_TLowInt &lt;- FindAllMarkers(object = seuratSub2, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE_TLowInt &lt;- overallDE_TLowInt %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub2$TcellGrp)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE_TLowInt %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE_TLowInt, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_TcellLow_vs_TcellInt.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_TcellLow_vs_TcellInt&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
</div>
<div id="t-cell-high-vs-t-cell-low-plus-int-1" class="section level3">
<h3>T cell high vs T cell low plus int</h3>
<pre class="r"><code>seuratSub2 &lt;- seuratSub
seuratSub2$TcellGrp2 &lt;- seuratSub2$TcellGrp
seuratSub2$TcellGrp2[which(seuratSub2$TcellGrp %in% c(&quot;TcellLow&quot;, &quot;TcellInt&quot;))] &lt;- &quot;TcellLowInt&quot;
Idents(seuratSub2) &lt;- seuratSub2$TcellGrp2
overallDE_T2 &lt;- FindAllMarkers(object = seuratSub2, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.05,
                                     logfc.threshold = 0.1,
                                     test.use = &quot;wilcox&quot;)
overallDE_T2 &lt;- overallDE_T2 %&gt;% filter(p_val_adj &lt; 0.05)

grpVec &lt;- unique(seuratSub2$TcellGrp2)
GOcons &lt;- lapply(grpVec, function(cl){
  DEgenesSub &lt;- overallDE_T2 %&gt;% 
    filter(cluster == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, gene)) %&gt;% 
    slice_min(., p_val_adj, n=250)
  egoSS &lt;- enrichGO(gene      = unique(DEgenesSub$ENS),
                OrgDb         = org.Hs.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Hs.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- grpVec

## write list with DE genes and GO terms
write.table(overallDE_T2, quote=F, row.names = F,col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_TcellHigh_vs_TcellLowPlusInt.txt&quot;))
GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons)
write.table(GOconsDat, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir,&quot;/data/GSEA/humanHeartsPlusGraz_intPatients_&quot;, 
                              &quot;FibroblastDE_TcellHigh_vs_TcellLowPlusInt&quot;,
                              &quot;_GO.txt&quot;))</code></pre>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] viridis_0.6.2               viridisLite_0.4.1          
 [3] enrichplot_1.16.2           DOSE_3.22.1                
 [5] org.Hs.eg.db_3.15.0         AnnotationDbi_1.58.0       
 [7] clusterProfiler_4.4.4       ggpubr_0.5.0               
 [9] pheatmap_1.0.12             ggsci_2.9                  
[11] runSeurat3_0.1.0            here_1.0.1                 
[13] magrittr_2.0.3              SeuratObject_4.1.3         
[15] Seurat_4.3.0                forcats_0.5.2              
[17] stringr_1.5.0               dplyr_1.0.10               
[19] purrr_0.3.5                 readr_2.1.3                
[21] tidyr_1.2.1                 tibble_3.1.8               
[23] ggplot2_3.4.0               tidyverse_1.3.2            
[25] SingleCellExperiment_1.18.1 SummarizedExperiment_1.26.1
[27] Biobase_2.56.0              GenomicRanges_1.48.0       
[29] GenomeInfoDb_1.32.4         IRanges_2.30.1             
[31] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[33] MatrixGenerics_1.8.1        matrixStats_0.63.0         

loaded via a namespace (and not attached):
  [1] scattermore_0.8        bit64_4.0.5            knitr_1.41            
  [4] irlba_2.3.5.1          DelayedArray_0.22.0    data.table_1.14.6     
  [7] KEGGREST_1.36.3        RCurl_1.98-1.9         generics_0.1.3        
 [10] cowplot_1.1.1          RSQLite_2.2.19         shadowtext_0.1.2      
 [13] RANN_2.6.1             future_1.29.0          bit_4.0.5             
 [16] tzdb_0.3.0             spatstat.data_3.0-0    xml2_1.3.3            
 [19] lubridate_1.9.0        httpuv_1.6.6           assertthat_0.2.1      
 [22] gargle_1.2.1           xfun_0.35              hms_1.1.2             
 [25] jquerylib_0.1.4        evaluate_0.18          promises_1.2.0.1      
 [28] fansi_1.0.3            dbplyr_2.2.1           readxl_1.4.1          
 [31] igraph_1.3.5           DBI_1.1.3              htmlwidgets_1.5.4     
 [34] spatstat.geom_3.0-3    googledrive_2.0.0      ellipsis_0.3.2        
 [37] backports_1.4.1        deldir_1.0-6           vctrs_0.5.1           
 [40] ROCR_1.0-11            abind_1.4-5            cachem_1.0.6          
 [43] withr_2.5.0            ggforce_0.4.1          progressr_0.11.0      
 [46] vroom_1.6.0            sctransform_0.3.5      treeio_1.20.2         
 [49] goftest_1.2-3          cluster_2.1.4          ape_5.6-2             
 [52] lazyeval_0.2.2         crayon_1.5.2           spatstat.explore_3.0-5
 [55] labeling_0.4.2         pkgconfig_2.0.3        tweenr_2.0.2          
 [58] nlme_3.1-160           rlang_1.0.6            globals_0.16.2        
 [61] lifecycle_1.0.3        miniUI_0.1.1.1         downloader_0.4        
 [64] modelr_0.1.10          cellranger_1.1.0       rprojroot_2.0.3       
 [67] polyclip_1.10-4        lmtest_0.9-40          Matrix_1.5-3          
 [70] aplot_0.1.9            carData_3.0-5          zoo_1.8-11            
 [73] reprex_2.0.2           ggridges_0.5.4         googlesheets4_1.0.1   
 [76] png_0.1-8              bitops_1.0-7           KernSmooth_2.23-20    
 [79] Biostrings_2.64.1      blob_1.2.3             workflowr_1.7.0       
 [82] qvalue_2.28.0          parallelly_1.32.1      spatstat.random_3.0-1 
 [85] rstatix_0.7.1          gridGraphics_0.5-1     ggsignif_0.6.4        
 [88] scales_1.2.1           memoise_2.0.1          plyr_1.8.8            
 [91] ica_1.0-3              zlibbioc_1.42.0        scatterpie_0.1.8      
 [94] compiler_4.2.1         RColorBrewer_1.1-3     fitdistrplus_1.1-8    
 [97] cli_3.4.1              XVector_0.36.0         listenv_0.8.0         
[100] patchwork_1.1.2        pbapply_1.6-0          MASS_7.3-58.1         
[103] tidyselect_1.2.0       stringi_1.7.8          highr_0.9             
[106] yaml_2.3.6             GOSemSim_2.22.0        ggrepel_0.9.2         
[109] grid_4.2.1             sass_0.4.4             fastmatch_1.1-3       
[112] tools_4.2.1            timechange_0.1.1       future.apply_1.10.0   
[115] parallel_4.2.1         rstudioapi_0.14        git2r_0.30.1          
[118] gridExtra_2.3          farver_2.1.1           Rtsne_0.16            
[121] ggraph_2.1.0           digest_0.6.30          shiny_1.7.3           
[124] Rcpp_1.0.9             car_3.1-1              broom_1.0.1           
[127] later_1.3.0            RcppAnnoy_0.0.20       httr_1.4.4            
[130] colorspace_2.0-3       rvest_1.0.3            fs_1.5.2              
[133] tensor_1.5             reticulate_1.26        splines_4.2.1         
[136] uwot_0.1.14            yulab.utils_0.0.5      tidytree_0.4.1        
[139] spatstat.utils_3.0-1   graphlayouts_0.8.4     sp_1.5-1              
[142] ggplotify_0.1.0        plotly_4.10.1          xtable_1.8-4          
[145] jsonlite_1.8.3         ggtree_3.4.4           tidygraph_1.2.2       
[148] ggfun_0.0.9            R6_2.5.1               pillar_1.8.1          
[151] htmltools_0.5.3        mime_0.12              glue_1.6.2            
[154] fastmap_1.1.0          BiocParallel_1.30.4    codetools_0.2-18      
[157] fgsea_1.22.0           utf8_1.2.2             lattice_0.20-45       
[160] bslib_0.4.1            spatstat.sparse_3.0-0  leiden_0.4.3          
[163] GO.db_3.15.0           limma_3.52.4           survival_3.4-0        
[166] rmarkdown_2.18         munsell_0.5.0          DO.db_2.9             
[169] GenomeInfoDbData_1.2.8 haven_2.5.1            reshape2_1.4.4        
[172] gtable_0.3.1          </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Thu Dec  8 08:57:34 2022&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] viridis_0.6.2               viridisLite_0.4.1          
 [3] enrichplot_1.16.2           DOSE_3.22.1                
 [5] org.Hs.eg.db_3.15.0         AnnotationDbi_1.58.0       
 [7] clusterProfiler_4.4.4       ggpubr_0.5.0               
 [9] pheatmap_1.0.12             ggsci_2.9                  
[11] runSeurat3_0.1.0            here_1.0.1                 
[13] magrittr_2.0.3              SeuratObject_4.1.3         
[15] Seurat_4.3.0                forcats_0.5.2              
[17] stringr_1.5.0               dplyr_1.0.10               
[19] purrr_0.3.5                 readr_2.1.3                
[21] tidyr_1.2.1                 tibble_3.1.8               
[23] ggplot2_3.4.0               tidyverse_1.3.2            
[25] SingleCellExperiment_1.18.1 SummarizedExperiment_1.26.1
[27] Biobase_2.56.0              GenomicRanges_1.48.0       
[29] GenomeInfoDb_1.32.4         IRanges_2.30.1             
[31] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[33] MatrixGenerics_1.8.1        matrixStats_0.63.0         

loaded via a namespace (and not attached):
  [1] scattermore_0.8        bit64_4.0.5            knitr_1.41            
  [4] irlba_2.3.5.1          DelayedArray_0.22.0    data.table_1.14.6     
  [7] KEGGREST_1.36.3        RCurl_1.98-1.9         generics_0.1.3        
 [10] cowplot_1.1.1          RSQLite_2.2.19         shadowtext_0.1.2      
 [13] RANN_2.6.1             future_1.29.0          bit_4.0.5             
 [16] tzdb_0.3.0             spatstat.data_3.0-0    xml2_1.3.3            
 [19] lubridate_1.9.0        httpuv_1.6.6           assertthat_0.2.1      
 [22] gargle_1.2.1           xfun_0.35              hms_1.1.2             
 [25] jquerylib_0.1.4        evaluate_0.18          promises_1.2.0.1      
 [28] fansi_1.0.3            dbplyr_2.2.1           readxl_1.4.1          
 [31] igraph_1.3.5           DBI_1.1.3              htmlwidgets_1.5.4     
 [34] spatstat.geom_3.0-3    googledrive_2.0.0      ellipsis_0.3.2        
 [37] backports_1.4.1        deldir_1.0-6           vctrs_0.5.1           
 [40] ROCR_1.0-11            abind_1.4-5            cachem_1.0.6          
 [43] withr_2.5.0            ggforce_0.4.1          progressr_0.11.0      
 [46] vroom_1.6.0            sctransform_0.3.5      treeio_1.20.2         
 [49] goftest_1.2-3          cluster_2.1.4          ape_5.6-2             
 [52] lazyeval_0.2.2         crayon_1.5.2           spatstat.explore_3.0-5
 [55] labeling_0.4.2         pkgconfig_2.0.3        tweenr_2.0.2          
 [58] nlme_3.1-160           rlang_1.0.6            globals_0.16.2        
 [61] lifecycle_1.0.3        miniUI_0.1.1.1         downloader_0.4        
 [64] modelr_0.1.10          cellranger_1.1.0       rprojroot_2.0.3       
 [67] polyclip_1.10-4        lmtest_0.9-40          Matrix_1.5-3          
 [70] aplot_0.1.9            carData_3.0-5          zoo_1.8-11            
 [73] reprex_2.0.2           ggridges_0.5.4         googlesheets4_1.0.1   
 [76] png_0.1-8              bitops_1.0-7           KernSmooth_2.23-20    
 [79] Biostrings_2.64.1      blob_1.2.3             workflowr_1.7.0       
 [82] qvalue_2.28.0          parallelly_1.32.1      spatstat.random_3.0-1 
 [85] rstatix_0.7.1          gridGraphics_0.5-1     ggsignif_0.6.4        
 [88] scales_1.2.1           memoise_2.0.1          plyr_1.8.8            
 [91] ica_1.0-3              zlibbioc_1.42.0        scatterpie_0.1.8      
 [94] compiler_4.2.1         RColorBrewer_1.1-3     fitdistrplus_1.1-8    
 [97] cli_3.4.1              XVector_0.36.0         listenv_0.8.0         
[100] patchwork_1.1.2        pbapply_1.6-0          MASS_7.3-58.1         
[103] tidyselect_1.2.0       stringi_1.7.8          highr_0.9             
[106] yaml_2.3.6             GOSemSim_2.22.0        ggrepel_0.9.2         
[109] grid_4.2.1             sass_0.4.4             fastmatch_1.1-3       
[112] tools_4.2.1            timechange_0.1.1       future.apply_1.10.0   
[115] parallel_4.2.1         rstudioapi_0.14        git2r_0.30.1          
[118] gridExtra_2.3          farver_2.1.1           Rtsne_0.16            
[121] ggraph_2.1.0           digest_0.6.30          shiny_1.7.3           
[124] Rcpp_1.0.9             car_3.1-1              broom_1.0.1           
[127] later_1.3.0            RcppAnnoy_0.0.20       httr_1.4.4            
[130] colorspace_2.0-3       rvest_1.0.3            fs_1.5.2              
[133] tensor_1.5             reticulate_1.26        splines_4.2.1         
[136] uwot_0.1.14            yulab.utils_0.0.5      tidytree_0.4.1        
[139] spatstat.utils_3.0-1   graphlayouts_0.8.4     sp_1.5-1              
[142] ggplotify_0.1.0        plotly_4.10.1          xtable_1.8-4          
[145] jsonlite_1.8.3         ggtree_3.4.4           tidygraph_1.2.2       
[148] ggfun_0.0.9            R6_2.5.1               pillar_1.8.1          
[151] htmltools_0.5.3        mime_0.12              glue_1.6.2            
[154] fastmap_1.1.0          BiocParallel_1.30.4    codetools_0.2-18      
[157] fgsea_1.22.0           utf8_1.2.2             lattice_0.20-45       
[160] bslib_0.4.1            spatstat.sparse_3.0-0  leiden_0.4.3          
[163] GO.db_3.15.0           limma_3.52.4           survival_3.4-0        
[166] rmarkdown_2.18         munsell_0.5.0          DO.db_2.9             
[169] GenomeInfoDbData_1.2.8 haven_2.5.1            reshape2_1.4.4        
[172] gtable_0.3.1          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
