<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2020-05-14" />

<title>Merge samples</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">humanCardiacFibroblasts</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Merge samples</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">14 May 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>humanCardiacFibroblasts/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210903code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210903)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210903code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210903)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststree70e878c1e0500e2587aa4df8dca31023d0734855targetblank70e878ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/70e878c1e0500e2587aa4df8dca31023d0734855" target="_blank">70e878c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommluetgehumanCardiacFibroblaststree70e878c1e0500e2587aa4df8dca31023d0734855targetblank70e878ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mluetge/humanCardiacFibroblasts/tree/70e878c1e0500e2587aa4df8dca31023d0734855" target="_blank">70e878c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/humanFibroblast/

Unstaged changes:
    Modified:   analysis/Vis_plus_analyse_NormlaHFMyo.Rmd
    Modified:   analysis/Vis_plus_analyse_NormlaHFMyo_woECMO4.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Vis_plus_analyse_NormlaHFMyo.Rmd</code>) and HTML (<code>docs/Vis_plus_analyse_NormlaHFMyo.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/analysis/Vis_plus_analyse_NormlaHFMyo.Rmd" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
<td>
update grand application plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/humanCardiacFibroblasts/70e878c1e0500e2587aa4df8dca31023d0734855/docs/Vis_plus_analyse_NormlaHFMyo.html" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
<td>
update grand application plots
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(SingleCellExperiment)
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(ggsci)
  library(ggpubr)
  library(pheatmap)
  library(viridis)
  library(sctransform)
})</code></pre>
</div>
<div id="sign-plot-funct" class="section level2">
<h2>sign plot funct</h2>
<pre class="r"><code>## adapted from CellMixS
visGroup_adapt &lt;- function (sce,group,dim_red = &quot;TSNE&quot;,col_group=pal_nejm()(8)) 
{
    if (!is(sce, &quot;SingleCellExperiment&quot;)) {
        stop(&quot;Error:&#39;sce&#39; must be a &#39;SingleCellExperiment&#39; object.&quot;)
    }
    if (!group %in% names(colData(sce))) {
        stop(&quot;Error: &#39;group&#39; variable must be in &#39;colData(sce)&#39;&quot;)
    }
    cell_names &lt;- colnames(sce)
    if (!dim_red %in% &quot;TSNE&quot;) {
        if (!dim_red %in% reducedDimNames(sce)) {
            stop(&quot;Please provide a dim_red method listed in reducedDims of sce&quot;)
        }
        red_dim &lt;- as.data.frame(reducedDim(sce, dim_red))
    }
    else {
        if (!&quot;TSNE&quot; %in% reducedDimNames(sce)) {
            if (&quot;logcounts&quot; %in% names(assays(sce))) {
                sce &lt;- runTSNE(sce)
            }
            else {
                sce &lt;- runTSNE(sce, exprs_values = &quot;counts&quot;)
            }
        }
        red_dim &lt;- as.data.frame(reducedDim(sce, &quot;TSNE&quot;))
    }
    colnames(red_dim) &lt;- c(&quot;red_dim1&quot;, &quot;red_dim2&quot;)
    df &lt;- data.frame(sample_id = cell_names, group_var = colData(sce)[, 
        group], red_Dim1 = red_dim$red_dim1, red_Dim2 = red_dim$red_dim2)
    t &lt;- ggplot(df, aes_string(x = &quot;red_Dim1&quot;, y = &quot;red_Dim2&quot;)) + 
        xlab(paste0(dim_red, &quot;_1&quot;)) + ylab(paste0(dim_red, &quot;_2&quot;)) + 
        theme_void() + theme(aspect.ratio = 1,
                             panel.grid.minor = element_blank(), 
        panel.grid.major = element_line(color = &quot;grey&quot;, size = 0.3))
    t_group &lt;- t + geom_point(size = 1.5, alpha = 0.8,
                              aes_string(color = &quot;group_var&quot;)) + 
        guides(color = guide_legend(override.aes = list(size = 1), 
            title = group)) + ggtitle(group)
    if (is.numeric(df$group_var)) {
        t_group &lt;- t_group + scale_color_viridis(option = &quot;D&quot;)
    }
    else {
        t_group &lt;- t_group + scale_color_manual(values = col_group)
    }
    t_group
}</code></pre>
</div>
<div id="set-dir" class="section level2">
<h2>set dir</h2>
<pre class="r"><code>basedir &lt;- here()
seurat &lt;- readRDS(file = paste0(basedir, 
                              &quot;/data/humanHearts_merged_seurat.rds&quot;))
seurat$ID[which(seurat$ID == &quot;ID23_25&quot;)] &lt;- &quot;ID2325&quot;

## subset on sel patients
selPat &lt;- c(&quot;ID2325&quot;, &quot;ID28&quot;, &quot;ID30&quot;, &quot;ID31&quot;, &quot;ID21&quot;, &quot;ID26&quot;, &quot;ECMO4&quot;, &quot;ID29&quot;)
seurat &lt;- subset(seurat, ID %in% selPat)
#seurat &lt;- rerunSeurat3(seurat)

seurat$grp &lt;- &quot;normal&quot;
seurat$grp[which(seurat$ID %in% c(&quot;ID21&quot;, &quot;ID26&quot;))] &lt;- &quot;HF&quot;
seurat$grp[which(seurat$ID %in% c(&quot;ID30&quot;, &quot;ID31&quot;, &quot;ID29&quot;))] &lt;- &quot;Myocarditis&quot;

#Idents(seurat) &lt;- seurat$RNA_snn_res.0.25

## integrate data across patients
Idents(seurat) &lt;- seurat$ID

seurat.list &lt;- SplitObject(object = seurat, split.by = &quot;ID&quot;)
for (i in 1:length(x = seurat.list)) {
    seurat.list[[i]] &lt;- NormalizeData(object = seurat.list[[i]],
                                      verbose = FALSE)
    seurat.list[[i]] &lt;- FindVariableFeatures(object = seurat.list[[i]], 
        selection.method = &quot;vst&quot;, nfeatures = 2000, verbose = FALSE)
}

seurat.anchors &lt;- FindIntegrationAnchors(object.list = seurat.list, dims = 1:15)
seurat.int &lt;- IntegrateData(anchorset = seurat.anchors, dims = 1:15)
DefaultAssay(object = seurat.int) &lt;- &quot;integrated&quot;

# rerun seurat
seurat.int &lt;- ScaleData(object = seurat.int, verbose = FALSE,
                        features = rownames(seurat.int))
seurat.int &lt;- RunPCA(object = seurat.int, npcs = 20, verbose = FALSE)
seurat.int &lt;- RunTSNE(object = seurat.int, reduction = &quot;pca&quot;, dims = 1:20)
seurat.int &lt;- RunUMAP(object = seurat.int, reduction = &quot;pca&quot;, dims = 1:20)

seurat.int &lt;- FindNeighbors(object = seurat.int, reduction = &quot;pca&quot;, dims = 1:20)
res &lt;- c(0.6,0.8,0.4,0.25)
for(i in 1:length(res)){
  seurat.int &lt;- FindClusters(object = seurat.int, resolution = res[i],
                             random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11281
Number of edges: 505870

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9194
Number of communities: 17
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11281
Number of edges: 505870

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9017
Number of communities: 19
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11281
Number of edges: 505870

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9402
Number of communities: 14
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11281
Number of edges: 505870

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9587
Number of communities: 12
Elapsed time: 1 seconds</code></pre>
<pre class="r"><code>DefaultAssay(object = seurat.int) &lt;- &quot;RNA&quot;
seurat &lt;- seurat.int
remove(seurat.int)</code></pre>
</div>
<div id="color-vectors" class="section level2">
<h2>color vectors</h2>
<pre class="r"><code>colPal &lt;- pal_igv()(length(levels(seurat)))
colTec &lt;- pal_jama()(length(unique(seurat$technique)))
colSmp &lt;- c(pal_uchicago()(8), pal_npg()(8), pal_aaas()(10))[1:length(unique(seurat$dataset))]
colLoc &lt;- pal_npg()(length(unique(seurat$location)))
colBatch &lt;- c(pal_jco()(10), pal_npg()(10))[1:length(unique(seurat$ID))]
colOrig &lt;- pal_futurama()(length(unique(seurat$origin)))
colIso &lt;- pal_nejm()(length(unique(seurat$isolation)))
colGrp &lt;- c(&quot;#b6bcbb&quot;, &quot;#a32d25&quot;, &quot;#2544a3&quot;)



names(colPal) &lt;- levels(seurat)
names(colTec) &lt;- unique(seurat$technique)
names(colSmp) &lt;- unique(seurat$dataset)
names(colLoc) &lt;- unique(seurat$location)
names(colBatch) &lt;- unique(seurat$ID)
names(colOrig) &lt;- unique(seurat$origin)
names(colIso) &lt;- unique(seurat$isolation)
names(colGrp) &lt;- c(&quot;normal&quot;, &quot;Myocarditis&quot;, &quot;HF&quot;)</code></pre>
</div>
<div id="vis-data" class="section level2 tabset">
<h2 class="tabset">vis data</h2>
<div id="clusters" class="section level3">
<h3>clusters</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cluster-1">
Past versions of “vis cluster-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-cluster-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis cluster-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal,
        pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20cluster-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cluster-2">
Past versions of “vis cluster-2.png”
</button>
</p>
<div id="fig-no-spaces-vis-cluster-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis cluster-2.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="technique" class="section level3">
<h3>technique</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;technique&quot;, cols=colTec)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20tecnique-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-tecnique-1">
Past versions of “vis tecnique-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-tecnique-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis tecnique-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample" class="section level3">
<h3>Sample</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;dataset&quot;, cols=colSmp)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20sample-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-sample-1">
Past versions of “vis sample-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis sample-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="id" class="section level3">
<h3>ID</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;ID&quot;, cols=colBatch)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20ID-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-ID-1">
Past versions of “vis ID-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-ID-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis ID-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;ID&quot;, cols=colBatch,
        pt.size=0.6, shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20ID-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-ID-2">
Past versions of “vis ID-2.png”
</button>
</p>
<div id="fig-no-spaces-vis-ID-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis ID-2.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="origin" class="section level3">
<h3>Origin</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;origin&quot;, cols=colOrig)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20origin-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-origin-1">
Past versions of “vis origin-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-origin-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis origin-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="isolation" class="section level3">
<h3>isolation</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;isolation&quot;, cols=colIso)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20isolation-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-isolation-1">
Past versions of “vis isolation-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-isolation-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis isolation-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="location" class="section level3">
<h3>location</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;location&quot;, cols=colLoc)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20location-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-location-1">
Past versions of “vis location-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-location-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis location-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="grp" class="section level3">
<h3>grp</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;grp&quot;, cols=colGrp)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20grp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-grp-1">
Past versions of “vis grp-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-grp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis grp-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, group.by = &quot;grp&quot;, cols=colGrp,
        pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20grp-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-grp-2">
Past versions of “vis grp-2.png”
</button>
</p>
<div id="fig-no-spaces-vis-grp-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis grp-2.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="grp-without-hf" class="section level3">
<h3>grp without HF</h3>
<pre class="r"><code>seuratSub &lt;- subset(seurat, grp == &quot;HF&quot;, invert=T)


DimPlot(seuratSub, reduction = &quot;umap&quot;, group.by = &quot;grp&quot;, cols=colGrp,
        pt.size=0.6, order = &quot;Myocarditis&quot;)+
  theme_void()</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20grp%20without%20HF-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-grp-without-HF-1">
Past versions of “vis grp without HF-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-grp-without-HF-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis grp without HF-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="grp-without-m" class="section level3">
<h3>grp without M</h3>
<pre class="r"><code>seuratSub2 &lt;- subset(seurat, grp == &quot;Myocarditis&quot;, invert=T)


DimPlot(seuratSub2, reduction = &quot;umap&quot;, group.by = &quot;grp&quot;, cols=colGrp,
        pt.size=0.6, order = &quot;HF&quot;)+
  theme_void()</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis%20grp%20without%20M-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-grp-without-M-1">
Past versions of “vis grp without M-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-grp-without-M-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/vis grp without M-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="de-genes-grp" class="section level2">
<h2>DE genes grp</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$grp
DEgenes &lt;- FindAllMarkers(object = seurat, assay =&quot;RNA&quot;,
                                     only.pos = TRUE, min.pct = 0.25,
                                     logfc.threshold = 0.25,
                                     test.use = &quot;wilcox&quot;)</code></pre>
<div id="top-20-marker-genes-per-grp" class="section level3">
<h3>top 20 marker genes per grp</h3>
<pre class="r"><code>cluster &lt;- levels(seurat)
selGenesAll &lt;- DEgenes %&gt;% group_by(cluster) %&gt;% 
  top_n(-20, p_val_adj) %&gt;% 
  top_n(20, avg_log2FC)
selGenesAll &lt;- selGenesAll %&gt;% mutate(geneIDval=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(nchar(geneIDval)&gt;1)

Idents(seurat) &lt;- seurat$grp
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesAll,
                  colVecIdent = colGrp, 
                  ordVec=levels(seurat),
                  gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                  cr=T, condCol=F)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/top%20DE%20genes%20EMB29-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-top-DE-genes-EMB29-1">
Past versions of “top DE genes EMB29-1.png”
</button>
</p>
<div id="fig-no-spaces-top-DE-genes-EMB29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/top DE genes EMB29-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seurat) &lt;- seurat$ID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesAll,
                  colVecIdent = colBatch, 
                  ordVec=levels(seurat),
                  gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                  cr=T, condCol=F)</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/top%20DE%20genes%20EMB29-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-top-DE-genes-EMB29-2">
Past versions of “top DE genes EMB29-2.png”
</button>
</p>
<div id="fig-no-spaces-top-DE-genes-EMB29-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/top DE genes EMB29-2.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cnt-tab" class="section level3">
<h3>cnt Tab</h3>
<pre class="r"><code>table(seurat$grp)</code></pre>
<pre><code>
         HF Myocarditis      normal 
       1064        2670        7547 </code></pre>
<pre class="r"><code>table(seurat$ID)</code></pre>
<pre><code>
 ECMO4   ID21 ID2325   ID26   ID28   ID29   ID30   ID31 
  3162    826   3131    238   1254   1242    236   1192 </code></pre>
<pre class="r"><code>table(seurat$ID, seurat$grp)</code></pre>
<pre><code>        
           HF Myocarditis normal
  ECMO4     0           0   3162
  ID21    826           0      0
  ID2325    0           0   3131
  ID26    238           0      0
  ID28      0           0   1254
  ID29      0        1242      0
  ID30      0         236      0
  ID31      0        1192      0</code></pre>
</div>
</div>
<div id="signature-cut-1.5" class="section level2">
<h2>signature cut 1.5</h2>
<div id="split-by-grp" class="section level3">
<h3>split by grp</h3>
<pre class="r"><code>signDat &lt;- read_delim(file = paste0(basedir,
                    &quot;/data/SelSignaturesTreat.txt&quot;),
                    delim = &quot;\t&quot;)
genes &lt;- data.frame(geneID=rownames(seurat)) %&gt;% 
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, geneID))
signDat &lt;- signDat %&gt;% left_join(.,genes, by=&quot;gene&quot;)
allSign &lt;- unique(signDat$signature)

DefaultAssay(object = seurat) &lt;- &quot;integrated&quot;
sce2 &lt;- as.SingleCellExperiment(seurat)

DefaultAssay(object = seurat) &lt;- &quot;RNA&quot;
sce &lt;- as.SingleCellExperiment(seurat)
reducedDims(sce) &lt;- list(PCA=reducedDim(sce2, &quot;PCA&quot;),
                         TSNE=reducedDim(sce2, &quot;TSNE&quot;),
                         UMAP=reducedDim(sce2, &quot;UMAP&quot;))

treatGrps &lt;- unique(sce$grp)

cutOff &lt;- 1.5
pal = viridis(100)
sc &lt;- scale_colour_gradientn(colours = pal, limits=c(0, cutOff))

lapply(unique(signDat$signature), function(sign){
  signGenes &lt;- signDat %&gt;% dplyr::filter(signature == sign)
  sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
  cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts)))/nrow(signGenes)
  sceSub$sign &lt;- cntMat
  sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
  sceSub$sign[which(sceSub$sign &lt; 0)] &lt;- 0
  lapply(treatGrps, function(treat){
    sceSubT &lt;- sceSub[, which(sceSub$grp == treat)]
    p &lt;- visGroup_adapt(sceSubT, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature - &#39;, treat)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
    p
  })
})</code></pre>
<pre><code>[[1]]
[[1]][[1]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-1">
Past versions of “signature cut 1.5-1.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[1]][[2]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-2">
Past versions of “signature cut 1.5-2.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-2.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[1]][[3]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-3">
Past versions of “signature cut 1.5-3.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-3.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[2]]
[[2]][[1]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-4">
Past versions of “signature cut 1.5-4.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-4.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]][[2]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-5">
Past versions of “signature cut 1.5-5.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-5.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]][[3]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-6">
Past versions of “signature cut 1.5-6.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-6.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[3]]
[[3]][[1]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-7">
Past versions of “signature cut 1.5-7.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-7.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]][[2]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-8">
Past versions of “signature cut 1.5-8.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-8.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]][[3]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-9">
Past versions of “signature cut 1.5-9.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-9.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[4]]
[[4]][[1]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-10">
Past versions of “signature cut 1.5-10.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-10.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]][[2]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-11">
Past versions of “signature cut 1.5-11.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-11.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]][[3]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-12">
Past versions of “signature cut 1.5-12.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-12.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>

[[5]]
[[5]][[1]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-13">
Past versions of “signature cut 1.5-13.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-13.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]][[2]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-14">
Past versions of “signature cut 1.5-14.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-14.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]][[3]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-15">
Past versions of “signature cut 1.5-15.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5-15.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="across-all" class="section level3">
<h3>across all</h3>
<pre class="r"><code>#sce &lt;- as.SingleCellExperiment(seurat)

cutOff &lt;- 1.5
pal = viridis(100)
sc &lt;- scale_colour_gradientn(colours = pal, limits=c(0, cutOff))

lapply(unique(signDat$signature), function(sign){
  signGenes &lt;- signDat %&gt;% dplyr::filter(signature == sign)
  sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
  cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts)))/nrow(signGenes)
  sceSub$sign &lt;- cntMat
  sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
  sceSub$sign[which(sceSub$sign &lt; 0)] &lt;- 0
    p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature - across all&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
    p

})</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5%20across%20all-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-across-all-1">
Past versions of “signature cut 1.5 across all-1.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-across-all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5 across all-1.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5%20across%20all-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-across-all-2">
Past versions of “signature cut 1.5 across all-2.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-across-all-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5 across all-2.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5%20across%20all-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-across-all-3">
Past versions of “signature cut 1.5 across all-3.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-across-all-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5 across all-3.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5%20across%20all-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-across-all-4">
Past versions of “signature cut 1.5 across all-4.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-across-all-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5 across all-4.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature%20cut%201.5%20across%20all-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-signature-cut-1.5-across-all-5">
Past versions of “signature cut 1.5 across all-5.png”
</button>
</p>
<div id="fig-no-spaces-signature-cut-1.5-across-all-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/humanCardiacFibroblasts/blob/70e878c1e0500e2587aa4df8dca31023d0734855/docs/figure/Vis_plus_analyse_NormlaHFMyo.Rmd/signature cut 1.5 across all-5.png" target="_blank">70e878c</a>
</td>
<td>
mluetge
</td>
<td>
2022-05-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="save-objects" class="section level2">
<h2>save objects</h2>
<pre class="r"><code>Idents(seurat) &lt;- Idents(seurat) &lt;- seurat$integrated_snn_res.0.25
saveRDS(seurat, file = paste0(basedir, 
                              &quot;/data/humanHearts_intAcrossPat_Normal_HF_Myocarditis.rds&quot;))

saveRDS(seuratSub, file = paste0(basedir, 
                              &quot;/data/humanHearts_intAcrossPat_Normal_Myocarditis.rds&quot;))

saveRDS(seuratSub2, file = paste0(basedir, 
                              &quot;/data/humanHearts_intAcrossPat_Normal_HF.rds&quot;))

write.table(DEgenes,
            file=paste0(basedir,
                        &quot;/data/humanHearts_intAcrossPat_NORMALvsHFvsMYO_overallDEGenes.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] sctransform_0.3.3           viridis_0.6.2              
 [3] viridisLite_0.4.0           pheatmap_1.0.12            
 [5] ggpubr_0.4.0                ggsci_2.9                  
 [7] runSeurat3_0.1.0            here_1.0.1                 
 [9] magrittr_2.0.3              sp_1.4-7                   
[11] SeuratObject_4.1.0          Seurat_4.1.1               
[13] forcats_0.5.1               stringr_1.4.0              
[15] dplyr_1.0.9                 purrr_0.3.4                
[17] readr_2.1.2                 tidyr_1.2.0                
[19] tibble_3.1.7                ggplot2_3.3.6              
[21] tidyverse_1.3.1             SingleCellExperiment_1.14.1
[23] SummarizedExperiment_1.22.0 Biobase_2.52.0             
[25] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[27] IRanges_2.26.0              S4Vectors_0.30.2           
[29] BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[31] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             reticulate_1.24        tidyselect_1.1.2      
  [4] htmlwidgets_1.5.4      grid_4.1.0             Rtsne_0.16            
  [7] munsell_0.5.0          codetools_0.2-18       ica_1.0-2             
 [10] future_1.25.0          miniUI_0.1.1.1         withr_2.5.0           
 [13] spatstat.random_2.2-0  colorspace_2.0-3       progressr_0.10.0      
 [16] highr_0.9              knitr_1.39             rstudioapi_0.13       
 [19] ROCR_1.0-11            ggsignif_0.6.3         tensor_1.5            
 [22] listenv_0.8.0          labeling_0.4.2         git2r_0.30.1          
 [25] GenomeInfoDbData_1.2.6 polyclip_1.10-0        bit64_4.0.5           
 [28] farver_2.1.0           rprojroot_2.0.3        parallelly_1.31.1     
 [31] vctrs_0.4.1            generics_0.1.2         xfun_0.30             
 [34] R6_2.5.1               bitops_1.0-7           spatstat.utils_2.3-1  
 [37] DelayedArray_0.18.0    assertthat_0.2.1       vroom_1.5.7           
 [40] promises_1.2.0.1       scales_1.2.0           rgeos_0.5-9           
 [43] gtable_0.3.0           globals_0.15.0         goftest_1.2-3         
 [46] workflowr_1.7.0        rlang_1.0.2            splines_4.1.0         
 [49] rstatix_0.7.0          lazyeval_0.2.2         spatstat.geom_2.4-0   
 [52] broom_0.8.0            yaml_2.3.5             reshape2_1.4.4        
 [55] abind_1.4-5            modelr_0.1.8           backports_1.4.1       
 [58] httpuv_1.6.5           tools_4.1.0            ellipsis_0.3.2        
 [61] spatstat.core_2.4-2    jquerylib_0.1.4        RColorBrewer_1.1-3    
 [64] ggridges_0.5.3         Rcpp_1.0.8.3           plyr_1.8.7            
 [67] zlibbioc_1.38.0        RCurl_1.98-1.6         rpart_4.1.16          
 [70] deldir_1.0-6           pbapply_1.5-0          cowplot_1.1.1         
 [73] zoo_1.8-10             haven_2.5.0            ggrepel_0.9.1         
 [76] cluster_2.1.3          fs_1.5.2               data.table_1.14.2     
 [79] RSpectra_0.16-1        scattermore_0.8        lmtest_0.9-40         
 [82] reprex_2.0.1           RANN_2.6.1             whisker_0.4           
 [85] fitdistrplus_1.1-8     hms_1.1.1              patchwork_1.1.1       
 [88] mime_0.12              evaluate_0.15          xtable_1.8-4          
 [91] readxl_1.4.0           gridExtra_2.3          compiler_4.1.0        
 [94] KernSmooth_2.23-20     crayon_1.5.1           htmltools_0.5.2       
 [97] mgcv_1.8-40            later_1.3.0            tzdb_0.3.0            
[100] lubridate_1.8.0        DBI_1.1.2              dbplyr_2.1.1          
[103] MASS_7.3-57            Matrix_1.4-1           car_3.0-13            
[106] cli_3.3.0              igraph_1.3.1           pkgconfig_2.0.3       
[109] plotly_4.10.0          spatstat.sparse_2.1-1  xml2_1.3.3            
[112] bslib_0.3.1            XVector_0.32.0         rvest_1.0.2           
[115] digest_0.6.29          RcppAnnoy_0.0.19       spatstat.data_2.2-0   
[118] rmarkdown_2.14         cellranger_1.1.0       leiden_0.3.10         
[121] uwot_0.1.11            shiny_1.7.1            lifecycle_1.0.1       
[124] nlme_3.1-157           jsonlite_1.8.0         carData_3.0-5         
[127] limma_3.48.3           fansi_1.0.3            pillar_1.7.0          
[130] lattice_0.20-45        fastmap_1.1.0          httr_1.4.3            
[133] survival_3.3-1         glue_1.6.2             png_0.1-7             
[136] bit_4.0.4              stringi_1.7.6          sass_0.4.1            
[139] irlba_2.3.5            future.apply_1.9.0    </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Mon May  9 15:30:27 2022&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] sctransform_0.3.3           viridis_0.6.2              
 [3] viridisLite_0.4.0           pheatmap_1.0.12            
 [5] ggpubr_0.4.0                ggsci_2.9                  
 [7] runSeurat3_0.1.0            here_1.0.1                 
 [9] magrittr_2.0.3              sp_1.4-7                   
[11] SeuratObject_4.1.0          Seurat_4.1.1               
[13] forcats_0.5.1               stringr_1.4.0              
[15] dplyr_1.0.9                 purrr_0.3.4                
[17] readr_2.1.2                 tidyr_1.2.0                
[19] tibble_3.1.7                ggplot2_3.3.6              
[21] tidyverse_1.3.1             SingleCellExperiment_1.14.1
[23] SummarizedExperiment_1.22.0 Biobase_2.52.0             
[25] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[27] IRanges_2.26.0              S4Vectors_0.30.2           
[29] BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[31] matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2             reticulate_1.24        tidyselect_1.1.2      
  [4] htmlwidgets_1.5.4      grid_4.1.0             Rtsne_0.16            
  [7] munsell_0.5.0          codetools_0.2-18       ica_1.0-2             
 [10] future_1.25.0          miniUI_0.1.1.1         withr_2.5.0           
 [13] spatstat.random_2.2-0  colorspace_2.0-3       progressr_0.10.0      
 [16] highr_0.9              knitr_1.39             rstudioapi_0.13       
 [19] ROCR_1.0-11            ggsignif_0.6.3         tensor_1.5            
 [22] listenv_0.8.0          labeling_0.4.2         git2r_0.30.1          
 [25] GenomeInfoDbData_1.2.6 polyclip_1.10-0        bit64_4.0.5           
 [28] farver_2.1.0           rprojroot_2.0.3        parallelly_1.31.1     
 [31] vctrs_0.4.1            generics_0.1.2         xfun_0.30             
 [34] R6_2.5.1               bitops_1.0-7           spatstat.utils_2.3-1  
 [37] DelayedArray_0.18.0    assertthat_0.2.1       vroom_1.5.7           
 [40] promises_1.2.0.1       scales_1.2.0           rgeos_0.5-9           
 [43] gtable_0.3.0           globals_0.15.0         goftest_1.2-3         
 [46] workflowr_1.7.0        rlang_1.0.2            splines_4.1.0         
 [49] rstatix_0.7.0          lazyeval_0.2.2         spatstat.geom_2.4-0   
 [52] broom_0.8.0            yaml_2.3.5             reshape2_1.4.4        
 [55] abind_1.4-5            modelr_0.1.8           backports_1.4.1       
 [58] httpuv_1.6.5           tools_4.1.0            ellipsis_0.3.2        
 [61] spatstat.core_2.4-2    jquerylib_0.1.4        RColorBrewer_1.1-3    
 [64] ggridges_0.5.3         Rcpp_1.0.8.3           plyr_1.8.7            
 [67] zlibbioc_1.38.0        RCurl_1.98-1.6         rpart_4.1.16          
 [70] deldir_1.0-6           pbapply_1.5-0          cowplot_1.1.1         
 [73] zoo_1.8-10             haven_2.5.0            ggrepel_0.9.1         
 [76] cluster_2.1.3          fs_1.5.2               data.table_1.14.2     
 [79] RSpectra_0.16-1        scattermore_0.8        lmtest_0.9-40         
 [82] reprex_2.0.1           RANN_2.6.1             whisker_0.4           
 [85] fitdistrplus_1.1-8     hms_1.1.1              patchwork_1.1.1       
 [88] mime_0.12              evaluate_0.15          xtable_1.8-4          
 [91] readxl_1.4.0           gridExtra_2.3          compiler_4.1.0        
 [94] KernSmooth_2.23-20     crayon_1.5.1           htmltools_0.5.2       
 [97] mgcv_1.8-40            later_1.3.0            tzdb_0.3.0            
[100] lubridate_1.8.0        DBI_1.1.2              dbplyr_2.1.1          
[103] MASS_7.3-57            Matrix_1.4-1           car_3.0-13            
[106] cli_3.3.0              igraph_1.3.1           pkgconfig_2.0.3       
[109] plotly_4.10.0          spatstat.sparse_2.1-1  xml2_1.3.3            
[112] bslib_0.3.1            XVector_0.32.0         rvest_1.0.2           
[115] digest_0.6.29          RcppAnnoy_0.0.19       spatstat.data_2.2-0   
[118] rmarkdown_2.14         cellranger_1.1.0       leiden_0.3.10         
[121] uwot_0.1.11            shiny_1.7.1            lifecycle_1.0.1       
[124] nlme_3.1-157           jsonlite_1.8.0         carData_3.0-5         
[127] limma_3.48.3           fansi_1.0.3            pillar_1.7.0          
[130] lattice_0.20-45        fastmap_1.1.0          httr_1.4.3            
[133] survival_3.3-1         glue_1.6.2             png_0.1-7             
[136] bit_4.0.4              stringi_1.7.6          sass_0.4.1            
[139] irlba_2.3.5            future.apply_1.9.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
